---
title: "Analysis of Haifeng 3D/3D culture samples"
output:
  html_notebook:
      code_folding: none
---

This document details the analysis of proteomics data acquired to look at A673 cells in 3D and 3D growth conditions with modulation of EWS-FLI1.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning=FALSE}
##########################################################################################
library('tidyverse')
library('DEqMS')
library('ggplot2')
library('RColorBrewer')
library('ggrepel')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToOthers'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToOthers'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

Read the protein annotation details.

```{r}
##########################################################################################
proteinAnnotation = readRDS(paste(baseRepository, '/proteomics20201115_haifengA673Dox2d3dSetB/uniprotHumanNov2020.fasta.annotated.rds', sep = ''))
```

<div style="margin-bottom:50px;"></div>

Read in the psm data and process into a peptide set.

```{r}
##########################################################################################
psm = read_tsv(paste(baseRepository, '/proteomics20201115_haifengA673Dox2d3dSetB/20201115_HaifengA673Dox2d3dSetB_TMT16_HpH_Default_PSM_Report.txt', sep = '')) %>%
  dplyr::select(`Protein(s)`, Sequence, `Modified Sequence`, `Spectrum File`, `Spectrum Title`) %>%
  mutate(fraction = factor(sub('.*HpH_(.*)\\.raw$', '\\1', `Spectrum File`), levels = seq(1,36,1))) %>%
  mutate(accession = sapply(strsplit(`Protein(s)`, ','), '[', 1)) %>%
  mutate(scan = as.numeric(sub('Spectrum_(.*)', '\\1', `Spectrum Title`))) %>%
  mutate(sequence = Sequence) %>%
  mutate(modSequence = `Modified Sequence`) %>%
  left_join(proteinAnnotation) %>%
  dplyr::select(fraction, scan, accession, gene, detectablePeptides, sequence, modSequence)
```

<div style="margin-bottom:50px;"></div>

Now we need the quantification data. I will use a function to process these data.

```{r}
#######################################################################
#######################################################################
##this function goes through the provided quant files and returns
##a parsed object for each. It returns signal to noise ratio for quant
#######################################################################
combineQuantFiles = function(filePath, ...){
  quantData = read_tsv(filePath) %>%
    dplyr::select(MS2ScanNumber, `126Intensity`:`134NIntensity`)
  colnames(quantData) = c('scan','tmt16plex_126C','tmt16plex_127N','tmt16plex_127C','tmt16plex_128N',
                  'tmt16plex_128C','tmt16plex_129N','tmt16plex_129C','tmt16plex_130N','tmt16plex_130C','tmt16plex_131N',
                  'tmt16plex_131C','tmt16plex_132N','tmt16plex_132C','tmt16plex_133N','tmt16plex_133C','tmt16plex_134N')
  ##
  fraction = sub('.*HpH_(.*)\\.raw_Matrix\\.txt$', '\\1', filePath)
  quantData$fraction = fraction
  print(paste('Processing file for fraction ', fraction, '.', sep = ''))
  ##
  return(quantData)
}
```

<div style="margin-bottom:50px;"></div>

We can use this function to parse the quant data into a single object.

```{r, message = FALSE, warning = FALSE}
##########################################################################################
quantFiles = as.list(list.files(paste(baseRepository, '/proteomics20201115_haifengA673Dox2d3dSetB/quantFiles/', sep = ''),
           pattern = '_Matrix.txt', full.names = TRUE))
##
quantDataSet = lapply(quantFiles, combineQuantFiles)
```

<div style="margin-bottom:50px;"></div>

Now combine the quant data into a single data frame and with the previously processed PSM data.

```{r}
##########################################################################################
allQuantData = do.call('rbind', quantDataSet)
psmQuant = psm %>%
  left_join(allQuantData)
```

<div style="margin-bottom:50px;"></div>

Now we will filter the expression data to discard any entries where the signal is just too low to be confident in the data. I generally will use a cutoff of a minimum signal of 10 per TMT channel. In this case we have 11 channels, so we will use a sum signal of 110 as a cutoff. Here we also filter out genes that have no assigned gene name as this will cause problems with DEqMS later on. If you don't want to do this, I suggest using 'accession' instead of 'gene' in the DEqMS analysis below.

Our sample layout is:

* 126C - pool
* 127N - empty
* 127C - empty
* 128N - noDox.1
* 128C - noDox.2
* 129N - noDox.3
* 129C - yesDox.1
* 130N - yesDox.2
* 130C - yesDox.3
* 131N - empty
* 131C - empty
* 132N - washOut.1
* 132C - washOut.2
* 133N - empty
* 133C - washOut.3
* 134N - pool

```{r}
##########################################################################################
psmQuant$sampleSignal = rowSums(psmQuant[,c(8,11:16,19:20,22:23)])
psmQuantFiltered = subset(psmQuant, psmQuant$sampleSignal >= 110 & !is.na(psmQuant$gene) & !grepl('-', psmQuant$gene))
```

<div style="margin-bottom:50px;"></div>

We will also normalize the data to deal with any gross loading differences. We save both the non-normalized data here as well and check the normalization with a plot.

```{r}
##########################################################################################
quantInput = psmQuantFiltered[,c(1:7,11:16,19:20,22)]
saveRDS(quantInput, paste(baseRepository, '/proteomics20201115_haifengA673Dox2d3dSetB/dataset_peptideQuantDataPreDEqMS.rds', sep = ''))
quantInputLog = quantInput[,c(7,4,8:16)]
quantInputLog[,3:11][quantInputLog[,3:11] == 0] = NA
quantInputLog[,3:11] = log2(quantInputLog[,3:11])
quantInputNormalized = medianSweeping(quantInputLog, group_col = 2)
boxplot(quantInputNormalized, las = 2, ylab = 'log2 ratio', main = 'normalized TMT data')
```

<div style="margin-bottom:50px;"></div>

All of the data look tight. Good. Lets continue with the DEqMS analysis. We need to build our sample table.

```{r}
##########################################################################################
cond = as.factor(c('noDox','noDox','noDox','yesDox','yesDox','yesDox','washOut','washOut','washOut'))
design = model.matrix(~0+cond) 
colnames(design) = gsub('cond', '', colnames(design))
```

<div style="margin-bottom:50px;"></div>

Now we can build the Limma model.

```{r}
##########################################################################################
geneMatrix = as.matrix(quantInputNormalized)
limmaFit1 = lmFit(geneMatrix, design)
limmaContrasts = c('yesDox - noDox', 'washOut - noDox', 'yesDox - washOut')
limmaContrastDesign =  makeContrasts(contrasts = limmaContrasts, levels = design)
limmaFit2 = eBayes(contrasts.fit(limmaFit1, contrasts = limmaContrastDesign))
```

<div style="margin-bottom:50px;"></div>

Now for the actual DEqMS work. We will first

```{r}
##########################################################################################
psmCountTable = as.data.frame(table(quantInput$gene))
rownames(psmCountTable) = psmCountTable$Var1
limmaFit2$count = psmCountTable[rownames(limmaFit2$coefficients),2]
limmaFit3 = spectraCounteBayes(limmaFit2)
head(limmaFit3$coefficients)
##
deqmsResults = outputResult(limmaFit3, coef_col = 1)
dim(deqmsResults)
write.table(deqmsResults, 
            paste(baseRepository, '/proteomics20201115_haifengA673Dox2d3dSetB/dataset_deqms_3d_yesDox-noDox.csv', sep = ''),
            sep = ',', row.names = FALSE, quote = FALSE)
##
deqmsResults = outputResult(limmaFit3, coef_col = 2)
dim(deqmsResults)
write.table(deqmsResults, 
            paste(baseRepository, '/proteomics20201115_haifengA673Dox2d3dSetB/dataset_deqms_3d_washOut-noDox.csv', sep = ''),
            sep = ',', row.names = FALSE, quote = FALSE)
##
deqmsResults = outputResult(limmaFit3, coef_col = 3)
dim(deqmsResults)
write.table(deqmsResults, 
            paste(baseRepository, '/proteomics20201115_haifengA673Dox2d3dSetB/dataset_deqms_3d_yesDox-washOut.csv', sep = ''),
            sep = ',', row.names = FALSE, quote = FALSE)
```

<div style="margin-bottom:50px;"></div>

Generate the variance plot to see the DEqMS model for the data.

```{r}
##########################################################################################
VarianceBoxplot(limmaFit3, n=20, xlab = 'PSM count', main = 'DEqMS analysis of TMT data')
```

<div style="margin-bottom:50px;"></div>

This looks as it should. Now let us make a volcano plot to show the data.

```{r}
##########################################################################################
deqmsResults = read_csv(paste(baseRepository, '/proteomics20201115_haifengA673Dox2d3dSetB/dataset_deqms_3d_yesDox-noDox.csv', sep = ''))
#deqmsResults = outputResult(limmaFit3, coef_col = 1)
deqmsResults$logFC = -deqmsResults$logFC
deqmsResults$plotPValue = -log10(deqmsResults$sca.P.Value)
deqmsResults$pColors = ifelse(deqmsResults$logFC <= -0.5 & deqmsResults$plotPValue >= 3, brewer.pal(4,'GnBu')[4],
                              ifelse(deqmsResults$logFC >= 0.5 & deqmsResults$plotPValue >= 3, brewer.pal(4,'OrRd')[4],
                                     brewer.pal(3,'Greys')[2]))
goi = c('IL1RAP','PRKCB','LOX')
deqmsResults$pText = ifelse(deqmsResults$gene %in% goi, deqmsResults$gene, '')
##
ggplot(deqmsResults, aes(logFC, plotPValue)) +
  geom_point(size = 1, color = deqmsResults$pColors) +
  geom_text_repel(label = deqmsResults$pText, nudge_x = 2, nudge_y = 1, max.overlaps = 10000) +
  labs(x = 'log2(EWS-FLI1 high vs. low)', y = '-log10(P-value)', title = '3D culture protein abundance') +
  scale_y_continuous(limits = c(0,12), breaks = seq(0,20,3)) +
  scale_x_continuous(limits = c(-6,6), breaks = seq(-6,6,2)) +
  geom_vline(xintercept = c(-0.5,0.5), linetype = 'dashed') +
  geom_hline(yintercept = 3, linetype = 'dashed') +
  theme_classic()
ggsave(paste(baseRepository, '/proteomics20201115_haifengA673Dox2d3dSetB/scatter_deqms_3d_yesDox-noDox.pdf', sep = ''),
       height = 2, width = 2, useDingbats = FALSE)




##volcano plot in our usualy style
deqmsResults$pColors = ifelse(deqmsResults$logFC <= -0.5 & deqmsResults$plotPValue >= 3, brewer.pal(4,'GnBu')[4],
                              ifelse(deqmsResults$logFC >= 0.5 & deqmsResults$plotPValue >= 3, brewer.pal(4,'OrRd')[4],
                                     brewer.pal(3,'Greys')[2]))
#goi = c('PRKCB','LOX','DLG2')
goi = c('IL1RAP','STEAP1','STEAP2','PRKCB','LOX','DLG2')
deqmsResults$pText = ifelse(deqmsResults$symbol %in% goi, deqmsResults$symbol, '')
##
ggplot(deqmsResults, aes(logFC, plotPValue)) +
  geom_point(size = 1, color = deqmsResults$pColors) +
  geom_text_repel(label = deqmsResults$pText, nudge_x = 2, nudge_y = 1, max.overlaps = 10000) +
  labs(x = 'log2(day0 - day7)', y = '-log10(P-value)', title = 'Differential protein abundance') +
  scale_y_continuous(limits = c(0,12), breaks = seq(0,20,4)) +
  scale_x_continuous(limits = c(-6,6), breaks = seq(-6,6,2)) +
  geom_vline(xintercept = c(-0.5,0.5), linetype = 'dashed') +
  geom_hline(yintercept = 3, linetype = 'dashed') +
  theme_classic()




##########################################################################################
deqmsResults = outputResult(limmaFit3, coef_col = 2)
deqmsResults$plotPValue = -log10(deqmsResults$sca.P.Value)
deqmsResults$pColors = ifelse(deqmsResults$logFC <= -0.5 & deqmsResults$plotPValue >= 3, brewer.pal(8,'RdBu')[8],
                              ifelse(deqmsResults$logFC >= 0.5 & deqmsResults$plotPValue >= 3, brewer.pal(8,'RdBu')[1],
                                     brewer.pal(8,'Greys')[6]))
deqmsResults$pText = ifelse(grepl('IL1RAP$',deqmsResults$gene), deqmsResults$gene, '')
##
ggplot(deqmsResults, aes(logFC, plotPValue)) +
  geom_point(size = 1, color = deqmsResults$pColors, alpha = 0.5) +
  geom_text_repel(label = deqmsResults$pText, nudge_x = -2) +
  labs(x = 'log2(washOut - noDox)', y = '-log10(P-value)', title = '3D washOut vs noDox protein abundance') +
  scale_y_continuous(limits = c(0,12), breaks = seq(0,18,3)) +
  scale_x_continuous(limits = c(-6,6), breaks = seq(-6,6,2)) +
  geom_vline(xintercept = c(-0.5,0.5), linetype = 'dashed') +
  geom_hline(yintercept = 3, linetype = 'dashed') +
  theme_classic()
ggsave(paste(baseRepository, '/proteomics20201115_haifengA673Dox2d3dSetB/scatter_deqms_3d_washOut-noDox.pdf', sep = ''),
       height = 4, width = 4, useDingbats = FALSE)


##########################################################################################
deqmsResults = outputResult(limmaFit3, coef_col = 3)
deqmsResults$plotPValue = -log10(deqmsResults$sca.P.Value)
deqmsResults$pColors = ifelse(deqmsResults$logFC <= -0.5 & deqmsResults$plotPValue >= 3, brewer.pal(8,'RdBu')[8],
                              ifelse(deqmsResults$logFC >= 0.5 & deqmsResults$plotPValue >= 3, brewer.pal(8,'RdBu')[1],
                                     brewer.pal(8,'Greys')[6]))
deqmsResults$pText = ifelse(grepl('IL1RAP$',deqmsResults$gene), deqmsResults$gene, '')
##
ggplot(deqmsResults, aes(logFC, plotPValue)) +
  geom_point(size = 1, color = deqmsResults$pColors, alpha = 0.5) +
  geom_text_repel(label = deqmsResults$pText, nudge_x = -2) +
  labs(x = 'log2(yesDox - washOut)', y = '-log10(P-value)', title = '3D yesDox vs washOut protein abundance') +
  scale_y_continuous(limits = c(0,12), breaks = seq(0,18,3)) +
  scale_x_continuous(limits = c(-6,6), breaks = seq(-6,6,2)) +
  geom_vline(xintercept = c(-0.5,0.5), linetype = 'dashed') +
  geom_hline(yintercept = 3, linetype = 'dashed') +
  theme_classic()
ggsave(paste(baseRepository, '/proteomics20201115_haifengA673Dox2d3dSetB/scatter_deqms_3d_yesDox-washOut.pdf', sep = ''),
       height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

## Session info

```{r}
sessionInfo()
```




