---
title: "osteosarcoma RNAseq"
author: "Christopher Hughes"
format: html
editor: visual
---

This document details the analysis of RNAseq data from osteosarcoma patients as part of the TARGET project. The first think I had to do was to download the counts data from [GDC](https://portal.gdc.cancer.gov/repository?facetTab=files&filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.program.name%22%2C%22value%22%3A%5B%22TARGET%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.project_id%22%2C%22value%22%3A%5B%22TARGET-OS%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.access%22%2C%22value%22%3A%5B%22open%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.analysis.workflow_type%22%2C%22value%22%3A%5B%22STAR%20-%20Counts%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.experimental_strategy%22%2C%22value%22%3A%5B%22RNA-Seq%22%5D%7D%7D%5D%7D). You can do this by just adding all of those files to your cart and clicking download. You should also download the 'sample sheet' as well as we will use it for annotation and file parsing.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r}
#| message: false
##########################################################################################
library('tidyverse')
library('tximport')
library('RColorBrewer')
library('ggplot2')
library('ggrepel')
```

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToOthers'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToOthers'
```

Now we are ready to proceed with the analysis.

## Data processing

The GDC data is provided in individual folders instead of on a file-by-file basis, so first we will build a sample table in order to grab these data.

```{r}
##########################################################################################
##read in the sample info sheet
sampleInfo = read_tsv(paste(baseRepository, '/sequencing20220629_osteosarcomaRnaSeqForBrian/gdc_sample_sheet.2022-06-29.tsv', sep = ''), 
                      show_col_types = FALSE) %>%
  dplyr::rename(folderId = `File ID`, fileId = `File Name`, caseId = `Case ID`) %>%
  dplyr::select(folderId, fileId, caseId) %>%
  dplyr::mutate(filePath = file.path(baseRepository, 
                                     'sequencing20220629_osteosarcomaRnaSeqForBrian',
                                     folderId,
                                     fileId))

##take a look at the data
sampleInfo

##check that all the files in the paths exist
all(file.exists(sampleInfo$filePath))
```

Now we can read in the data and process it for anything of interest. 

```{r}
#| message: false
#| warning: false
##########################################################################################
##read in the rna data and reformat it
##I select the tpm values here, but there are other options in these files (e.g. fpkm, or raw counts)
rnaData = tibble()
for (i in 1:nrow(sampleInfo)){
  rnaTemp = read_tsv(sampleInfo$filePath[i], col_names = FALSE, skip = 6, show_col_types = FALSE) %>%
    dplyr::select(X1,X2,X7) %>%
    dplyr::rename(ensg = X1,
                  symbol = X2,
                  tpm = X7) %>%
    dplyr::mutate(sampleId = sampleInfo$caseId[i])
  ##
  rnaData = rbind(rnaData, rnaTemp)
}

##
rnaData

##save the data
saveRDS(rnaData, paste(baseRepository, '/sequencing20220629_osteosarcomaRnaSeqForBrian/dataset_targetOsteoTpmValues.rds', sep = ''))
write.table(rnaData, paste(baseRepository, '/sequencing20220629_osteosarcomaRnaSeqForBrian/dataset_targetOsteoTpmValues.tsv', sep = ''),
            col.names = TRUE, row.names = FALSE, quote = FALSE, sep = '\t')
```

This looks OK. Now we can look at genes of interest, in this case, TMEM119.

```{r}
##########################################################################################
##read the data for our gene of interest
genesOfInterest = c('TMEM119','EIF4A1','YBX1', 'DLG2')
goi = readRDS(paste(baseRepository, '/sequencing20220629_osteosarcomaRnaSeqForBrian/dataset_targetOsteoTpmValues.rds', sep = '')) %>%
  dplyr::filter(symbol %in% genesOfInterest)

##plot the data
ggplot(goi, aes(x = symbol, y = log2(tpm + 1), fill = symbol)) +
  geom_point(size = 1.5, color = brewer.pal(3,'Greys')[2], position = position_jitterdodge(jitter.width = 0.2)) +
  geom_boxplot(width = 0.2, outlier.shape = NA) +
  scale_fill_manual(values = rep(brewer.pal(3,'Spectral')[3],4)) +
  labs(x = 'gene', y = 'log2(TPM + 1)', title = 'TARGET osteo RNA') +
  theme_classic() +
  theme(legend.position = 'none',
        axis.text.x = element_text(angle = 45, hjust = 1))
```

Done for now.

## Session info

```{r}
##########################################################################################
sessionInfo()
```
