---
title: "Coverage maps for DGL2"
output:
  html_notebook:
      code_folding: none
---

This document details working with aligned RNAseq data to create coverage maps for DLG2.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning = FALSE}
##########################################################################################
library('tidyverse')
library('ggplot2')
library('RColorBrewer')
library('Rsamtools')
library('rtracklayer')
library('GenomicAlignments')
library('bamsignals')
library('wiggleplotr')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

First I extract the genomic range of all annotated exons for DLG2 from the hg38 GTF file we used in alignment. I then save this object for later use.

```{r}
##########################################################################################
gtf = rtracklayer::import('C:/Users/chughes/Documents/bccrc/databases/hg38/gencode_gtf/hg38.gtf')
dlg2Gtf = as.data.frame(gtf) %>%
  filter(grepl('DLG2', gene_name) & grepl('exon', type) & grepl('protein_coding', transcript_type))
saveRDS(dlg2Gtf, paste(baseRepository, '/sequencing20200107_humanProteinAtlasData/dlg2GtfAllExons.rds', sep = ''))
```

<div style="margin-bottom:50px;"></div>

Read in the previous created GTF file and process the exon data to create a non-redudant set of exons.

```{r}
##########################################################################################
exonRangesRaw = readRDS(paste(baseRepository, '/sequencing20200107_humanProteinAtlasData/dlg2GtfAllExons.rds', sep = ''))
exonRangesGr = GRanges(exonRangesRaw)
exonRangesCollapse = as.data.frame(reduce(exonRangesGr)) %>%
  mutate(name = 'DLG2') %>%
  dplyr::select(seqnames, start, end, name, strand)
write.table(exonRangesCollapse, paste(baseRepository, '/sequencing20200107_humanProteinAtlasData/dlg2GtfCollapsedExons.bed', sep = ''),
            quote = FALSE, col.names = FALSE, row.names = FALSE, sep = '\t')
```

<div style="margin-bottom:50px;"></div>

Using this bed regions file, I ran mosdepth on the regions as described elsewhere in this folder, and I will read these data in now and process them for plotting purposes. 

```{r}
##########################################################################################
##this function goes through the provided coverage files
##and outputs a parsed object
#######################################################################
combineMosdepthFiles = function(filePath, ...){
  mosdepthData = read_tsv(filePath, col_names = c('seqname', 'start', 'end', 'symbol', 'coverage')) %>%
    mutate('accession' = sub('.*mosdepthAnalysis\\/(.*)\\.regions\\.bed\\.gz', '\\1', filePath)) %>%
    mutate('exon' = seq(43,1,-1)) %>%
    mutate('relativeCoverage' = coverage / max(coverage))
  print(paste('Processing file for fraction ', sub('.*mosdepthAnalysis\\/(.*)\\.regions\\.bed\\.gz', '\\1', filePath), '.', sep = ''))
  ##
  return(mosdepthData)
}

##get the list of input files
filePath = list.files(paste(baseRepository, '/sequencing20200107_humanProteinAtlasData/mosdepthAnalysis/', sep = ''), 
                      pattern = '.regions.bed.gz$', full.names = TRUE)
##now proces the data and combine into a single set
coverageSet = do.call('rbind', lapply(filePath, combineMosdepthFiles))
```

<div style="margin-bottom:50px;"></div>

Now we are ready to make some plots. I think best is probably a boxplot, so let's try it out.

```{r}
##########################################################################################
coverageSummary = coverageSet %>%
  group_by(exon) %>%
  summarise(mean.cov = median(relativeCoverage, na.rm = TRUE),
            sd.cov = sd(relativeCoverage, na.rm = TRUE),
            n.cov = n())
##
ggplot(coverageSet, aes(exon, relativeCoverage, group = exon)) +
  geom_boxplot(size = 0.5, width = 0.5, outlier.shape = NA) +
  geom_point(data = coverageSummary, aes(exon, mean.cov, group = 1), color = brewer.pal(6,'Spectral')[1], size = 0.5) +
  geom_path(data = coverageSummary, aes(exon, mean.cov, group = 1), color = brewer.pal(6,'Spectral')[1], size = 0.5) +
  scale_y_continuous(limits = c(0,1.2), breaks = seq(0,2,0.25)) +
  labs(x = 'Exon number', y = 'Coverage', title = 'DLG2 coverage, Brain') +
  theme_classic()
ggsave(paste(baseRepository, '/sequencing20200107_humanProteinAtlasData/mosdepthAnalysis/boxplot_dlg2CoverageBrainRna.pdf', sep = ''),
         height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

This might look better as a line plot?

```{r}
##########################################################################################
test = coverageSet %>% 
  group_by(exon) %>%
  summarise(mean.cov = mean(coverage, na.rm = TRUE),
            sd.cov = sd(coverage, na.rm = TRUE),
            n.cov = n()) %>%
  mutate(se.cov = sd.cov / sqrt(n.cov),
         lower.ci.cov = mean.cov - qt(1 - (0.05 / 2), n.cov - 1) * se.cov,
         upper.ci.cov = mean.cov + qt(1 - (0.05 / 2), n.cov - 1) * se.cov)
ggplot(test, aes(exon, mean.cov)) +
  geom_path() +
  geom_ribbon(aes(ymin = lower.ci.cov, ymax = upper.ci.cov), linetype = 2, alpha = 0.5)
```

<div style="margin-bottom:50px;"></div>

Hmm. Not really. It is nice to be able to see the individual exons as you can in the boxplot.

I am going to revisit these plots. I think the boxplots are Ok, but they are a bit confusing. I think we should just pick an isoform and show coverage from a bigwig file because the important thing isn't that there are a bunch of isoforms, but rather that there is an N-terminal truncation. We want to highlight that.

First, read the GTF file that we can use to get the DLG2 annotation details.

```{r}
##########################################################################################
totalGtf = as.data.frame(rtracklayer::import.gff('C:/Users/chughes/Documents/bccrc/databases/hg38/gencode_gtf/hg38.gtf.gz'))
```

<div style="margin-bottom:50px;"></div>

Now we will create the annotation matrix input required for the wiggleplot. We use a single transcript here (the main isoform for DLG2) just for simplicity.

```{r}
##########################################################################################
dlg2Gtf = totalGtf %>%
  filter(grepl('ENST00000398309.6', transcript_id))
##
dlg2Meta = dlg2Gtf %>%
  filter(grepl('transcript', type)) %>%
  dplyr::select(transcript_id, gene_id, gene_name, strand)
##
dlg2Exons = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('exon', type)), split.field = 'transcript_id')
##
dlg2Cds = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('CDS', type)), split.field = 'transcript_id')
##
sampleData = tibble(sample_id = c('brain1','brain2','brain3'), 
                    condition = factor(c('brain','brain','brain'), levels = c('brain')),
                    scaling_factor = 1)
##
bigwigFiles = c(paste(baseRepository, '/sequencing20200107_humanProteinAtlasData/ERR315432.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20200107_humanProteinAtlasData/ERR315455.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20200107_humanProteinAtlasData/ERR315477.chr11.bw', sep = ''))
##
sampleData = sampleData %>%
  dplyr::mutate(bigWig = bigwigFiles)
as.data.frame(sampleData)
##
trackData = dplyr::mutate(sampleData, track_id = condition, colour_group = condition)
#trackData = dplyr::mutate(sampleData, track_id = sample_id, colour_group = condition)
##
plotCoverage(dlg2Exons, dlg2Cds, 
             dlg2Meta, trackData,
             heights = c(2,1), fill_palette = getGenotypePalette(),
             rescale_introns = TRUE)
ggsave(paste(baseRepository, '/sequencing20200107_humanProteinAtlasData/boxplot_dlg2CoverageBrainRna.pdf', sep = ''),
         height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

## Session info

```{r}
##########################################################################################
sessionInfo()
```











