---
title: "Coverage maps for DGL2"
output:
  html_notebook:
      code_folding: none
---

This document details working with aligned RNAseq data to create coverage maps for DLG2.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning = FALSE}
##########################################################################################
library('tidyverse')
library('ggplot2')
library('RColorBrewer')
library('Rsamtools')
library('rtracklayer')
library('GenomicAlignments')
library('bamsignals')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

I think the easiest way to do this might be to create a BamViews object. First we need to get the genomic range for DLG2.

```{r}
##########################################################################################
gtf = rtracklayer::import('C:/Users/chughes/Documents/bccrc/databases/hg38/gencode_gtf/hg38.gtf')
dlg2Gtf = as.data.frame(gtf) %>%
  filter(grepl('ENST00000398309.6', transcript_id)) %>%
  filter(grepl('transcript', type))
dlg2Range = makeGRangesFromDataFrame(dlg2Gtf)
seqlevels(dlg2Range) = "chr11"
```

<div style="margin-bottom:50px;"></div>

Now lets see if we can assemble a bamviews object using these ranges.

```{r}
##########################################################################################
bamInputFiles = list.files(paste(baseRepository, '/sequencing20200107_humanProteinAtlasData/', sep = ''), pattern = '*.bam$', full.names = TRUE)

##
dlg2BamView = BamViews(bamPaths = bamInputFiles, 
                       bamIndicies = bamInputFiles,
                       bamRanges = dlg2Range)


which <- dlg2Range
what <- c("rname", "strand", "pos", "qwidth", "seq")
param <- ScanBamParam(which = which, what = what)     
bamFile <- bamInputFiles[1]
bam <- scanBam(bamFile, param=param)   



which <- GRanges(seqnames = c("chr11"), ranges = IRanges(c(83454886), c(83468831)))

bampath <- bamInputFiles[1]
features2 = unnest_parts(features) %>%
  filter(grepl('exon', feature_type))
sigs <- bamProfile(bampath, features2, verbose=FALSE)
x = tibble('readCoverage' = unlist(as.list(sigs)))
x$position = seq(1, nrow(x), 1)
ggplot(x, aes(position, readCoverage)) +
  geom_path()


c(83454886), c(83468831)
c(83460521), c(83464006)
```



```{r}
#BiocManager::install('remotes')
#BiocManager::install("sa-lee/superintronic")
library('superintronic')

#
features <- plyranges::read_gff('C:/Users/chughes/Documents/bccrc/databases/hg38/gencode_gtf/hg38.gtf') %>% 
  collect_parts() %>% 
  filter(gene_name == "DLG2")


#
bamInputFiles = list.files(paste(baseRepository, '/sequencing20200107_humanProteinAtlasData/', sep = ''), pattern = '*.bam$', full.names = TRUE)
design <- tibble('sample' = c('ERR315432','ERR315455','ERR315477')) %>%
  mutate(bam = bamInputFiles)

cvg <- compute_coverage_long(design, source = "bam", .parallel = MulticoreParam(3))




##need to do this on the linux server because of BiocParallel it seems
library('tidyverse')
library('BiocParallel')
library('superintronic')
library('GenomeInfoDb')
library('GenomicRanges')
features <- plyranges::read_gff('/projects/ptx_analysis/chughes/databases/refgenieIndexes/hg38/gencode_gtf/default/hg38.gtf') %>% 
  collect_parts() %>% 
  dplyr::filter(gene_name == "DLG2")
bamInputFile = '/projects/ptx_results/Sequencing/publishedStudies/humanProteinAtlasRnaSequencingData/starResults/ERR315432.sorted.bam'
design = tibble('sample' = c('ERR315432')) %>%
  dplyr::mutate(bam = bamInputFile)
cvg = compute_coverage_long(design, source = 'bam')
cvgStd = keepStandardChromosomes(cvg, pruning.mode = 'coarse')
cvg_over_features = cvgStd %>% 
  dplyr::select(-bam) %>% 
  join_parts(features)

saveRDS(cvg_over_features, '/projects/ptx_results/Sequencing/publishedStudies/humanProteinAtlasRnaSequencingData/starResults/ERR315432.dlg2.coverage.rds')
cvg_over_features = readRDS(paste(baseRepository, '/sequencing20200107_humanProteinAtlasData/ERR315432.dlg2.coverage.rds', sep = ''))

p = cvg_over_features %>%
  filter(feature_rank < 7) %>%
  mutate(strand = feature_strand) %>% 
  view_coverage(score = score, colour = feature_type) + 
  scale_color_brewer(palette = "Dark2") +
  guides(colour = FALSE) +
  labs(title = "Coverage over SRM")
p
gene_track <- view_segments(unnest_parts(features)[1:12], 
                            colour = feature_type)
gene_track
p / gene_track

exons = tibble(as.data.frame(cvg_over_features)) %>%
  filter(grepl('exon', feature_type))
exons$index = seq(1,nrow(exons),1)
ggplot(exons, aes(index, score)) +
  geom_path()
  
```









































