---
title: "Analysis of A673 EF1 samples with IRS"
output:
  html_notebook:
      code_folding: none
---

This document details the analysis of proteomics data acquired to look at A673 cells in 2D growth conditions with modulation of EWS-FLI1 across a timecourse.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning=FALSE}
##########################################################################################
library('tidyverse')
library('RColorBrewer')
library('ggplot2')
library('tximport')
library('GenomicFeatures')
library('DESeq2')
library('org.Hs.eg.db')
library('ggrepel')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

Read in the sample annotation details and make the txdb that we can use later on for annotation gene identifiers.

```{r, message = FALSE}
##########################################################################################
samples = read_tsv(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/sampleInfo.txt', sep = ''))

##
files = file.path(baseRepository, 'sequencing20210421_a673TimecourseRnaSeqOutput', paste(samples$barcode, '_', samples$batch, '_quant', sep = ''), 'quant.sf')
all(file.exists(files))

##
myTxdb = makeTxDbFromGFF('C:/Users/chughes/Documents/bccrc/databases/hg38/gencode_gtf/2230c535660fb4774114bfa966a62f823fdb6d21acf138d4.gtf.gz')
k = keys(myTxdb, keytype = 'TXNAME')
tx2gene = AnnotationDbi::select(myTxdb, k, 'GENEID', 'TXNAME')
```

<div style="margin-bottom:50px;"></div>

Now import the Salmon data.

```{r}
##########################################################################################
txi = tximport(files, 
               type = 'salmon', 
               tx2gene = tx2gene)
names(txi)
head(txi$counts)

##
txiAbundance = as.data.frame(txi$abundance)
colnames(txiAbundance) = paste(samples$sampleName,'_',samples$batch, sep = '')
ens.str = substr(rownames(txiAbundance), 1, 15)
txiAbundance$symbol = mapIds(org.Hs.eg.db, keys = ens.str, column = 'SYMBOL', keytype = 'ENSEMBL', multiVals = 'first')
txiAbundanceAvg = as_tibble(txiAbundance) %>%
  dplyr::select(symbol, day0_setA:day22_setC) %>%
  pivot_longer(cols = day0_setA:day22_setC, names_to = 'sampleName', values_to = 'tpm') %>%
  dplyr::mutate(day = factor(sub('(.*)_set[ABC]$', '\\1', sampleName), levels = c('day0','day7','day9','day10','day11','day14','day17','day22'))) %>%
  dplyr::mutate(log2Tpm = log2(tpm + 1)) %>%
  dplyr::select(-sampleName) %>%
  group_by(symbol, day) %>%
  summarize(meanTpm = mean(log2Tpm, na.rm = TRUE)) %>%
  pivot_wider(id_cols = symbol, names_from = day, values_from = meanTpm)
saveRDS(txiAbundanceAvg, paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_txiAllSamplesSummarizedMeans.rds', sep = ''))

##
tpmCalc = function(x){
  log2Tpm = log2(x + 1)
}
txiAbundanceNoAvg = as_tibble(txiAbundance) %>%
  dplyr::select(symbol, day0_setA:day22_setC)
  #dplyr::mutate(across(c(day0_setA:day22_setC), tpmCalc))
saveRDS(txiAbundanceNoAvg, paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_txiAllSamples.rds', sep = ''))
```

<div style="margin-bottom:50px;"></div>

Construct the DESeq2 dataset.

```{r}
##########################################################################################
ddsTxi = DESeqDataSetFromTximport(txi,
                                  colData = samples,
                                  design = ~ sampleName)
##
dds = DESeq(ddsTxi)
keep = rowSums(counts(dds)) >= 10
dds = dds[keep,]
```

<div style="margin-bottom:50px;"></div>

Perform comparisons between samples.

```{r}
##########################################################################################
datasetFirst = 'day7'
datasetSecond = 'day0'
res = results(dds, contrast = c('sampleName', datasetFirst, datasetSecond))
ens.str = substr(rownames(res), 1, 15)
res$symbol = mapIds(org.Hs.eg.db,
                     keys=ens.str,
                     column="SYMBOL",
                     keytype="ENSEMBL",
                     multiVals="first")
resOrdered = res[order(res$pvalue),]

##
saveRDS(as.data.frame(resOrdered), 
        paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_deseq2_', datasetFirst, '-', datasetSecond, '.rds', sep = ''))
write.csv(as.data.frame(resOrdered), 
          file = paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_deseq2_', datasetFirst, '-', datasetSecond, '.csv', sep = ''))

##
#resLFC = lfcShrink(dds, coef = paste('treatment_',datasetFirst,'_vs_',datasetSecond, sep = ''), type = "apeglm")
#resLFC
#plotMA(resLFC, ylim = c(-2,2))
rnaExp = as.data.frame(resOrdered)
rnaExp$logPValue = -log10(rnaExp$padj)
#rnaExp$logPValueScaled = ifelse(rnaExp$logPValue > 50, 50, rnaExp$logPValue)
rnaExp$pColors = ifelse(rnaExp$padj <= 0.001 & rnaExp$log2FoldChange >= 1, brewer.pal(3,'OrRd')[3], 
                        ifelse(rnaExp$padj <= 0.001 & rnaExp$log2FoldChange <= -1, brewer.pal(3,'Blues')[3], brewer.pal(3,'Greys')[2]))
goi = c('DLG2','LOX','PRKCB')
rnaExp$pText = ifelse(rnaExp$symbol %in% goi, rnaExp$symbol, '')
##
ggplot(rnaExp, aes(log2FoldChange, logPValue)) +
  geom_point(size = 1, color = rnaExp$pColors, alpha = 0.75) +
  labs(x = paste('log2(',datasetFirst,' - ',datasetSecond,')', sep = ''), y = '-log10(Adjusted p-value)', title = paste(datasetFirst,' vs ',datasetSecond, sep='')) +
  geom_text_repel(label = rnaExp$pText, nudge_x = -3, nudge_y = -150, max.overlaps = 10000) +
  scale_x_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) +
  #scale_y_continuous(limits = c(0,20), breaks = seq(0,20,4)) +
  geom_vline(xintercept = c(-1,1), linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.001), linetype = 'dashed') +
  theme_classic()
##
ggsave(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/scatter_deseq2_', datasetFirst, '-', datasetSecond, '.pdf', sep = ''),
       height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

Compare the different samples using a PCA. 

```{r}
##########################################################################################
vsd = vst(dds, blind = FALSE) #variance stabilizing normalization of the data
#extract the data for the MDS analysis itself
mdsPlot = plotPCA(vsd, intgroup=c('sampleName','batch'), returnData = TRUE)
mdsPlot$sampleName = factor(mdsPlot$sampleName, levels = c('day0','day7','day9','day10','day11','day14','day17','day22'))
percentVar = round(100 * attr(mdsPlot, 'percentVar'))
#make the ggplot
ggplot(mdsPlot, aes(PC1, PC2, color = sampleName, shape = batch)) +
  geom_point(size = 6, alpha = 0.75) +
  scale_color_manual(values = rep(c(brewer.pal(3,'Blues')[3], brewer.pal(3,'OrRd')[3], brewer.pal(7,'Blues')[1:6]),3)) +
  scale_x_continuous(limits = c(-15,13), breaks = seq(-20,20,4)) +
  scale_y_continuous(limits = c(-5,15), breaks = seq(-20,20,4)) +
  labs(x = paste0('PC1: ', percentVar[1], '% variance'), 
       y = paste0('PC2: ', percentVar[2], '% variance'), 
       title = 'A673 RNA-seq groups') +
  theme_classic()
#save the plot
ggsave(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/pcaplot_a673ShEwsFli1AbundanceVsn.pdf', sep = ''), 
       height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

## Session info

```{r}
##########################################################################################
sessionInfo()
```



