---
title: "EwS signature gene set"
output:
  html_notebook:
      code_folding: none
---

This document details the comparative analysis of RNAseq and proteomics data from the A673 timecourse of EWS-FLI1 expression in order to derive a signature set of EWS genes.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning=FALSE}
##########################################################################################
library('tidyverse')
library('corrr')
library('vroom')
library('ggplot2')
library('RColorBrewer')
library('ggrepel')
library('BSgenome.Hsapiens.UCSC.hg38')
library('survival')
library('survminer')
library('pheatmap')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

Read in and combine the protein and RNA data.

```{r}
##########################################################################################
pro = read_csv(paste(baseRepository, '/proteomics20210421_a673TimecourseSearchOutput/dataset_deqms_day7-day0.csv', sep = '')) %>%
  dplyr::select(gene, logFC, sca.P.Value) %>%
  dplyr::rename(proFC = logFC, proPVal = sca.P.Value)
rna = read_csv(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_deseq2_day7-day0.csv', sep = '')) %>%
  dplyr::select(symbol, log2FoldChange, pvalue) %>%
  dplyr::rename(gene = symbol, rnaFC = log2FoldChange, rnaPVal = pvalue)

##
rnaPro = rna %>%
  left_join(pro)
rnaPro
```

<div style="margin-bottom:50px;"></div>

Read in and prepare the DepMap expression data.

```{r}
##########################################################################################
ccle = readRDS(paste(generalDatasets, '/depmap20Q4/CCLE_expression_parsed.rds', sep = ''))
```

<div style="margin-bottom:50px;"></div>

Extract the set of EWS regulated genes from the combined RNA and protein data.

```{r}
##########################################################################################
upSet = rnaPro %>%
  dplyr::filter(rnaFC < -2 & rnaPVal <= 0.001) %>%
  dplyr::filter(proFC < -2 & proPVal <= 0.001)

##
lineageOfInterest = c('Bone','Breast','Lung','Pancreas')
upCcle = ccle %>%
  #dplyr::filter(gene %in% upSet$gene & stripped_cell_line_name %in% testCells) %>%
  dplyr::filter(gene %in% upSet$gene & grepl('bone', lineage)) %>% 
  dplyr::select(gene, stripped_cell_line_name, log2Tpm) %>%
  pivot_wider(names_from = stripped_cell_line_name, values_from = log2Tpm) %>%
  dplyr::select(-gene) %>%
  correlate() %>%
  stretch()
  #network_plot(min_cor = 0.7, repel = TRUE, legend = FALSE)

graphCcle = upCcle %>%
  filter(abs(r) > 0.7) %>% 
  graph_from_data_frame(directed = FALSE)

ggraph(graphCcle) +
  geom_edge_link() +
  geom_node_point() +
  geom_node_text(aes(label = name), repel = TRUE) +
  theme_graph()

ggraph(graphCcle) +
  geom_edge_link(aes(edge_alpha = abs(r), edge_width = abs(r), color = r)) +
  guides(edge_alpha = "none", edge_width = "none") +
  scale_edge_colour_gradientn(limits = c(-1, 1), colors = c("firebrick2", "dodgerblue2")) +
  geom_node_point(color = "white", size = 5) +
  geom_node_text(aes(label = name), repel = TRUE, size = 3) +
  theme_classic() +
  labs(title = "Correlations between cells")
```

<div style="margin-bottom:50px;"></div>

What if we instead do sampling of the correlation using different gene sets. For example, if we take genes up in RNA samples and sample the correlation values between Ewing sarcoma and all other cancers?

```{r}
##########################################################################################
rna = read_csv(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_deseq2_day7-day0.csv', sep = '')) %>%
  dplyr::select(symbol, log2FoldChange, pvalue) %>%
  dplyr::rename(gene = symbol, rnaFC = log2FoldChange, rnaPVal = pvalue) %>%
  dplyr::filter(rnaFC <= -2 & rnaPVal <= 0.001) %>%
  left_join(ccle) %>%
  dplyr::select(gene, rnaFC, rnaPVal, log2Tpm, stripped_cell_line_name, lineage, lineage_subtype) %>%
  dplyr::filter(!is.na(stripped_cell_line_name))


##
ewsCells = unique(rna[grepl('Ewing', rna$lineage_subtype), 'stripped_cell_line_name']$stripped_cell_line_name)
boneCells = unique(rna[grepl('bone', rna$lineage), 'stripped_cell_line_name']$stripped_cell_line_name)
cellOutputData = data.frame()

set.seed(13579)
for (i in 1:10000){
  #message(i)
  ewsCellLocation = sample(seq(1,length(ewsCells)), 1)
  otherCells = unique(rna[rna$stripped_cell_line_name != ewsCells[ewsCellLocation], 'stripped_cell_line_name']$stripped_cell_line_name)
  otherCellLocation = sample(seq(1,length(otherCells)), 1)
  ##
  cellData = rna %>%
    dplyr::filter(stripped_cell_line_name == ewsCells[ewsCellLocation] | stripped_cell_line_name == otherCells[otherCellLocation]) %>%
    dplyr::select(gene, log2Tpm, stripped_cell_line_name) %>%
    pivot_wider(names_from = stripped_cell_line_name, values_from = log2Tpm) %>%
    dplyr::select(-gene)
  ##
  cellCorrelation = cor(cellData[,1], cellData[,2], use = 'pairwise.complete.obs', method = 'pearson')
  isEwing = ifelse(otherCells[otherCellLocation] %in% ewsCells, TRUE, FALSE)
  isBone = ifelse(otherCells[otherCellLocation] %in% boneCells, TRUE, FALSE)
  ##
  prepOutputData = data.frame('cellOne' = ewsCells[ewsCellLocation],
                              'cellTwo' = otherCells[otherCellLocation],
                              'correlation' = cellCorrelation[1],
                              'isEwing' = isEwing,
                              'isBone' = isBone)
  cellOutputData = rbind(cellOutputData, prepOutputData)
}
saveRDS(cellOutputData, paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_correlationRnaOnly.rds', sep = ''))
```

<div style="margin-bottom:50px;"></div>

Plot these data.

```{r}
##########################################################################################
corrData = readRDS(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_correlationRnaOnly.rds', sep = ''))
ggplot(corrData, aes(correlation, group = isEwing, fill = isEwing)) +
  geom_density(color = brewer.pal(11,'RdGy')[10], size = 0.5) +
  scale_fill_manual(values = brewer.pal(8,'Spectral')[c(1,8)]) +
  labs(x = 'Pearson correlation', y = 'Density', title = 'RNA expression signature') +
  geom_vline(xintercept = c(median(corrData[corrData$isEwing == TRUE, 'correlation'], na.rm = TRUE))) +
  geom_vline(xintercept = c(median(corrData[corrData$isEwing == FALSE, 'correlation'], na.rm = TRUE))) +
  scale_x_continuous(limits = c(-0.25,1), breaks = seq(-1,1,0.2)) +
  theme_classic() +
  theme(legend.position = 'none')
ggsave(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/density_correlationRnaOnly.pdf', sep = ''),
       height = 4, width = 4, useDingbats = FALSE)
table(corrData$isEwing)
median(corrData[corrData$isEwing == TRUE, 'correlation'], na.rm = TRUE)
median(corrData[corrData$isEwing == FALSE, 'correlation'], na.rm = TRUE)
```

<div style="margin-bottom:50px;"></div>

Repeat this with a protein only signature.

```{r}
##########################################################################################
pro = read_csv(paste(baseRepository, '/proteomics20210421_a673TimecourseSearchOutput/dataset_deqms_day7-day0.csv', sep = '')) %>%
  dplyr::select(gene, logFC, sca.P.Value) %>%
  dplyr::rename(proFC = logFC, proPVal = sca.P.Value) %>%
  dplyr::filter(proFC <= -2 & proPVal <= 0.001) %>%
  left_join(ccle) %>%
  dplyr::select(gene, proFC, proPVal, log2Tpm, stripped_cell_line_name, lineage, lineage_subtype) %>%
  dplyr::filter(!is.na(stripped_cell_line_name))


##
ewsCells = unique(pro[grepl('Ewing', pro$lineage_subtype), 'stripped_cell_line_name']$stripped_cell_line_name)
boneCells = unique(pro[grepl('bone', pro$lineage), 'stripped_cell_line_name']$stripped_cell_line_name)
cellOutputData = data.frame()

set.seed(13579)
for (i in 1:10000){
  #message(i)
  ewsCellLocation = sample(seq(1,length(ewsCells)), 1)
  otherCells = unique(pro[pro$stripped_cell_line_name != ewsCells[ewsCellLocation], 'stripped_cell_line_name']$stripped_cell_line_name)
  otherCellLocation = sample(seq(1,length(otherCells)), 1)
  ##
  cellData = pro %>%
    dplyr::filter(stripped_cell_line_name == ewsCells[ewsCellLocation] | stripped_cell_line_name == otherCells[otherCellLocation]) %>%
    dplyr::select(gene, log2Tpm, stripped_cell_line_name) %>%
    pivot_wider(names_from = stripped_cell_line_name, values_from = log2Tpm) %>%
    dplyr::select(-gene)
  ##
  cellCorrelation = cor(cellData[,1], cellData[,2], use = 'pairwise.complete.obs', method = 'pearson')
  isEwing = ifelse(otherCells[otherCellLocation] %in% ewsCells, TRUE, FALSE)
  isBone = ifelse(otherCells[otherCellLocation] %in% boneCells, TRUE, FALSE)
  ##
  prepOutputData = data.frame('cellOne' = ewsCells[ewsCellLocation],
                              'cellTwo' = otherCells[otherCellLocation],
                              'correlation' = cellCorrelation[1],
                              'isEwing' = isEwing,
                              'isBone' = isBone)
  cellOutputData = rbind(cellOutputData, prepOutputData)
}
saveRDS(cellOutputData, paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_correlationProteinOnly.rds', sep = ''))
```

<div style="margin-bottom:50px;"></div>

Plot these data.

```{r}
##########################################################################################
corrData = readRDS(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_correlationProteinOnly.rds', sep = ''))
ggplot(corrData, aes(correlation, group = isEwing, fill = isEwing)) +
  geom_density(color = brewer.pal(11,'RdGy')[10], size = 0.5) +
  scale_fill_manual(values = brewer.pal(8,'Spectral')[c(1,8)]) +
  labs(x = 'Pearson correlation', y = 'Density', title = 'Protein expression signature') +
  geom_vline(xintercept = c(median(corrData[corrData$isEwing == TRUE, 'correlation'], na.rm = TRUE))) +
  geom_vline(xintercept = c(median(corrData[corrData$isEwing == FALSE, 'correlation'], na.rm = TRUE))) +
  scale_x_continuous(limits = c(-0.25,1), breaks = seq(-1,1,0.2)) +
  theme_classic() +
  theme(legend.position = 'none')
ggsave(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/density_correlationProteinOnly.pdf', sep = ''),
       height = 4, width = 4, useDingbats = FALSE)
table(corrData$isEwing)
median(corrData[corrData$isEwing == TRUE, 'correlation'], na.rm = TRUE)
median(corrData[corrData$isEwing == FALSE, 'correlation'], na.rm = TRUE)
```

<div style="margin-bottom:50px;"></div>

Repeat this with a combined signature.

```{r}
##########################################################################################
pro = read_csv(paste(baseRepository, '/proteomics20210421_a673TimecourseSearchOutput/dataset_deqms_day7-day0.csv', sep = '')) %>%
  dplyr::select(gene, logFC, sca.P.Value) %>%
  dplyr::rename(proFC = logFC, proPVal = sca.P.Value)
rna = read_csv(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_deseq2_day7-day0.csv', sep = '')) %>%
  dplyr::select(symbol, log2FoldChange, pvalue) %>%
  dplyr::rename(gene = symbol, rnaFC = log2FoldChange, rnaPVal = pvalue)

##
rnaPro = rna %>%
  left_join(pro) %>%
  dplyr::filter(proFC <= -2 & proPVal <= 0.001 & rnaFC <= -2 & rnaPVal <= 0.001) %>%
  left_join(ccle) %>%
  dplyr::select(gene, proFC, proPVal, rnaFC, rnaPVal, log2Tpm, stripped_cell_line_name, lineage, lineage_subtype) %>%
  dplyr::filter(!is.na(stripped_cell_line_name))
 length(unique(rnaPro$gene))
  
##
ewsCells = unique(rnaPro[grepl('Ewing', rnaPro$lineage_subtype), 'stripped_cell_line_name']$stripped_cell_line_name)
boneCells = unique(rnaPro[grepl('bone', rnaPro$lineage), 'stripped_cell_line_name']$stripped_cell_line_name)
cellOutputData = data.frame()

set.seed(13579)
for (i in 1:10000){
  #message(i)
  ewsCellLocation = sample(seq(1,length(ewsCells)), 1)
  otherCells = unique(rnaPro[rnaPro$stripped_cell_line_name != ewsCells[ewsCellLocation], 'stripped_cell_line_name']$stripped_cell_line_name)
  otherCellLocation = sample(seq(1,length(otherCells)), 1)
  ##
  cellData = rnaPro %>%
    dplyr::filter(stripped_cell_line_name == ewsCells[ewsCellLocation] | stripped_cell_line_name == otherCells[otherCellLocation]) %>%
    dplyr::select(gene, log2Tpm, stripped_cell_line_name) %>%
    pivot_wider(names_from = stripped_cell_line_name, values_from = log2Tpm) %>%
    dplyr::select(-gene)
  ##
  cellCorrelation = cor(cellData[,1], cellData[,2], use = 'pairwise.complete.obs', method = 'pearson')
  isEwing = ifelse(otherCells[otherCellLocation] %in% ewsCells, TRUE, FALSE)
  isBone = ifelse(otherCells[otherCellLocation] %in% boneCells, TRUE, FALSE)
  ##
  prepOutputData = data.frame('cellOne' = ewsCells[ewsCellLocation],
                              'cellTwo' = otherCells[otherCellLocation],
                              'correlation' = cellCorrelation[1],
                              'isEwing' = isEwing,
                              'isBone' = isBone)
  cellOutputData = rbind(cellOutputData, prepOutputData)
}
saveRDS(cellOutputData, paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_correlationCombined.rds', sep = ''))
```

<div style="margin-bottom:50px;"></div>

Plot these data.

```{r}
##########################################################################################
corrData = readRDS(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_correlationCombined.rds', sep = ''))
ggplot(corrData, aes(correlation, group = isEwing, fill = isEwing)) +
  geom_density(color = brewer.pal(11,'RdGy')[10], size = 0.5) +
  scale_fill_manual(values = brewer.pal(8,'Spectral')[c(1,8)]) +
  labs(x = 'Pearson correlation', y = 'Density', title = 'Combined signature') +
  geom_vline(xintercept = c(median(corrData[corrData$isEwing == TRUE, 'correlation'], na.rm = TRUE))) +
  geom_vline(xintercept = c(median(corrData[corrData$isEwing == FALSE, 'correlation'], na.rm = TRUE))) +
  scale_x_continuous(limits = c(-0.25,1), breaks = seq(-1,1,0.2)) +
  theme_classic() +
  theme(legend.position = 'none')
ggsave(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/density_correlationCombined.pdf', sep = ''),
       height = 4, width = 4, useDingbats = FALSE)
table(corrData$isEwing)
median(corrData[corrData$isEwing == TRUE, 'correlation'], na.rm = TRUE)
median(corrData[corrData$isEwing == FALSE, 'correlation'], na.rm = TRUE)
```

<div style="margin-bottom:50px;"></div>

So it doesn't seem to improve much as you include more of these genes. But what is this tail that we see for the Ewing cases? Are there a subset of Ewing cells that just aren't like the others?

```{r}
##########################################################################################
corrData = readRDS(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_correlationCombined.rds', sep = ''))
ewsCorr = corrData[corrData$isEwing == TRUE,]

##
pro = read_csv(paste(baseRepository, '/proteomics20210421_a673TimecourseSearchOutput/dataset_deqms_day7-day0.csv', sep = '')) %>%
  dplyr::select(gene, logFC, sca.P.Value) %>%
  dplyr::rename(proFC = logFC, proPVal = sca.P.Value)
rna = read_csv(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_deseq2_day7-day0.csv', sep = '')) %>%
  dplyr::select(symbol, log2FoldChange, pvalue) %>%
  dplyr::rename(gene = symbol, rnaFC = log2FoldChange, rnaPVal = pvalue)
rnaPro = rna %>%
  left_join(pro) %>%
  dplyr::filter(proFC <= -2 & proPVal <= 0.001 & rnaFC <= -2 & rnaPVal <= 0.001) %>%
  left_join(ccle) %>%
  dplyr::filter(grepl('Ewing', lineage_subtype)) %>%
  dplyr::select(gene, log2Tpm, stripped_cell_line_name) %>%
  dplyr::filter(!is.na(stripped_cell_line_name)) %>%
  pivot_wider(names_from = stripped_cell_line_name, values_from = log2Tpm) %>%
  dplyr::select(-gene) %>%
  correlate() %>%
  network_plot(colors = brewer.pal(8,'Spectral')[c(8,1)], legend = FALSE, min_cor = 0.7) +
  ggplot2::labs(title = 'Ewing cell correlations') +
  ggplot2::geom_curve(size = 0.1) +
  ggplot2::theme_classic() +
  ggplot2::theme(legend.position = 'none') +
  ggplot2::ggsave(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/network_ewingCellLinesCombinedSignature.pdf', sep = ''),
                  height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

CHLA57 definitely seems to be different. Can we improve this score by including other information? Perhaps something where we include RNA/protein data with cell line specificity, FLI1 binding peaks, and MSC regulation. Read in the annotation file we will need first.

```{r}
##########################################################################################
totalGtf = as.data.frame(rtracklayer::import.gff('C:/Users/chughes/Documents/bccrc/databases/hg38/gencode_gtf/hg38.gtf.gz'))

##
#expressionData = esExpression
#phenotypeData = esPhenotype
#geneOfInterest = 'MYOM2'
survivalData = function(expressionData, phenotypeData, geneOfInterest){
  ##
  geneOfInterestExprs = expressionData %>%
    dplyr::filter(symbol == geneOfInterest)
  
  ##check for data
  if (nrow(geneOfInterestExprs) >= 1){
    #print('Gene found.')
    geneOfInterestExprs$medExprs = apply(geneOfInterestExprs[,which(grepl('GS',colnames(geneOfInterestExprs)))[1]:ncol(geneOfInterestExprs)], 
                                       1, function(x) mean(x, na.rm = TRUE))
    geneOfInterestSort = geneOfInterestExprs %>%
      dplyr::arrange(desc(medExprs))
    ########
    #keep the top probe for the gene, based on the expression calculated above
    geneSurvivalInput = geneOfInterestSort[1,1:(ncol(geneOfInterestSort) - 1)] %>%
      pivot_longer(cols = colnames(geneOfInterestSort)[which(grepl('GS',colnames(geneOfInterestSort)))[1]]:colnames(geneOfInterestSort)[(ncol(geneOfInterestSort) - 1)], 
                 names_to = 'geo_accession', values_to = 'rnaExprs') %>%
      right_join(phenotypeData) %>%
      dplyr::arrange(rnaExprs)
      ########
    geneCutPoint = surv_cutpoint(geneSurvivalInput,
                               time = 'ovs',
                               event = 'status',
                               variables = 'rnaExprs',
                               minprop = 0.15)
    geneSurvivalCat = surv_categorize(geneCutPoint)
    geneSurvivalInput$geneLevel = geneSurvivalCat$rnaExprs
    geneSurvivalInput$geneLevel = factor(geneSurvivalInput$geneLevel, levels = c('low','high'))
    #calculation of the different survival metrics based on our data
    survivalFit = survfit(Surv(ovs, status) ~ geneLevel, data = geneSurvivalInput)
    coxStats = coxph(Surv(ovs, status) ~ geneLevel, data = geneSurvivalInput)
    coxZScore = coef(coxStats)/sqrt(diag(vcov(coxStats)))
    coxOutputValue = coxZScore
  }
  
  
  ###
  if (nrow(geneOfInterestExprs) < 1){
    #print('Gene not found in first pass, searching secondary identifier annotation.')
    geneOfInterestExprs = expressionData %>%
      dplyr::filter(arraySymbolOne == geneOfInterest)
    
    if (nrow(geneOfInterestExprs) < 1){
      #print('Gene not found in second pass, searching third identifier annotation.')
      geneOfInterestExprs = expressionData %>%
        dplyr::filter(arraySymbolTwo == geneOfInterest)
    }
    
    if (nrow(geneOfInterestExprs) < 1){
      #print('Gene not found. Moving on to next entry.')
      return(NA)
    } else {
      geneOfInterestExprs$medExprs = apply(geneOfInterestExprs[,which(grepl('GS',colnames(geneOfInterestExprs)))[1]:ncol(geneOfInterestExprs)], 
                                       1, function(x) mean(x, na.rm = TRUE))
      geneOfInterestSort = geneOfInterestExprs %>%
        dplyr::arrange(desc(medExprs))
      ########
      #keep the top probe for the gene, based on the expression calculated above
      geneSurvivalInput = geneOfInterestSort[1,1:(ncol(geneOfInterestSort) - 1)] %>%
        pivot_longer(cols = colnames(geneOfInterestSort)[which(grepl('GS',colnames(geneOfInterestSort)))[1]]:colnames(geneOfInterestSort)[(ncol(geneOfInterestSort) - 1)], 
                 names_to = 'geo_accession', values_to = 'rnaExprs') %>%
        right_join(phenotypeData) %>%
        dplyr::arrange(rnaExprs)
        ########
      geneCutPoint = surv_cutpoint(geneSurvivalInput,
                               time = 'ovs',
                               event = 'status',
                               variables = 'rnaExprs',
                               minprop = 0.15)
      geneSurvivalCat = surv_categorize(geneCutPoint)
      geneSurvivalInput$geneLevel = geneSurvivalCat$rnaExprs
      geneSurvivalInput$geneLevel = factor(geneSurvivalInput$geneLevel, levels = c('low','high'))
      #calculation of the different survival metrics based on our data
      survivalFit = survfit(Surv(ovs, status) ~ geneLevel, data = geneSurvivalInput)
      coxStats = coxph(Surv(ovs, status) ~ geneLevel, data = geneSurvivalInput)
      coxZScore = coef(coxStats)/sqrt(diag(vcov(coxStats)))
      coxOutputValue = coxZScore
    }
  }
  return(coxOutputValue)
}
```

<div style="margin-bottom:50px;"></div>

Now proceed with score definition.

```{r, message = FALSE}
##########################################################################################
##
##our protein data
pro = read_csv(paste(baseRepository, '/proteomics20210421_a673TimecourseSearchOutput/dataset_deqms_day7-day0.csv', sep = '')) %>%
  dplyr::select(gene, logFC, sca.P.Value) %>%
  dplyr::rename(proFC = logFC, proPVal = sca.P.Value) %>%
  dplyr::filter(proFC <= -1 | proFC >= 1) %>%
  dplyr::filter(proPVal <= 0.001) %>%
  dplyr::mutate(proClass = ifelse(proFC > 0, 'dn', 'up')) %>%
  dplyr::filter(!is.na(gene))

##
##our rna data
rna = read_csv(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_deseq2_day7-day0.csv', sep = '')) %>%
  dplyr::select(symbol, log2FoldChange, pvalue) %>%
  dplyr::rename(gene = symbol, rnaFC = log2FoldChange, rnaPVal = pvalue) %>%
  dplyr::filter(rnaFC <= -2 | rnaFC >= 2) %>%
  dplyr::filter(rnaPVal <= 0.001) %>%
  dplyr::mutate(rnaClass = ifelse(rnaFC > 0, 'dn', 'up')) %>%
  dplyr::filter(!is.na(gene))

##
##depmap cell line expression, need to define a specificity score here
rnaProGenes = unique(c(pro$gene, rna$gene))
cells = ccle %>%
  dplyr::select(gene, log2Tpm, lineage_subtype) %>%
  dplyr::mutate(isEwing = ifelse(grepl('Ewing', lineage_subtype), 'yes', 'no')) %>%
  dplyr::select(-lineage_subtype) %>%
  dplyr::filter(gene %in% rnaProGenes)

ewsSpecScores = data.frame()
for (i in 1:length(rnaProGenes)){
  targetGene = cells %>%
    dplyr::filter(gene == rnaProGenes[i]) %>%
    group_by(isEwing) %>%
    dplyr::summarise(medRna = median(log2Tpm, na.rm = TRUE))
  if (nrow(targetGene) < 2){
    #message('no value for ', rnaProGenes[i])
    targetGeneScore = data.frame('gene' = rnaProGenes[i],
                               'medRna' = NA)
    ewsSpecScores = rbind(ewsSpecScores, targetGeneScore)
  } else {
      targetGeneScore = data.frame('gene' = rnaProGenes[i],
                               'medRna' = targetGene[targetGene$isEwing == 'no', 'medRna'] - targetGene[targetGene$isEwing == 'yes', 'medRna'])
      ewsSpecScores = rbind(ewsSpecScores, targetGeneScore)
  }
}
ewsCellScores = ewsSpecScores %>%
  dplyr::rename(cellScore = medRna)


##
##msc regulation
msc = read_tsv(paste(baseRepository, '/sequencing20210419_survivalAndBafComplexIntersection/GSE94277_counts_matrix.txt.gz', sep = '')) %>%
  dplyr::select(gene, MSC1.control.RNAseq:MSC1b.FLI1.RNAseq, MSC5.shGFP.control.RNAseq:MSC5b.shBRG1.EWSFLI1.RNAseq) %>%
  dplyr::filter(gene %in% rnaProGenes) %>%
  dplyr::mutate(fli1RepA = (MSC1.FLI1.RNAseq + 1) / (MSC1.EWSFLI1.RNAseq + 1), fli1RepB = (MSC1b.FLI1.RNAseq + 1) / (MSC1b.EWSFLI1.RNAseq + 1)) %>%
  dplyr::mutate(brg1RepA = (MSC5.shBRG1.EWSFLI1.RNAseq + 1) / (MSC5.shGFP.EWSFLI1.RNAseq + 1), brg1RepB = (MSC5b.shBRG1.EWSFLI1.RNAseq + 1) / (MSC5b.shGFP.EWSFLI1.RNAseq + 1)) %>%
  dplyr::rowwise() %>%
  dplyr::mutate(fli1Score = log2(mean(c(fli1RepA, fli1RepB), na.rm = TRUE)), brg1Score = log2(mean(c(brg1RepA, brg1RepB), na.rm = TRUE))) %>%
  dplyr::select(gene, fli1Score, brg1Score)


##
##cell surface
cellSurface = read_tsv(paste(generalDatasets, '/bauschFluckInSilicoCellSurfaceProteomePmid30373828/combinedSurfaceomeProteins.txt', sep = '')) %>%
  dplyr::rename(gene = `UniProt gene`, surfaceAnno = `Surfaceome Label`) %>%
  dplyr::select(gene, surfaceAnno) %>%
  unique()


##
##fli1 peaks
macsHighPeaks = readRDS(paste(baseRepository, '/sequencing20210114_a673ChipSeqReprocessingAynaud/SRR8832674_peaksGgaaAnnotated.rds', sep = '')) %>%
  dplyr::mutate(midPoint = start + round(length / 2, 0)) %>%
  dplyr::filter(fold_enrichment >= 1 & ggaaCounts > 0)
highPeaks = data.frame()
for (i in 1:length(rnaProGenes)){
  goiGtf = totalGtf %>%
    dplyr::filter(grepl(paste('^',rnaProGenes[i],'$',sep=''), gene_name)) %>%
    dplyr::filter(grepl('gene', type) & grepl('protein_coding', gene_type)) %>%
    dplyr::mutate(modStart = ifelse(strand == '+', abs(start - 100000), start)) %>%
    dplyr::mutate(modEnd = ifelse(strand == '-', abs(end + 100000), end))
  goiPeaks = macsHighPeaks %>%
    dplyr::filter(chr == goiGtf$seqnames[1]) %>%
    dplyr::filter(start > goiGtf$modStart & end < goiGtf$modEnd) %>%
    dplyr::arrange(-ggaaCounts)
  ##
  if(nrow(goiPeaks) < 1){
    #message('no value for ', rnaProGenes[i])
    ewsPeaks = data.frame('gene' = rnaProGenes[i],
                          'strand' = as.character(goiGtf$strand[1]),
                          'chromosome' = as.character(goiGtf$seqnames[1]),
                          'geneSize' = goiGtf$width[1] / 1000,
                          'highFli1PeakScore' = 0,
                          'highOutsidePeakScore' = 0,
                          'highInsidePeakScore' = 0,
                          'outsideMsat' = 0,
                          'outsidePeakName' = NA,
                          'insideMsat' = 0,
                          'insidePeakName' = NA,
                          'numberOfGgaaPeaks' = 0)
    highPeaks = rbind(highPeaks, ewsPeaks)
  } else {
    ##
    beforeGene = goiPeaks %>%
      dplyr::filter(abs_summit <= min(goiGtf$start) | abs_summit >= max(goiGtf$end))
    ##
    insideGene = goiPeaks %>%
      dplyr::filter(abs_summit >= min(goiGtf$start) & abs_summit <= max(goiGtf$end))
    ##
    ewsPeaks = data.frame('gene' = rnaProGenes[i],
                          'strand' = as.character(goiGtf$strand[1]),
                          'chromosome' = as.character(goiGtf$seqnames[1]),
                          'geneSize' = goiGtf$width[1] / 1000,
                          'highFli1PeakScore' = goiPeaks$fold_enrichment[1],
                          'highOutsidePeakScore' = ifelse(nrow(beforeGene) >= 1, beforeGene$fold_enrichment[1], 0),
                          'highInsidePeakScore' = ifelse(nrow(insideGene) >= 1, insideGene$fold_enrichment[1], 0),
                          'outsideMsat' = ifelse(nrow(beforeGene) >= 1, beforeGene$ggaaCounts[1], 0),
                          'outsidePeakName' = ifelse(nrow(beforeGene) >= 1, beforeGene$name[1], NA),
                          'insideMsat' = ifelse(nrow(insideGene) >= 1, insideGene$ggaaCounts[1], 0),
                          'insidePeakName' = ifelse(nrow(insideGene) >= 1, insideGene$name[1], NA),
                          'numberOfGgaaPeaks' = nrow(goiPeaks))
    highPeaks = rbind(highPeaks, ewsPeaks)
  }
}


##
macsLowPeaks = read_tsv(paste(baseRepository, '/sequencing20210114_a673ChipSeqReprocessingAynaud/SRR8832669_peaks.xls', sep = ''), 
                     skip = 30, 
                     col_names = c('chr','start','end','length','abs_summit','pileup','logPValue','fold_enrichment','logQValue','name')) %>%
  dplyr::mutate(midPoint = start + round(length / 2, 0)) %>%
  dplyr::filter(fold_enrichment >= 10)
lowPeaks = data.frame()
for (i in 1:length(rnaProGenes)){
  goiGtf = totalGtf %>%
    dplyr::filter(grepl(paste('^',rnaProGenes[i],'$',sep=''), gene_name)) %>%
    dplyr::filter(grepl('gene', type) & grepl('protein_coding', gene_type)) %>%
    dplyr::mutate(modStart = ifelse(strand == '+', abs(start - 100000), start)) %>%
    dplyr::mutate(modEnd = ifelse(strand == '-', abs(end + 100000), end))
  goiPeaks = macsLowPeaks %>%
    dplyr::filter(chr == goiGtf$seqnames[1]) %>%
    dplyr::filter(start > goiGtf$modStart & end < goiGtf$modEnd) %>%
    dplyr::arrange(-fold_enrichment)
  ##
  if(nrow(goiPeaks) < 1){
    #message('no value for ', rnaProGenes[i])
    ewsPeaks = data.frame('gene' = rnaProGenes[i],
                        'lowFli1PeakScore' = 0,
                        'lowOutsidePeakScore' = 0,
                        'lowInsidePeakScore' = 0)
    lowPeaks = rbind(lowPeaks, ewsPeaks)
  } else {
    beforeGene = goiPeaks %>%
      dplyr::filter(abs_summit <= min(goiGtf$start) | abs_summit >= max(goiGtf$end))
    insideGene = goiPeaks %>%
      dplyr::filter(abs_summit >= min(goiGtf$start) & abs_summit <= max(goiGtf$end))
    ewsPeaks = data.frame('gene' = rnaProGenes[i],
                        'lowFli1PeakScore' = goiPeaks$fold_enrichment[1],
                        'lowOutsidePeakScore' = ifelse(nrow(beforeGene) >= 1, beforeGene$fold_enrichment[1], 0),
                        'lowInsidePeakScore' = ifelse(nrow(insideGene) >= 1, insideGene$fold_enrichment[1], 0))
    lowPeaks = rbind(lowPeaks, ewsPeaks)
  }
}



##
##survival data, Savola
esExpression = readRDS(paste(baseRepository, '/microarray20201203_dlg2SavolaSurvival/dataset_savolaExpression.rds', sep = ''))
esPhenotype = readRDS(paste(baseRepository, '/microarray20201203_dlg2SavolaSurvival/dataset_savolaPhenotype.rds', sep = ''))
savolaSurvival = data.frame()
for (i in 1:length(rnaProGenes)){
  survivalScore = data.frame('gene' = rnaProGenes[i],
                             'savolaScore' = survivalData(esExpression, esPhenotype, rnaProGenes[i]))
  row.names(survivalScore) = NULL
  savolaSurvival = rbind(savolaSurvival, survivalScore)
}

##survival data, Postel-Vinay
esExpression = readRDS(paste(baseRepository, '/microarray20201203_dlg2PostelVinaySurvival/dataset_postelVinayExpression.rds', sep = ''))
esPhenotype = readRDS(paste(baseRepository, '/microarray20201203_dlg2PostelVinaySurvival/dataset_postelVinayPhenotype.rds', sep = ''))
postelVinaySurvival = data.frame()
for (i in 1:length(rnaProGenes)){
  survivalScore = data.frame('gene' = rnaProGenes[i],
                             'postelVinayScore' = survivalData(esExpression, esPhenotype, rnaProGenes[i]))
  row.names(survivalScore) = NULL
  postelVinaySurvival = rbind(postelVinaySurvival, survivalScore)
}


##
rnaProCells = rna %>%
  left_join(pro) %>%
  left_join(ewsCellScores) %>%
  left_join(cellSurface) %>%
  left_join(msc) %>%
  left_join(highPeaks) %>%
  left_join(lowPeaks) %>%
  left_join(savolaSurvival) %>%
  left_join(postelVinaySurvival) %>%
  dplyr::mutate(aggregateSurvival = savolaScore + postelVinayScore)

##
saveRDS(rnaProCells, paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_ewsSignatureRawData.rds', sep = ''))
write.table(rnaProCells, paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_ewsSignatureRawData.csv', sep = ''),
            col.names = TRUE, row.names = FALSE, quote = FALSE, sep = ',')
```

<div style="margin-bottom:50px;"></div>

Now that we have our dataset, make some example plots of key genes just to demonstrate the traits.

```{r}
##########################################################################################  
sigData = readRDS(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_ewsSignatureRawData.rds', sep = ''))


##
goi = sigData %>%
  dplyr::filter(gene == 'DLG2') %>%
  dplyr::select(rnaFC, proFC, cellScore, fli1Score, brg1Score, highFli1PeakScore, lowFli1PeakScore, savolaScore, postelVinayScore) %>%
  dplyr::mutate(highFli1PeakScore = log2(highFli1PeakScore), lowFli1PeakScore = log2(lowFli1PeakScore)) %>%
  pivot_longer(cols = rnaFC:postelVinayScore, names_to = 'category', values_to = 'scores') %>%
  dplyr::mutate(category = factor(category, levels = category)) %>%
  dplyr::mutate(scores = ifelse(is.na(scores), 0, scores))
##
ggplot(goi, aes(category, scores)) +
  geom_bar(stat = 'identity', width = 0.8) +
  labs(x = 'Category', y = 'Score', title = 'DLG2 EwS scores') +
  scale_y_continuous(limits = c(-5,6), breaks = seq(-6,6,2)) +
  theme_classic()
ggsave(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/barplot_ewsScoresDlg2.pdf', sep = ''),
                  height = 4, width = 3, useDingbats = FALSE)



##
goi = sigData %>%
  dplyr::filter(gene == 'MGLL') %>%
  dplyr::select(rnaFC, proFC, cellScore, fli1Score, brg1Score, highFli1PeakScore, lowFli1PeakScore, savolaScore, postelVinayScore) %>%
  dplyr::mutate(highFli1PeakScore = log2(highFli1PeakScore), lowFli1PeakScore = log2(lowFli1PeakScore)) %>%
  pivot_longer(cols = rnaFC:postelVinayScore, names_to = 'category', values_to = 'scores') %>%
  dplyr::mutate(category = factor(category, levels = category)) %>%
  dplyr::mutate(scores = ifelse(is.na(scores), 0, scores))
##
ggplot(goi, aes(category, scores)) +
  geom_bar(stat = 'identity', width = 0.8) +
  labs(x = 'Category', y = 'Score', title = 'MGLL EwS scores') +
  scale_y_continuous(limits = c(-5,6), breaks = seq(-6,6,2)) +
  theme_classic()
ggsave(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/barplot_ewsScoresMgll.pdf', sep = ''),
                  height = 4, width = 3, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

What if we sum these scores up to calculate the strongest EwS signature genes?

```{r}
##########################################################################################  
sigData = readRDS(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_ewsSignatureRawData.rds', sep = '')) %>%
  dplyr::mutate(rnaFC = scale(rnaFC), proFC = scale(proFC), fli1Score = scale(fli1Score), brg1Score = scale(brg1Score), aggregateSurvival = scale(aggregateSurvival), cellScore = scale(cellScore)) %>%
  dplyr::mutate(highFli1PeakScore = log2(highFli1PeakScore + 1), lowFli1PeakScore = log2(lowFli1PeakScore + 1))
sigData$ewsScore = rowSums(sigData[,c(2,5,8,10,11,28)], na.rm = TRUE)

##
goi = c('STEAP1','STEAP2','IL1RAP','PRKCB','DLG2')
ewsScoreData = sigData %>%
  dplyr::arrange(ewsScore) %>%
  dplyr::mutate(ewsRank = seq(1,nrow(sigData),1)) %>%
  dplyr::mutate(pText = ifelse(gene %in% goi, gene, ''))
saveRDS(ewsScoreData, paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_ewsSignatureScoredData.rds', sep = ''))
write.table(ewsScoreData, 
            paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_ewsSignatureScoredData.csv', sep = ''),
            sep = ',', quote = FALSE, col.names = TRUE, row.names = FALSE)


##
ggplot(ewsScoreData, aes(ewsRank, ewsScore)) +
  geom_point(size = 3, pch = 21, color = brewer.pal(11,'RdGy')[10], fill = brewer.pal(8,'Spectral')[1], alpha = 0.75) +
  geom_text_repel(label = ewsScoreData$pText, nudge_y = -5, nudge_x = 50, max.overlaps = 200) +
  labs(x = 'Gene rank', y = 'EwS score', title = 'EwS signature genes') +
  geom_hline(yintercept = 10, linetype = 'dashed') +
  theme_classic()
ggsave(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/scatterplot_ewsScoresAllGenes.pdf', sep = ''),
                  height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

Make a heatmap of the EwS signature genes.

```{r}
##########################################################################################
ewsSignatureGenes = readRDS(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_ewsSignatureGeneScores.rds', sep = '')) %>%
  dplyr::filter(ewsScore >= 20) %>%
  dplyr::select(gene)
rna = readRDS(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_txiAllSamplesSummarizedMeans.rds', sep = '')) %>%
  dplyr::filter(symbol %in% ewsSignatureGenes$gene)

##
rnaMatrix = as.matrix(rna[,2:9])
rownames(rnaMatrix) = rna$symbol
pheatmap(rnaMatrix, main = 'base',
         cluster_cols = FALSE,
         scale = 'row',
         show_rownames = FALSE,
         angle_col = 90,
         cutree_rows = 2,
         cellwidth = 8,
         cellheight = 2,
         width = 4,
         height = 4,
         filename = paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/heatmap_ewsSignatureGenesRnaMeanTpmScaled.pdf', sep = ''))

```

<div style="margin-bottom:50px;"></div>

Repeat the correlation analysis with the new signature.

```{r}
##########################################################################################
ewsSignature = readRDS(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_ewsSignatureGeneScores.rds', sep = '')) %>%
  dplyr::filter(ewsScore >= 20) %>%
  left_join(ccle) %>%
  dplyr::select(gene, log2Tpm, stripped_cell_line_name, lineage, lineage_subtype) %>%
  dplyr::filter(!is.na(stripped_cell_line_name))
length(unique(ewsSignature$gene))

##
ewsCells = unique(ewsSignature[grepl('Ewing', ewsSignature$lineage_subtype), 'stripped_cell_line_name']$stripped_cell_line_name)
boneCells = unique(ewsSignature[grepl('bone', ewsSignature$lineage), 'stripped_cell_line_name']$stripped_cell_line_name)
cellOutputData = data.frame()

set.seed(13579)
for (i in 1:10000){
  #message(i)
  ewsCellLocation = sample(seq(1,length(ewsCells)), 1)
  otherCells = unique(ewsSignature[ewsSignature$stripped_cell_line_name != ewsCells[ewsCellLocation], 'stripped_cell_line_name']$stripped_cell_line_name)
  otherCellLocation = sample(seq(1,length(otherCells)), 1)
  ##
  cellData = ewsSignature %>%
    dplyr::filter(stripped_cell_line_name == ewsCells[ewsCellLocation] | stripped_cell_line_name == otherCells[otherCellLocation]) %>%
    dplyr::select(gene, log2Tpm, stripped_cell_line_name) %>%
    pivot_wider(names_from = stripped_cell_line_name, values_from = log2Tpm) %>%
    dplyr::select(-gene)
  ##
  cellCorrelation = cor(cellData[,1], cellData[,2], use = 'pairwise.complete.obs', method = 'pearson')
  isEwing = ifelse(otherCells[otherCellLocation] %in% ewsCells, TRUE, FALSE)
  isBone = ifelse(otherCells[otherCellLocation] %in% boneCells, TRUE, FALSE)
  ##
  prepOutputData = data.frame('cellOne' = ewsCells[ewsCellLocation],
                              'cellTwo' = otherCells[otherCellLocation],
                              'correlation' = cellCorrelation[1],
                              'isEwing' = isEwing,
                              'isBone' = isBone)
  cellOutputData = rbind(cellOutputData, prepOutputData)
}
saveRDS(cellOutputData, paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_correlationHomemadeSignature.rds', sep = ''))
```

<div style="margin-bottom:50px;"></div>

Plot these data.

```{r}
##########################################################################################
corrData = readRDS(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_correlationHomemadeSignature.rds', sep = '')) %>%
  dplyr::filter(!grepl('CHLA57', cellOne) & !grepl('CHLA57', cellTwo))
ggplot(corrData, aes(correlation, group = isEwing, fill = isEwing)) +
  geom_density(color = brewer.pal(11,'RdGy')[10], size = 0.5) +
  scale_fill_manual(values = brewer.pal(8,'Spectral')[c(1,8)]) +
  labs(x = 'Pearson correlation', y = 'Density', title = 'EwS signature') +
  geom_vline(xintercept = c(median(corrData[corrData$isEwing == TRUE, 'correlation'], na.rm = TRUE))) +
  geom_vline(xintercept = c(median(corrData[corrData$isEwing == FALSE, 'correlation'], na.rm = TRUE))) +
  scale_x_continuous(limits = c(-0.6,1), breaks = seq(-1,1,0.2)) +
  theme_classic() +
  theme(legend.position = 'none')
ggsave(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/density_correlationHomemadeSignature.pdf', sep = ''),
       height = 4, width = 4, useDingbats = FALSE)
table(corrData$isEwing)
median(corrData[corrData$isEwing == TRUE, 'correlation'], na.rm = TRUE)
median(corrData[corrData$isEwing == FALSE, 'correlation'], na.rm = TRUE)
```




































