---
title: "CHLA cell line proteomics"
output:
  html_notebook:
      code_folding: none
---

This document details the analysis of sequencing data acquired for comparison of the CHLA9 and CHLA10 cell lines. I am following the protocol outlined [here](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html), as well as another detailed [here](https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html), and [here](http://master.bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html).

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning=FALSE}
##########################################################################################
library('tidyverse')
library('RColorBrewer')
library('ggplot2')
library('tximport')
library('GenomicFeatures')
library('DESeq2')
library("AnnotationDbi")
library("org.Hs.eg.db")
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToOthers'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToOthers'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

First get the sample info.

```{r, message = FALSE}
##########################################################################################
samples = read_csv(paste(baseRepository, '/sequencing20210218_chlaCellLineRnaseq/chla-rna-seq_sequencing-annotation.csv', sep = '')) %>%
  dplyr::mutate(cell = factor(cell))
```

<div style="margin-bottom:50px;"></div>

Now specify the correct files for each sample.

```{r}
##########################################################################################
files = file.path(baseRepository, 'sequencing20210218_chlaCellLineRnaseq', paste(samples$barcode,'_quant',sep = ''), 'quant.sf')
names(files) = samples$barcode
all(file.exists(files))
#tx2gene <- read_csv(file.path(dir, "tx2gene.gencode.v27.csv"))
```

<div style="margin-bottom:50px;"></div>

Now build the tx2gene table. First we need to create a txdb from our GTF. 

```{r}
##########################################################################################
myTxdb = makeTxDbFromGFF('C:/Users/chughes/Documents/bccrc/databases/hg38/gencode_gtf/hg38.gtf.gz')
```

<div style="margin-bottom:50px;"></div>

Now we can select the entries we want.

```{r}
##########################################################################################
k = keys(myTxdb, keytype = 'TXNAME')
tx2gene = AnnotationDbi::select(myTxdb, k, 'GENEID', 'TXNAME')
```

<div style="margin-bottom:50px;"></div>

Import the quantification data.

```{r}
##########################################################################################
txi = tximport(files, type = 'salmon', tx2gene = tx2gene)
```

<div style="margin-bottom:50px;"></div>

Now we can construct the DESeq2 dataset and output a report.

```{r}
##########################################################################################
ddsTxi = DESeqDataSetFromTximport(txi,
                                  colData = samples,
                                  design = ~ cell)
##
dds = DESeq(ddsTxi)
res = results(dds, contrast = c('cell','chla9','chla10'))
ens.str = substr(rownames(res), 1, 15)
res$symbol = mapIds(org.Hs.eg.db,
                     keys=ens.str,
                     column="SYMBOL",
                     keytype="ENSEMBL",
                     multiVals="first")
resOrdered = res[order(res$pvalue),]
write.csv(as.data.frame(resOrdered), file = paste(baseRepository, '/sequencing20210218_chlaCellLineRnaseq/dataset_deseq2_chla9-chla10.csv', sep = ''))

##
resLFC = lfcShrink(dds, coef = "cell_chla9_vs_chla10", type = "apeglm")
resLFC
plotMA(resLFC, ylim = c(-2,2))
```

<div style="margin-bottom:50px;"></div>

## Session info

```{r}
##########################################################################################
sessionInfo()
```






