---
title: "ccleEwsSpecificity"
author: "Christopher Hughes"
format: html
---

## Details

This document describes processing of RNA-seq data from the DepMap project (depmap.org) to investigate patterns of DLG2 expression across different cancer types.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r}
#| message: false
#| warning: false
##########################################################################################
library('tidyverse')
library('vroom')
library('RColorBrewer')
```

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/manuscriptPreparationDlg2'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/manuscriptPreparationDlg2'
```

## Data processing

Read the sample annotation detail.

```{r}
##########################################################################################
cellAnnotation = read_csv(paste(generalDatasets, '/depmap22Q4/Model.csv', sep = ''), show_col_types = FALSE)
```

Read in the RNA data and do some basic processing on it.

```{r}
##########################################################################################
rna = vroom(paste(generalDatasets, '/depmap22Q4/OmicsExpressionProteinCodingGenesTPMLogp1.csv', sep = ''), show_col_types = FALSE) %>%
  dplyr::rename(ModelID = `...1`) %>% 
  tidyr::pivot_longer(cols = `TSPAN6 (7105)`:`CDR1 (1038)`, names_to = 'geneId', values_to = 'log2Tpm') %>%
  dplyr::filter(log2Tpm > 0 & !is.na(log2Tpm)) %>%
  dplyr::mutate(symbol = sub('(.*) \\(.*$', '\\1', geneId)) %>%
  dplyr::select(ModelID, symbol, log2Tpm) %>%
  dplyr::left_join(cellAnnotation)

##save the data
saveRDS(rna, paste(baseRepository, '/rnaSeqData/depmapEwsRnaSeq/datasetOutputs/dataset_depmap22Q4_rnaExpressionParsed.rds', sep = ''))
```

Read in the previously parsed RNAseq data and subset out DLG2 expression values.

```{r}
##########################################################################################
##read the rna data
rna = readRDS(paste(baseRepository, '/rnaSeqData/depmapEwsRnaSeq/datasetOutputs/dataset_depmap22Q4_rnaExpressionParsed.rds', sep = '')) %>%
  dplyr::select(symbol, log2Tpm, OncotreeLineage, OncotreeSubtype, MolecularSubtype) %>%
  dplyr::filter(symbol == 'DLG2') %>%
  dplyr::mutate(isEwing = ifelse(grepl('Ewing', OncotreeSubtype), 'yes', 'no'))
```

Make a plot of DLG2 RNA expression across Depmap cancer types.

```{r}
##########################################################################################
##subset lineage types that don't have at least 6 cell lines measured
rnaSubset = rna %>%
  dplyr::filter(!(OncotreeSubtype %in% names(which(table(OncotreeSubtype) <= 6))))

##make the plot
ggplot(rnaSubset, aes(x = reorder(OncotreeSubtype, log2Tpm, median, na.rm = TRUE), log2Tpm)) +
  #geom_boxplot(size = 0.5, outlier.shape = NA) +
  geom_boxplot(size = 0.5, outlier.size = 0.5) +
  labs(x = 'Lineage subtype in DepMap', y = 'log2(Transcripts per million)', title = 'DLG2 RNA in DepMap') +
  scale_y_continuous(limits = c(0,8), breaks = seq(0,8,2)) +
  geom_hline(yintercept = 2, linetype = 'dashed') +
  theme_classic() +
  #theme(axis.text.x = element_blank())
  theme(axis.text.x = element_text(angle = 45, size = 6, hjust = 1))
ggsave(paste(baseRepository, '/rnaSeqData/depmapEwsRnaSeq/plots/boxplot_dlg2RnaOncotreeSubtypeDepmap.pdf', sep = ''),
       height = 2, width = 2, useDingbats = FALSE)
```

So, there is only really SCLC that has any real expression of DLG2. So, we can make a better OncotreeSubtype plot using a smaller set of data.


```{r}
##########################################################################################
##plot the RNA expression data by lineage
tissuesOfInterest = c('Ewing Sarcoma', 'Small Cell Lung Cancer')
rnaSubset = rna %>%
  dplyr::filter(!(OncotreeSubtype %in% names(which(table(OncotreeSubtype) <= 6)))) %>%
  dplyr::mutate(tissueClass = factor(ifelse(OncotreeSubtype %in% tissuesOfInterest, OncotreeSubtype, 'Other'), levels = c('Ewing Sarcoma', 'Small Cell Lung Cancer', 'Other')))

##
ggplot(rnaSubset, aes(x = reorder(tissueClass, -log2Tpm, median, na.rm = TRUE), log2Tpm, color = tissueClass)) +
  geom_violin(size = 0.25, width = 0.75) +
  geom_boxplot(size = 0.25, outlier.size = 0.25, width = 0.1) +
  labs(x = 'Tissue Specification in DepMap', y = 'log2(TPM + 1)', title = 'DLG2 in DepMap') +
  scale_color_manual(values = c(rep(brewer.pal(4,'OrRd')[4],2), brewer.pal(3,'Greys')[3])) +
  scale_y_continuous(limits = c(0,8), breaks = seq(0,8,2)) +
  theme_classic() +
  theme(legend.position = 'none')
ggsave(paste(baseRepository, '/rnaSeqData/depmapEwsRnaSeq/plots/boxplot_dlg2RnaOncotreeSubtypeDepmapDefined.pdf', sep = ''),
       height = 2, width = 2, useDingbats = FALSE)
```

Make a plot for DLG2 across the lineage_molecular_subtype column.

```{r}
##########################################################################################
##prepare the data
rnaSubset = rna %>%
  dplyr::filter(OncotreeSubtype == 'Ewing Sarcoma') %>%
  dplyr::filter(!is.na(MolecularSubtype)) %>%
  dplyr::filter(!grepl('FEV', MolecularSubtype))

##plot the results
ggplot(rnaSubset, aes(x = reorder(MolecularSubtype, log2Tpm, median, na.rm = TRUE), log2Tpm)) +
  geom_boxplot(aes(fill = MolecularSubtype), size = 0.5, outlier.shape = NA, width = 0.5) +
  geom_point(aes(color = MolecularSubtype), size = 1, position = position_jitterdodge(jitter.width = 0.25)) +
  labs(x = 'Lineage subtype in DepMap', y = 'log2(TPM + 1)', title = 'DLG2 by fusion') +
  scale_y_continuous(limits = c(0,8), breaks = seq(0,8,2)) +
  scale_color_manual(values = c(rep(brewer.pal(3,'OrRd')[3],2))) +
  scale_fill_manual(values = c(rep(brewer.pal(3,'Greys')[2],2))) +
  #geom_hline(yintercept = 2, linetype = 'dashed') +
  theme_classic() +
  theme(legend.position = 'none')
ggsave(paste(baseRepository, '/rnaSeqData/depmapEwsRnaSeq/plots/boxplot_dlg2RnaExpressionByFusionDepmap.pdf', sep = ''),
       height = 2, width = 2, useDingbats = FALSE)

##there isn't really enough ERG fusions in DepMap to make this a useful plot
```








targetGeneList = unique(rna$gene)
ewsRnaScores = data.frame()
for (i in 1:length(targetGeneList)){
#for (i in 8225:8225){
  targetGeneData = rna %>%
    dplyr::filter(gene == targetGeneList[i]) %>%
    dplyr::group_by(isEwing) %>%
    dplyr::summarise(medRna = median(log2Tpm, na.rm = TRUE))
  if (nrow(targetGeneData) < 2){
    #message('no value for ', rnaProGenes[i])
    targetGeneScore = data.frame('gene' = targetGeneList[i],
                               'medRna' = NA)
    ewsRnaScores = rbind(ewsRnaScores, targetGeneScore)
  } else {
    targetGeneScore = data.frame('gene' = targetGeneList[i],
                             'medRna' = targetGeneData[targetGeneData$isEwing == 'yes', 'medRna'] - targetGeneData[targetGeneData$isEwing == 'no', 'medRna'])
    ewsRnaScores = rbind(ewsRnaScores, targetGeneScore)
  }
}
ewsRnaScores = ewsRnaScores %>%
  dplyr::rename(ewsRnaScore = medRna)

##save the data
saveRDS(ewsRnaScores, paste(baseRepository, '/database20220929_ewsExpressionDatabase/ccleEwsSpecificity/dataset_depmapEwsRnaSpecificityScores.rds', sep = ''))
write.table(ewsRnaScores, paste(baseRepository, '/database20220929_ewsExpressionDatabase/ccleEwsSpecificity/dataset_depmapEwsRnaSpecificityScores.tsv', sep = ''),
            col.names = TRUE, row.names = FALSE, quote = FALSE, sep = '\t')
```

Now we can define a protein score using data from PMID: 35839778. Because of how the data is reported here, I added a cell counter to beef up the enrichment scores. For example, for DLG2, which we know is enriched in EwS, only 44 values are reported for the over 900 cell lines analyzed. The non-reported values I presume are NA values, and thus speak to the enrichment of DLG2 in this case. So we take a simple proportion of how many values that are reported are from EwS cells, and scale the enrichment score based on this. 

```{r}
##########################################################################################
##read in the cell line annotation data
cellInfo = read_csv(paste(generalDatasets, '/proteomics20220728_cancerCellLineMapsPmid35839778/model_list_20220628.csv', sep = ''), show_col_types = FALSE) %>%
  dplyr::select(model_id:gender) %>%
  dplyr::filter(grepl('Ewing', cancer_type))

##read in the protein data
proData = read_csv(paste(generalDatasets, '/proteomics20220728_cancerCellLineMapsPmid35839778/proteomics_all_20220713.csv', sep = ''), show_col_types = FALSE) %>%
  dplyr::mutate(isEwing = ifelse(model_name %in% cellInfo$model_name, 'yes', 'no'))

##calculate the score for each protein
targetGeneList = unique(proData$symbol)
ewsProScores = data.frame()
for (i in 1:length(targetGeneList)){
#for (i in 7:7){
  targetGeneData = proData %>%
    dplyr::filter(symbol == targetGeneList[i]) %>%
    dplyr::mutate(cellCount = 1) %>%
    dplyr::group_by(isEwing) %>%
    dplyr::summarise(medPro = median(protein_intensity, na.rm = TRUE),
                     numCells = sum(cellCount, na.rm = TRUE))
  if (nrow(targetGeneData) < 2){
    #message('no value for ', rnaProGenes[i])
    targetGeneScore = data.frame('gene' = targetGeneList[i],
                               'medPro' = NA,
                               'numCells' = NA)
    ewsProScores = rbind(ewsProScores, targetGeneScore)
  } else {
      targetGeneScore = data.frame('gene' = targetGeneList[i],
                               'medPro' = targetGeneData[targetGeneData$isEwing == 'yes', 'medPro'] - targetGeneData[targetGeneData$isEwing == 'no', 'medPro'],
                               'numCells' = (targetGeneData[targetGeneData$isEwing == 'yes', 'numCells'] / sum(targetGeneData$numCells, na.rm = TRUE)) * 10)
      ewsProScores = rbind(ewsProScores, targetGeneScore)
  }
}
ewsProScores = ewsProScores %>%
  dplyr::rename(ewsProScore = medPro) %>%
  dplyr::mutate(ewsProScore = ewsProScore * numCells)

##save the data
saveRDS(ewsProScores, paste(baseRepository, '/database20220929_ewsExpressionDatabase/ccleEwsSpecificity/dataset_proteomicsEwsProSpecificityScores.rds', sep = ''))
write.table(ewsProScores, paste(baseRepository, '/database20220929_ewsExpressionDatabase/ccleEwsSpecificity/dataset_proteomicsEwsProSpecificityScores.tsv', sep = ''),
            col.names = TRUE, row.names = FALSE, quote = FALSE, sep = '\t')
```

Now we can read in both of these sets and combine them.

```{r}
##########################################################################################
##read the data
rnaScores = readRDS(paste(baseRepository, '/database20220929_ewsExpressionDatabase/ccleEwsSpecificity/dataset_depmapEwsRnaSpecificityScores.rds', sep = ''))
proScores = readRDS(paste(baseRepository, '/database20220929_ewsExpressionDatabase/ccleEwsSpecificity/dataset_proteomicsEwsProSpecificityScores.rds', sep = ''))

##join the data
ewsScores = rnaScores %>%
  dplyr::left_join(proScores) %>%
  dplyr::select(-numCells) %>%
  dplyr::rename(ccleEwsSpecificity = ewsRnaScore, symbol = gene)

##save the data
saveRDS(ewsScores, paste(baseRepository, '/database20220929_ewsExpressionDatabase/ccleEwsSpecificity/dataset_proteomicsEwsRnaProSpecificityScores.rds', sep = ''))
write.table(ewsScores, paste(baseRepository, '/database20220929_ewsExpressionDatabase/ccleEwsSpecificity/dataset_proteomicsEwsRnaProSpecificityScores.tsv', sep = ''),
            col.names = TRUE, row.names = FALSE, quote = FALSE, sep = '\t')
```

Finish up.

### Session info

```{r}
##########################################################################################
sessionInfo()
```
