<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.260">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Christopher Hughes">

<title>Lipid raft proteome</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<script src="20220914File1_fragpipeDataProcessing_files/libs/clipboard/clipboard.min.js"></script>
<script src="20220914File1_fragpipeDataProcessing_files/libs/quarto-html/quarto.js"></script>
<script src="20220914File1_fragpipeDataProcessing_files/libs/quarto-html/popper.min.js"></script>
<script src="20220914File1_fragpipeDataProcessing_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="20220914File1_fragpipeDataProcessing_files/libs/quarto-html/anchor.min.js"></script>
<link href="20220914File1_fragpipeDataProcessing_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link id="quarto-text-highlighting-styles" href="20220914File1_fragpipeDataProcessing_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet">
<script src="20220914File1_fragpipeDataProcessing_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="20220914File1_fragpipeDataProcessing_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="20220914File1_fragpipeDataProcessing_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Lipid raft proteome</h1>
</div>





<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Christopher Hughes </p>
          </div>
  </div>
    
    
  </div>
  

</header>

<p>This document details the analysis of proteomics data from sampes that Ahmed Gad sent me that are composed of lipid rafts. These are SILAC labeled samples derived from a mixed cell culture (e.g.&nbsp;one cell type is light, the other is heavy).</p>
<section id="setting-up-the-environment" class="level2">
<h2 class="anchored" data-anchor-id="setting-up-the-environment">Setting up the environment</h2>
<p>These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'tidyverse'</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'stringr'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>baseWorkspace <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToOthers'</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>baseRepository <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToOthers'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="preparing-the-data" class="level2">
<h2 class="anchored" data-anchor-id="preparing-the-data">Preparing the data</h2>
<p>We need to read in the data from Fragpipe and process it into a single table. I am going to use the ‘ion_label_quant’ table for each sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="do">##prepare a sample table</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>sampleInfo <span class="ot">=</span> <span class="fu">tibble</span>(<span class="st">'sampleName'</span> <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">'sample'</span>, <span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">8</span>,<span class="dv">1</span>), <span class="at">sep =</span> <span class="st">''</span>),</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'annotation'</span> <span class="ot">=</span> <span class="st">'unknown'</span>,</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'filePath'</span> <span class="ot">=</span> <span class="fu">file.path</span>(<span class="st">'D:/msDataTemp/proteomics20220723_ahmedLipidRafts/fragpipeAnalysis'</span>, <span class="fu">paste</span>(<span class="st">'sample'</span>, <span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">8</span>,<span class="dv">1</span>), <span class="at">sep =</span> <span class="st">''</span>), <span class="st">'ion.tsv'</span>))</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">file.exists</span>(sampleInfo<span class="sc">$</span>filePath))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">##read in the data for each sample and combine</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>allIons <span class="ot">=</span> <span class="fu">tibble</span>()</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(sampleInfo)){</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  ionTemp <span class="ot">=</span> <span class="fu">read_tsv</span>(sampleInfo<span class="sc">$</span>filePath[i], <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">sequence =</span> <span class="st">`</span><span class="at">Peptide Sequence</span><span class="st">`</span>,</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">modificationSequence =</span> <span class="st">`</span><span class="at">Modified Sequence</span><span class="st">`</span>,</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">abundance =</span> Intensity,</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">accession =</span> <span class="st">`</span><span class="at">Protein ID</span><span class="st">`</span>,</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">symbol =</span> Gene) <span class="sc">%&gt;%</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(accession, symbol, sequence, modificationSequence, abundance) <span class="sc">%&gt;%</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">'contam_'</span>, accession), abundance <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">'8.0142'</span>, modificationSequence), <span class="st">'heavy'</span>,</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">'10.0082'</span>, modificationSequence), <span class="st">'heavy'</span>, <span class="st">'light'</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">labelCount =</span> <span class="fu">str_count</span>(modificationSequence, <span class="st">'</span><span class="sc">\\</span><span class="st">[8</span><span class="sc">\\</span><span class="st">.0142'</span>) <span class="sc">+</span> <span class="fu">str_count</span>(modificationSequence, <span class="st">'</span><span class="sc">\\</span><span class="st">[10</span><span class="sc">\\</span><span class="st">.0082'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">numPsms =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>modificationSequence) <span class="sc">%&gt;%</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(accession, symbol, sequence, label) <span class="sc">%&gt;%</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">abundance =</span> <span class="fu">mean</span>(abundance, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">numPsms =</span> <span class="fu">sum</span>(numPsms)) <span class="sc">%&gt;%</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">numPeptides =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(accession, symbol, label) <span class="sc">%&gt;%</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">abundance =</span> <span class="fu">median</span>(abundance, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">numPsms =</span> <span class="fu">sum</span>(numPsms), <span class="at">numPeptides =</span> <span class="fu">sum</span>(numPeptides)) <span class="sc">%&gt;%</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> accession<span class="sc">:</span>symbol, <span class="at">names_from =</span> <span class="st">'label'</span>, <span class="at">values_from =</span> abundance<span class="sc">:</span>numPeptides) <span class="sc">%&gt;%</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sampleSet =</span> sampleInfo<span class="sc">$</span>sampleName[i])</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>  allIons <span class="ot">=</span> <span class="fu">rbind</span>(allIons, ionTemp)</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'accession', 'symbol', 'sequence'. You can
override using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol', 'sequence'. You can
override using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol', 'sequence'. You can
override using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol', 'sequence'. You can
override using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol', 'sequence'. You can
override using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol', 'sequence'. You can
override using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol', 'sequence'. You can
override using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol', 'sequence'. You can
override using the `.groups` argument.
`summarise()` has grouped output by 'accession', 'symbol'. You can override
using the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">##save the data</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(allIons, <span class="fu">paste</span>(baseRepository, <span class="st">'/proteomics20220723_ahmedLipidRafts/dataset_proteinDataAllSamples.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(allIons, <span class="fu">paste</span>(baseRepository, <span class="st">'/proteomics20220723_ahmedLipidRafts/dataset_proteinDataAllSamples.tsv'</span>, <span class="at">sep =</span> <span class="st">''</span>),</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We are done here for now. Wrap up below.</p>
<section id="session-info" class="level3">
<h3 class="anchored" data-anchor-id="session-info">Session info</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.1.3 (2022-03-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_Canada.1252  LC_CTYPE=English_Canada.1252   
[3] LC_MONETARY=English_Canada.1252 LC_NUMERIC=C                   
[5] LC_TIME=English_Canada.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_0.5.2   stringr_1.4.1   dplyr_1.0.9     purrr_0.3.4    
[5] readr_2.1.2     tidyr_1.2.0     tibble_3.1.8    ggplot2_3.3.6  
[9] tidyverse_1.3.2

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.2    xfun_0.32           haven_2.5.1        
 [4] gargle_1.2.0        colorspace_2.0-3    vctrs_0.4.1        
 [7] generics_0.1.3      htmltools_0.5.3     yaml_2.3.5         
[10] utf8_1.2.2          rlang_1.0.4         pillar_1.8.1       
[13] withr_2.5.0         glue_1.6.2          DBI_1.1.3          
[16] bit64_4.0.5         dbplyr_2.2.1        modelr_0.1.9       
[19] readxl_1.4.1        lifecycle_1.0.1     munsell_0.5.0      
[22] gtable_0.3.0        cellranger_1.1.0    rvest_1.0.3        
[25] htmlwidgets_1.5.4   evaluate_0.16       knitr_1.40         
[28] tzdb_0.3.0          fastmap_1.1.0       parallel_4.1.3     
[31] fansi_1.0.3         broom_1.0.1         backports_1.4.1    
[34] scales_1.2.1        googlesheets4_1.0.1 vroom_1.5.7        
[37] jsonlite_1.8.0      bit_4.0.4           fs_1.5.2           
[40] hms_1.1.2           digest_0.6.29       stringi_1.7.8      
[43] grid_4.1.3          cli_3.3.0           tools_4.1.3        
[46] magrittr_2.0.3      crayon_1.5.1        pkgconfig_2.0.3    
[49] ellipsis_0.3.2      xml2_1.3.3          reprex_2.0.2       
[52] googledrive_2.0.0   lubridate_1.8.0     assertthat_0.2.1   
[55] rmarkdown_2.16      httr_1.4.4          rstudioapi_0.14    
[58] R6_2.5.1            compiler_4.1.3     </code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    setTimeout(function() {
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>