---
title: "Analysis of TurboId data"
output:
  html_notebook:
      code_folding: none
---

This document details the analysis of proteomics data acquired on samples processed using TurboID. The raw MS data were compiled to peptide identifications using SearchGUI and PeptideShaker as detailed elsewhere in this folder.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE}
library('tidyverse')
library('ggplot2')
library('RColorBrewer')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToOthers'
baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToOthers'
```

<div style="margin-bottom:50px;"></div>

We need to call a source file that has some functions we will need. For simplicity and ease of tracking, I make a copy of this file in each directory that I will work in in order to be able to trace the exact one functions that are used, in case I change them and mistakenly don't push it to GitHub.

```{r}
source(paste(baseWorkspace, '/proteomics20200820_rnaseH1TurboIdSean/userDefinedFunctions.R', sep = ''))
```

## Data processing

I want to process the data using SAINTq in the end. Problematically, I don't know much about these data, just that it is an RNaseH1 pulldown. I am not entirely sure of the TMT layout that was used. First, I want to annotate the identification data with gene names, so I will use the annotation index that was created as part of the rawMsDataProcessing that is detailed in other files in this folder. I will read it in to the R session now.

```{r}
##########################################################################################
proteinAnnotation = readRDS(paste(baseRepository, '/proteomics20200820_rnaseH1TurboIdSean/uniprotHumanAug2020.fasta.annotated.rds', sep = ''))
```

<div style="margin-bottom:50px;"></div>

Now I will read in the peptide search results and keep only the information I need. I will also annotate the data using the above object.

```{r}
##########################################################################################
psm = read_tsv(paste(baseRepository, '/proteomics20200820_rnaseH1TurboIdSean/n_20200114_BMBP_SS_Stirling_TurboID_TMT-IP_1_Default_PSM_Report.txt', sep = '')) %>%
  dplyr::select(`Protein(s)`, Sequence, `Modified Sequence`, `Spectrum File`, `Spectrum Scan Number`) %>%
  mutate(accession = sapply(str_split(`Protein(s)`, ','), '[', 1)) %>%
  mutate(scan = `Spectrum Scan Number`) %>%
  mutate(sequence = Sequence) %>%
  mutate(modSequence = `Modified Sequence`) %>%
  left_join(proteinAnnotation) %>%
  dplyr::select(scan, accession, gene, detectablePeptides, sequence, modSequence)
```

<div style="margin-bottom:50px;"></div>

Now we need the quantification data. These are hosted in a folder called quantFiles in the same directory as the data.

```{r, message = FALSE, warning = FALSE}
##########################################################################################
quantFiles = as.list(list.files(paste(baseRepository, '/proteomics20200820_rnaseH1TurboIdSean/quantFiles/', sep = ''),
           pattern = '_Matrix.txt', full.names = TRUE))
##
quantDataSet = lapply(quantFiles, combineQuantFiles)
```

<div style="margin-bottom:50px;"></div>

Now combine the identification and the quantification data into a single object.

```{r}
##########################################################################################
allQuantData = do.call('rbind', quantDataSet)
psmQuant = psm %>%
  left_join(allQuantData) %>%
  group_by(accession, gene, sequence) %>%
  summarise_at(vars(tmt10plex_126:tmt10plex_131C), median, na.rm = TRUE) %>%
  as_tibble()
```

<div style="margin-bottom:50px;"></div>

Save the peptide object in case we want to use it later.

```{r}
##########################################################################################
saveRDS(psmQuant, paste(baseRepository, '/proteomics20200820_rnaseH1TurboIdSean/datasetPeptidesNonNormalized.rds', sep = ''))
write.table(psmQuant, paste(baseRepository, '/proteomics20200820_rnaseH1TurboIdSean/datasetPeptidesNonNormalized.csv', sep = ''),
            quote = FALSE, sep = ',', col.names = TRUE, row.names = FALSE)
```

<div style="margin-bottom:50px;"></div>

So the problem I have now is that I don't know the TMT layout. Maybe I can guess it from the quantification data. Make a boxplot and see what it looks like.

```{r}
##########################################################################################
psmLong = psmQuant %>%
  pivot_longer(cols = tmt10plex_126:tmt10plex_131C, names_to = 'label', values_to = 's2n')
##
ggplot(psmLong, aes(label, log2(s2n))) +
  geom_boxplot(size = 1, outlier.size = 1, outlier.alpha = 0.5)
```

<div style="margin-bottom:50px;"></div>

So that isn't super informative. It looks like all channels have some sort of sample in them. Maybe if we just look at RNaseH1.

```{r}
##########################################################################################
psmLong = psmQuant %>%
  pivot_longer(cols = tmt10plex_126:tmt10plex_131C, names_to = 'label', values_to = 's2n') %>%
  filter(grepl('RNASEH1$', gene))
##
ggplot(psmLong, aes(label, log2(s2n))) +
  geom_boxplot(size = 1, outlier.size = 1, outlier.alpha = 0.5) +
  labs(title = 'RNaseH1 peptides')
```

<div style="margin-bottom:50px;"></div>

Alright so it definitely looks like the later 6 channels are possibly IP samples, with the exception of maybe 131N? I need to ask what the layout here is. Once I get these data back, I can do more work with these data.

Ok, I just got the TMT layout data. It is:

|label|sample|
|----|----|
|TMT126|WKKD1|
|TMT127N|TurboId1|
|TMT127C|WKKD2|
|TMT128N|TurboId2|
|TMT128C|WKKD3|
|TMT129N|WT1|
|TMT129C|D210W1|
|TMT130N|WT2|
|TMT130C|D210W2|
|TMT131N|WT3|
|TMT131C|D210W3|

So now we have the information to do the SAINTq analysis (information available [here](http://saint-apms.sourceforge.net/Main.html). We need to make the input file, as well as the parameters file. First, the input file will be created. This is just for the wild-type compared to the turboID control, for now.

```{r}
##########################################################################################
##annotation file
annoCondition = c('','','','C','C','T','T','T')
annoBait = c('','','','TURBOID','TURBOID','RNASEH1','RNASEH1','RNASEH1')
annoFile = rbind(annoCondition, annoBait)
write.table(annoFile, paste(baseRepository, '/proteomics20200820_rnaseH1TurboIdSean/datasetAnnotationFile.tsv', sep = ''),
            quote = FALSE, sep = '\t', col.names = FALSE, row.names = FALSE)
##intensity file
psmFile = psmQuant %>%
  mutate('Protein' = gene, 'Peptide' = sequence, 'Fragment' = '+2y5+1') %>%
  mutate('TURBOID_rep1' = tmt10plex_127N, 'TURBOID_rep2' = tmt10plex_128N) %>%
  mutate('RNASEH1_rep1' = tmt10plex_129N, 'RNASEH1_rep2' = tmt10plex_130N, 'RNASEH1_rep3' = tmt10plex_131N) %>%
  dplyr::select(Protein, Peptide, Fragment, TURBOID_rep1, TURBOID_rep2, RNASEH1_rep1, RNASEH1_rep2, RNASEH1_rep3)
write.table(psmFile, paste(baseRepository, '/proteomics20200820_rnaseH1TurboIdSean/datasetQuantificationFile.tsv', sep = ''),
            quote = FALSE, sep = '\t', col.names = TRUE, row.names = FALSE)
```

<div style="margin-bottom:50px;"></div>

Now we need to go to the command line and combine these two files. On my system, in terminal I use the commands:

```
> cd /mnt/c/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToOthers/proteomics20200820_rnaseH1TurboIdSean/
> cat datasetAnnotationFile.tsv datasetQuantificationFile.tsv > datasetSaintQuantificationFile.tsv
```

The last thing we need is the parameters file for SAINTq. I used the example parameters file they provided with the tool itself, and just modified the file paths using Notepad software. To run SAINTq, I navigated to the directory where I saved it and pointed it at the parameters file that I modified.

```
> /mnt/c/Users/chris/OneDrive/Documents/software/saintq_v0.0.4/saintq/binaries/saintq.exe param_MSPLIT_pep_level

```

I repeated the above analysis for the other comparisons using the same code as above.

```{r}
##########################################################################################
##annotation file
annoCondition = c('','','','C','C','T','T','T')
annoBait = c('','','','TURBOID','TURBOID','RNASEH1','RNASEH1','RNASEH1')
annoFile = rbind(annoCondition, annoBait)
write.table(annoFile, paste(baseRepository, '/proteomics20200820_rnaseH1TurboIdSean/datasetAnnotationFile.tsv', sep = ''),
            quote = FALSE, sep = '\t', col.names = FALSE, row.names = FALSE)
##intensity file
psmFile = psmQuant %>%
  mutate('Protein' = gene, 'Peptide' = sequence, 'Fragment' = '+2y5+1') %>%
  mutate('TURBOID_rep1' = tmt10plex_127N, 'TURBOID_rep2' = tmt10plex_128N) %>%
  mutate('RNASEH1_rep1' = tmt10plex_129C, 'RNASEH1_rep2' = tmt10plex_130C, 'RNASEH1_rep3' = tmt10plex_131C) %>%
  dplyr::select(Protein, Peptide, Fragment, TURBOID_rep1, TURBOID_rep2, RNASEH1_rep1, RNASEH1_rep2, RNASEH1_rep3)
write.table(psmFile, paste(baseRepository, '/proteomics20200820_rnaseH1TurboIdSean/datasetQuantificationFile.tsv', sep = ''),
            quote = FALSE, sep = '\t', col.names = TRUE, row.names = FALSE)
```

<div style="margin-bottom:50px;"></div>

```{r}
##########################################################################################
##annotation file
annoCondition = c('','','','C','C','T','T','T')
annoBait = c('','','','TURBOID','TURBOID','RNASEH1','RNASEH1','RNASEH1')
annoFile = rbind(annoCondition, annoBait)
write.table(annoFile, paste(baseRepository, '/proteomics20200820_rnaseH1TurboIdSean/datasetAnnotationFile.tsv', sep = ''),
            quote = FALSE, sep = '\t', col.names = FALSE, row.names = FALSE)
##intensity file
psmFile = psmQuant %>%
  mutate('Protein' = gene, 'Peptide' = sequence, 'Fragment' = '+2y5+1') %>%
  mutate('TURBOID_rep1' = tmt10plex_127N, 'TURBOID_rep2' = tmt10plex_128N) %>%
  mutate('RNASEH1_rep1' = tmt10plex_126, 'RNASEH1_rep2' = tmt10plex_127C, 'RNASEH1_rep3' = tmt10plex_128C) %>%
  dplyr::select(Protein, Peptide, Fragment, TURBOID_rep1, TURBOID_rep2, RNASEH1_rep1, RNASEH1_rep2, RNASEH1_rep3)
write.table(psmFile, paste(baseRepository, '/proteomics20200820_rnaseH1TurboIdSean/datasetQuantificationFile.tsv', sep = ''),
            quote = FALSE, sep = '\t', col.names = TRUE, row.names = FALSE)
```

<div style="margin-bottom:50px;"></div>

## Session info

```{r}
sessionInfo()
```


















