---
title: "edgeR analysis of A673 data"
output:
  html_notebook:
      code_folding: none
---

This document details comparisons between the time points in the A673 data to extract differential expression information. I followed the walkthrough give on [this page](http://www.bioconductor.org/packages/release/bioc/vignettes/DEqMS/inst/doc/DEqMS-package-vignette.html#differential-protein-expression-analysis-with-deqms-using-a-protein-table). 

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning=FALSE}
##########################################################################################
library('tidyverse')
library('ggplot2')
library('RColorBrewer')
library('limma')
library('DEqMS')
library('matrixStats')
library('ggrepel')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

Read in the normalized data.

```{r}
##########################################################################################
dataIrsRaw = as.data.frame(readRDS(paste(baseRepository, '/proteomics20201223_a673ShEwsFli1DoxTimecourseSetA/dataset_proteinSetABCIrsNormalized.rds', sep = '')))
row.names(dataIrsRaw) = dataIrsRaw$accession
#dataIrsRaw = dataIrsRaw[complete.cases(dataIrsRaw),]
dataIrs = dataIrsRaw[,c(2,10,18,3,11,19)] #I only keep the two time points I want to compare here
dataIrsLog = log2(dataIrs)
dataIrsLog = na.omit(dataIrsLog)
```

<div style="margin-bottom:50px;"></div>

Check to see that the medians are centered before proceeding. They should be fine from the IRS work. 

```{r}
##########################################################################################
boxplot(dataIrsLog,las=2,main="IRS normalized samples")
```

<div style="margin-bottom:50px;"></div>

Make the design table.

```{r}
##########################################################################################
cond = as.factor(c(rep('day0',3), rep('day7',3)))

# The function model.matrix is used to generate the design matrix
design = model.matrix(~0+cond) # 0 means no intercept for the linear model
colnames(design) = gsub('cond','',colnames(design))
```

<div style="margin-bottom:50px;"></div>

Make the contrasts.

```{r}
##########################################################################################
x = c('day7-day0')
contrast =  makeContrasts(contrasts = x,levels = design)
fit1 = lmFit(dataIrsLog, design)
fit2 = contrasts.fit(fit1,contrasts = contrast)
fit3 = eBayes(fit2)
```

<div style="margin-bottom:50px;"></div>

Perform the DEqMS analysis.

```{r}
##########################################################################################
count_columns = seq(31,33,1)
psm.count.table = data.frame(count = rowMins(
  as.matrix(dataIrsRaw[,count_columns])), row.names =  dataIrsRaw$accession)
fit3$count = psm.count.table[rownames(fit3$coefficients),'count']
fit4 = spectraCounteBayes(fit3)
```

<div style="margin-bottom:50px;"></div>

Outputs of spectraCounteBayes:
object is augmented form of “fit” object from eBayes in Limma, with the additions being:
sca.t - Spectra Count Adjusted posterior t-value
sca.p - Spectra Count Adjusted posterior p-value
sca.dfprior - DEqMS estimated prior degrees of freedom
sca.priorvar- DEqMS estimated prior variance
sca.postvar - DEqMS estimated posterior variance
model - fitted model

Visualize the fit curve.

```{r}
##########################################################################################
# n=30 limits the boxplot to show only proteins quantified by <= 30 PSMs.
VarianceBoxplot(fit4,n=30,main="A673 IRS data",xlab="PSM count")
VarianceScatterplot(fit4,main="A673 IRS data")
```

<div style="margin-bottom:50px;"></div>

Extract the results and output to a file.

```{r}
##########################################################################################
deqmsResults = tibble(outputResult(fit4,coef_col = 1)) %>%
  rename(accession  = gene) %>%
  left_join(dataIrsRaw[,c('accession','gene')])
#if you are not sure which coef_col refers to the specific contrast,type
head(fit4$coefficients)
head(deqmsResults)
write.table(deqmsResults, 
            paste(baseRepository, '/proteomics20201223_a673ShEwsFli1DoxTimecourseSetA/dataset_deqms_day7-day0.csv', sep = ''),
            sep = ',', row.names = FALSE, quote = FALSE)
```

<div style="margin-bottom:50px;"></div>

Make a volcano plot with our usual style.

```{r}
##########################################################################################
deqmsResults$plotPValue = -log10(deqmsResults$sca.P.Value)
deqmsResults$pColors = ifelse(deqmsResults$logFC <= -0.5 & deqmsResults$plotPValue >= 3, brewer.pal(8,'RdBu')[8],
                              ifelse(deqmsResults$logFC >= 0.5 & deqmsResults$plotPValue >= 3, brewer.pal(8,'RdBu')[1],
                                     brewer.pal(8,'Greys')[6]))
goi = c('FLI1','PRKCB','IL1RAP','DLG2','TDO2')
deqmsResults$pText = ifelse(deqmsResults$gene %in% goi, deqmsResults$gene, '')
##
ggplot(deqmsResults, aes(logFC, plotPValue)) +
  geom_point(size = 1, color = deqmsResults$pColors, alpha = 0.5) +
  geom_text_repel(label = deqmsResults$pText, nudge_x = -2) +
  labs(x = 'log2(day7 - day0)', y = '-log10(P-value)', title = 'Differential protein abundance') +
  scale_y_continuous(limits = c(0,14), breaks = seq(0,18,3)) +
  scale_x_continuous(limits = c(-6,6), breaks = seq(-6,6,2)) +
  geom_vline(xintercept = c(-0.5,0.5), linetype = 'dashed') +
  geom_hline(yintercept = 3, linetype = 'dashed') +
  theme_classic()
ggsave(paste(baseRepository, '/proteomics20201223_a673ShEwsFli1DoxTimecourseSetA/scatter_deqmsIrsDay7-Day0.pdf', sep = ''),
       height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

These data look better than the edgeR output. I am not sure how confident I am using edgeR on continuous data like this.

Compare with the RNAseq data from Boulay et al. 

```{r}
##########################################################################################
pro = read_csv(paste(baseRepository, '/proteomics20201223_a673ShEwsFli1DoxTimecourseSetA/dataset_deqms_day7-day0.csv', sep = ''))
rna = read_csv(paste(baseRepository, '/sequencing20201204_ewsFli1RnaSeqBoulayDataReprocessing/dataset_boulaySalmonRnaSeqProcessedData.csv', sep = '')) %>%
  dplyr::select(symbol, a673Fc) %>%
  dplyr::rename('gene' = symbol) %>%
  left_join(pro)
##
geneExpData = rna
targetGenes = c('STAG2','DLG2','IL1RAP','PRKCB','TDO2')
geneExpData$pText = ifelse(geneExpData$gene %in% targetGenes, geneExpData$gene, '')
geneExpData$pColors = ifelse(geneExpData$a673Fc <= -1 & geneExpData$logFC <= -1, brewer.pal(8,'RdBu')[8],
                            ifelse(geneExpData$a673Fc >= 1 & geneExpData$logFC >= 1, brewer.pal(8,'RdBu')[1],
                                    brewer.pal(8,'Greys')[6]))
ggplot(geneExpData, aes(a673Fc, logFC)) +
  geom_point(size = 1, color = geneExpData$pColors, alpha = 0.5) +
  geom_text_repel(label = geneExpData$pText, nudge_x = -4) +
  labs(x = 'log2(shEWS-FLI1 vs shGFP RNA)', y = 'log2(shEWS-FLI1 vs shGFP protein)', title = 'RNA vs proteomics') +
  geom_vline(xintercept = c(-1,1), linetype = 'dashed') +
  geom_hline(yintercept = c(-1,1), linetype = 'dashed') +
  scale_y_continuous(limits = c(-6,6), breaks = seq(-6,6,2)) +
  scale_x_continuous(limits = c(-8,8), breaks = seq(-8,8,2)) +
  theme_classic()
ggsave(paste(baseRepository, '/proteomics20201223_a673ShEwsFli1DoxTimecourseSetA/scatter_rnaExpressionVsProteinAbundanceBoulay.pdf', sep = ''),
      height = 4, width = 4, useDingbats = FALSE)
##
cor(geneExpData$a673Fc, geneExpData$logFC, use = 'pairwise.complete.obs', method = 'pearson')
```

<div style="margin-bottom:50px;"></div>

Compare with the RNAseq data from Riggi et al. 

```{r}
##########################################################################################
pro = read_csv(paste(baseRepository, '/proteomics20201223_a673ShEwsFli1DoxTimecourseSetA/dataset_deqms_day7-day0.csv', sep = ''))
rna = read_csv(paste(baseRepository, '/sequencing20201208_ewsFil1RnaSeqRiggiPmid25453903/dataset_riggiSalmonRnaSeqProcessedData.csv', sep = '')) %>%
  dplyr::select(symbol, a673Fc) %>%
  dplyr::rename('gene' = symbol) %>%
  left_join(pro)
##
geneExpData = rna
targetGenes = c('STAG2','DLG2','IL1RAP','PRKCB','TDO2')
geneExpData$pText = ifelse(geneExpData$gene %in% targetGenes, geneExpData$gene, '')
geneExpData$pColors = ifelse(geneExpData$a673Fc <= -1 & geneExpData$logFC <= -1, brewer.pal(8,'RdBu')[8],
                            ifelse(geneExpData$a673Fc >= 1 & geneExpData$logFC >= 1, brewer.pal(8,'RdBu')[1],
                                    brewer.pal(8,'Greys')[6]))
ggplot(geneExpData, aes(a673Fc, logFC)) +
  geom_point(size = 1, color = geneExpData$pColors, alpha = 0.5) +
  geom_text_repel(label = geneExpData$pText, nudge_x = -4) +
  labs(x = 'log2(shEWS-FLI1 vs shGFP RNA)', y = 'log2(shEWS-FLI1 vs shGFP protein)', title = 'RNA vs proteomics') +
  geom_vline(xintercept = c(-1,1), linetype = 'dashed') +
  geom_hline(yintercept = c(-1,1), linetype = 'dashed') +
  scale_y_continuous(limits = c(-6,6), breaks = seq(-6,6,2)) +
  scale_x_continuous(limits = c(-8,8), breaks = seq(-8,8,2)) +
  theme_classic()
ggsave(paste(baseRepository, '/proteomics20201223_a673ShEwsFli1DoxTimecourseSetA/scatter_rnaExpressionVsProteinAbundanceRiggi.pdf', sep = ''),
      height = 4, width = 4, useDingbats = FALSE)
##
cor(geneExpData$a673Fc, geneExpData$logFC, use = 'pairwise.complete.obs', method = 'pearson')
```


## Session info

```{r}
##########################################################################################
sessionInfo()
```



