---
title: "Coverage maps for DGL2"
output:
  html_notebook:
      code_folding: none
---

This document details working with aligned RNAseq data to create coverage maps for DLG2 using the NCBI annotation rather than ENSEMBL.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning = FALSE}
##########################################################################################
library('tidyverse')
library('ggplot2')
library('RColorBrewer')
library('rtracklayer')
library('wiggleplotr')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

First, read the GTF file that we can use to get the DLG2 annotation details.

```{r}
##########################################################################################
totalGtf = as.data.frame(rtracklayer::import.gff('C:/Users/chughes/Documents/bccrc/databases/hg38/refseq_gtf/GRCh38_latest_genomic_feb2021.gtf.gz'))
```

<div style="margin-bottom:50px;"></div>

Subset the GTF. The main transcript I am going to use is NM_001142699.3. 

```{r}
##########################################################################################
dlg2Gtf = totalGtf %>%
  filter(grepl('NM_001300983.1', transcript_id)) %>%
  dplyr::rename(gene_name = gene_id, gene_id = seqnames) %>%
  mutate('seqnames' = 'chr11')
##
dlg2Meta = tibble('transcript_id' = 'NM_001300983.1',
                  'gene_id' = 'NC_000011.10',
                  'gene_name' = 'DLG2',
                  'strand' = '-')
##
dlg2Exons = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('exon', type)), split.field = 'transcript_id')
##
dlg2Cds = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('CDS', type)), split.field = 'transcript_id')
##
sampleData = tibble(sample_id = c('ewing1','ewing2','ewing3','ewing4','ewing5','ewing6','ewing7','ewing8','ewing9','ewing10','ewing11'), 
                    condition = factor(c(rep('ewing',11)), levels = c('ewing')),
                    scaling_factor = 1)
##
bigwigFiles = c(paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615497_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8616012_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8616213_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8616214_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615592_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615679_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615832_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615859_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615273_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615499_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615521_refseq.chr11.bw', sep = ''))
##
sampleData = sampleData %>%
  dplyr::mutate(bigWig = bigwigFiles)
as.data.frame(sampleData)
##
trackData = dplyr::mutate(sampleData, track_id = condition, colour_group = condition)
#trackData = dplyr::mutate(sampleData, track_id = sample_id, colour_group = condition)
##
plotCoverage(dlg2Exons, dlg2Cds, 
             dlg2Meta, trackData,
             heights = c(2,1), fill_palette = getGenotypePalette(),
             rescale_introns = TRUE)
ggsave(paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/coverage_dlg2CoverageRefseqEwingRna.pdf', sep = ''),
         height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

Now for SCLC.

```{r}
##########################################################################################
dlg2Gtf = totalGtf %>%
  filter(grepl('NM_001300983.1', transcript_id)) %>%
  dplyr::rename(gene_name = gene_id, gene_id = seqnames) %>%
  mutate('seqnames' = 'chr11')
##
dlg2Meta = tibble('transcript_id' = 'NM_001300983.1',
                  'gene_id' = 'NC_000011.10',
                  'gene_name' = 'DLG2',
                  'strand' = '-')
##
dlg2Exons = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('exon', type)), split.field = 'transcript_id')
##
dlg2Cds = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('CDS', type)), split.field = 'transcript_id')
##
sampleData = tibble(sample_id = c('sclc1','sclc2','sclc3','sclc4','sclc5'), 
                    condition = factor(c(rep('sclc',5)), levels = c('sclc')),
                    scaling_factor = 1)
##
bigwigFiles = c(paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/sclcLines/SRR8616152_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/sclcLines/SRR8616161_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/sclcLines/SRR8616112_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/sclcLines/SRR8615423_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/sclcLines/SRR8615341_refseq.chr11.bw', sep = ''))
##
sampleData = sampleData %>%
  dplyr::mutate(bigWig = bigwigFiles)
as.data.frame(sampleData)
##
#trackData = dplyr::mutate(sampleData, track_id = condition, colour_group = condition)
trackData = dplyr::mutate(sampleData, track_id = sample_id, colour_group = condition)
##
plotCoverage(dlg2Exons, dlg2Cds, 
             dlg2Meta, trackData,
             heights = c(2,1), fill_palette = getGenotypePalette(),
             rescale_introns = TRUE)
ggsave(paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/sclcLines/coverage_dlg2CoverageRefseqSclcRnaIndividual.pdf', sep = ''),
         height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

Now for SCLC.

```{r}
##########################################################################################
dlg2Gtf = totalGtf %>%
  filter(grepl('NM_001300983.1', transcript_id)) %>%
  dplyr::rename(gene_name = gene_id, gene_id = seqnames) %>%
  mutate('seqnames' = 'chr11')
##
dlg2Meta = tibble('transcript_id' = 'NM_001300983.1',
                  'gene_id' = 'NC_000011.10',
                  'gene_name' = 'DLG2',
                  'strand' = '-')
##
dlg2Exons = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('exon', type)), split.field = 'transcript_id')
##
dlg2Cds = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('CDS', type)), split.field = 'transcript_id')
##
sampleData = tibble(sample_id = c('other1','other2','other3','other4','other5'), 
                    condition = factor(c(rep('other',5)), levels = c('other')),
                    scaling_factor = 1)
##
bigwigFiles = c(paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/otherLineages/SRR8615955_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/otherLineages/SRR8615971_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/otherLineages/SRR8615675_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/otherLineages/SRR8615747_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/otherLineages/SRR8615812_refseq.chr11.bw', sep = ''))
##
sampleData = sampleData %>%
  dplyr::mutate(bigWig = bigwigFiles)
as.data.frame(sampleData)
##
#trackData = dplyr::mutate(sampleData, track_id = condition, colour_group = condition)
trackData = dplyr::mutate(sampleData, track_id = sample_id, colour_group = condition)
##
plotCoverage(dlg2Exons, dlg2Cds, 
             dlg2Meta, trackData,
             heights = c(2,1), fill_palette = getGenotypePalette(),
             rescale_introns = TRUE)
ggsave(paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/otherLineages/coverage_dlg2CoverageRefseqOtherRnaIndividual.pdf', sep = ''),
         height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

For the Ewing data, I want to display that there are short isoforms annotated and show the coverage for these, as well as the junctions beyond what I think is the first coding exon. 

```{r}
##########################################################################################
transOfInterest = c('NM_001377978.1', 'NM_001377977.1', 'XM_017017289.1')
dlg2Gtf = totalGtf %>%
  filter(transcript_id %in% transOfInterest) %>%
  dplyr::rename(gene_name = gene_id, gene_id = seqnames) %>%
  mutate('seqnames' = 'chr11')
##
dlg2Meta = tibble('transcript_id' = unique(dlg2Gtf$transcript_id),
                  'gene_id' = unique(dlg2Gtf$gene_id),
                  'gene_name' = 'DLG2',
                  'strand' = '-') %>%
  mutate(transcript_id = factor(transcript_id, levels = c('NM_001377977.1', 'NM_001377978.1', 'XM_017017289.1')))
##
dlg2Exons = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('exon', type)), split.field = 'transcript_id')
##
dlg2Cds = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('CDS', type)), split.field = 'transcript_id')
##
sampleData = tibble(sample_id = c('ewing1','ewing2','ewing3','ewing4','ewing5','ewing6','ewing7','ewing8','ewing9','ewing10','ewing11'), 
                    condition = factor(c(rep('ewing',11)), levels = c('ewing')),
                    scaling_factor = 1)
##
bigwigFiles = c(paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615497_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8616012_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8616213_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8616214_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615592_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615679_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615832_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615859_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615273_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615499_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615521_refseq.chr11.bw', sep = ''))
##
sampleData = sampleData %>%
  dplyr::mutate(bigWig = bigwigFiles)
as.data.frame(sampleData)
##
trackData = dplyr::mutate(sampleData, track_id = condition, colour_group = condition)
#trackData = dplyr::mutate(sampleData, track_id = sample_id, colour_group = condition)
##
plotCoverage(dlg2Exons, dlg2Cds, 
             dlg2Meta, trackData,
             heights = c(2,1), fill_palette = getGenotypePalette(),
             rescale_introns = TRUE)
ggsave(paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/coverage_dlg2ShortCoverageRefseqEwingRna.pdf', sep = ''),
         height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

I want to remake the plot for junctions with these refseq annotations.

```{r}
##########################################################################################
transOfInterest = c('NM_001377978.1', 'NM_001377977.1', 'XM_017017289.1')
dlg2Gtf = totalGtf %>%
  filter(transcript_id %in% transOfInterest) %>%
  dplyr::rename(gene_name = gene_id, gene_id = seqnames) %>%
  mutate('seqnames' = 'chr11')
##
dlg2Meta = tibble('transcript_id' = unique(dlg2Gtf$transcript_id),
                  'gene_id' = unique(dlg2Gtf$gene_id),
                  'gene_name' = 'DLG2',
                  'strand' = '-') %>%
  mutate(transcript_id = factor(transcript_id, levels = c('NM_001377977.1', 'NM_001377978.1', 'XM_017017289.1')))
##
dlg2Exons = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('exon', type)), split.field = 'transcript_id')
##
dlg2Cds = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('CDS', type)), split.field = 'transcript_id')
##
sampleData = tibble(sample_id = factor(c('a673','sknmc','mhhes1'), levels = c('a673','sknmc','mhhes1')), 
                    condition = factor(c(rep('ewing',3)), levels = c('ewing')),
                    scaling_factor = c(1,0.4,0.4))
##
bigwigFiles = c(paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8616012_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615497_refseq.chr11.bw', sep = ''),
                paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/SRR8615521_refseq.chr11.bw', sep = ''))
##
sampleData = sampleData %>%
  dplyr::mutate(bigWig = bigwigFiles)
as.data.frame(sampleData)
##
#trackData = dplyr::mutate(sampleData, track_id = condition, colour_group = condition)
trackData = dplyr::mutate(sampleData, track_id = sample_id, colour_group = condition)
##
plotCoverage(dlg2Exons, dlg2Cds, 
             dlg2Meta, trackData,
             heights = c(2,1), fill_palette = getGenotypePalette(),
             rescale_introns = FALSE, 
             region_coords = c(83626956, 83760000),
             coverage_type = 'both')
ggsave(paste(baseRepository, '/sequencing20201217_ccleSequencingDataReprocessing/sashimi/coverage_dlg2RefseqForJunctionsEwingRna.pdf', sep = ''),
         height = 4, width = 4, useDingbats = FALSE)
```


<div style="margin-bottom:50px;"></div>

## Session info

```{r}
##########################################################################################
sessionInfo()
```









