---
title: "Expression analysis of Sole data"
output:
  html_notebook:
      code_folding: none
---

This document details working with Salmon data obtained from analysis of data from the Sole manuscript.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning = FALSE}
##########################################################################################
library('tidyverse')
library('ggplot2')
library('RColorBrewer')
library('tximport')
library('GenomicFeatures')
library('ggrepel')
library('org.Hs.eg.db')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

The first thing to do is to make a sample annotation table.

```{r}
##########################################################################################
samples = read_csv(paste(baseRepository, '/sequencing20211220_stag2CtcfSurdezPmid33930311/rnaSeq/SraRunTable1.txt',sep='')) %>% 
  dplyr::select(Run, `GEO_Accession (exp)`, Genotype, Cell_Line) %>%
  dplyr::rename(accession = 'Run', geo = `GEO_Accession (exp)`) %>%
  dplyr::arrange(accession) %>%
  dplyr::filter(!is.na(Genotype)) %>%
  dplyr::mutate(Genotype = factor(Genotype, levels = unique(Genotype)))
```

<div style="margin-bottom:50px;"></div>

Next we point at our quant files.

```{r}
##########################################################################################
quantFiles = file.path(paste(baseRepository, '/sequencing20211220_stag2CtcfSurdezPmid33930311/rnaSeq/quants', sep = ''), samples$accession, 'quant.sf')
names(quantFiles) = samples$accession
all(file.exists(quantFiles))
```

<div style="margin-bottom:50px;"></div>

Build the transcriptome index.

```{r}
##########################################################################################
txdb = makeTxDbFromGFF('D:/databases/projectEwsDlg2/baseGenomeFiles/genome.gtf')
k = keys(txdb, keytype = "TXNAME")
tx2gene = select(txdb, k, c("GENEID", "TXSTART", "TXEND"), "TXNAME")
```

<div style="margin-bottom:50px;"></div>

Read the transcript data.

```{r}
##########################################################################################
txi = tximport(quantFiles, type = 'salmon', tx2gene = tx2gene, countsFromAbundance = 'lengthScaledTPM')
names(txi)
txiAnnotated = tibble(as.data.frame(txi$counts)) %>%
  mutate('ENSEMBL' = sub('(.*)\\.[0-9]+$', '\\1', row.names(txi$counts)))
```

<div style="margin-bottom:50px;"></div>

Make a plot of these data.

```{r}
##########################################################################################
geneSymbols = select(org.Hs.eg.db, txiAnnotated$ENSEMBL, c('ENSEMBL','SYMBOL'), 'ENSEMBL') 
geneExpData = txiAnnotated %>%
  left_join(geneSymbols) %>%
  mutate(symbol = SYMBOL)

##
targetGenes = c('DLG2')
targetData = geneExpData %>%
  dplyr::filter(symbol %in% targetGenes) %>%
  tidyr::pivot_longer(cols = SRR14217762:SRR9326210, names_to = 'accession', values_to = 'tpm') %>%
  dplyr::select(symbol,accession,tpm) %>%
  dplyr::mutate(tpm = log2(tpm + 1)) %>%
  dplyr::left_join(samples)


##
sampleColors = c(rep(brewer.pal(3,'Blues')[3],41),
                 rep(brewer.pal(3,'OrRd')[3],2),
                 rep(brewer.pal(3,'Blues')[3],11),
                 rep(brewer.pal(3,'OrRd')[3],2),
                 rep(brewer.pal(3,'Blues')[3],3))
                 
ggplot(targetData, aes(Genotype, tpm, color = Genotype)) +
  geom_point(aes(x=Genotype, y=tpm), data = targetData, size = 2, alpha = 0.75, position = position_jitter(w = 0.1, h = 0)) +
  scale_color_manual(values = samples$sampleColors) +
  labs(y = 'log2(TPM + 1)', title = 'DLG2 in EWIma cells') +
  scale_y_continuous(limits = c(0,16), breaks = seq(0,16,2)) +
  theme_classic() +
  theme(legend.position = 'none'
        #axis.title.x=element_blank(),
        #axis.text.x = element_text(angle = 45, hjust = 1)
        )
ggsave(paste(baseRepository, '/sequencing20211220_stag2CtcfSurdezPmid33930311/rnaSeq/scatter_dlg2RnaExpressionLevels.pdf', sep = ''),
      height = 2, width = 2, useDingbats = FALSE)
```


<div style="margin-bottom:50px;"></div>

## Session info

```{r}
##########################################################################################
sessionInfo()
```









