<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.260">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Christopher Hughes">

<title>sorensenSurface</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<script src="sorensenSurface_files/libs/clipboard/clipboard.min.js"></script>
<script src="sorensenSurface_files/libs/quarto-html/quarto.js"></script>
<script src="sorensenSurface_files/libs/quarto-html/popper.min.js"></script>
<script src="sorensenSurface_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="sorensenSurface_files/libs/quarto-html/anchor.min.js"></script>
<link href="sorensenSurface_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link id="quarto-text-highlighting-styles" href="sorensenSurface_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet">
<script src="sorensenSurface_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="sorensenSurface_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="sorensenSurface_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">sorensenSurface</h1>
</div>





<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Christopher Hughes </p>
          </div>
  </div>
    
    
  </div>
  

</header>

<section id="setting-up-the-environment" class="level2">
<h2 class="anchored" data-anchor-id="setting-up-the-environment">Setting up the environment</h2>
<p>These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'tidyverse'</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'limma'</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'RColorBrewer'</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'DESeq2'</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'DEqMS'</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'ggrepel'</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'stringr'</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'Biostrings'</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'OrgMassSpecR'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>generalDatasets <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>baseWorkspace <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>baseRepository <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-processing" class="level2">
<h2 class="anchored" data-anchor-id="data-processing">Data processing</h2>
<p>Read in the peptide spectral match data. The TMT layout data are in the associated tmt_layout_surface.txt file.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="do">##set the file locations</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>sampleInfo <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'plex1'</span>,<span class="st">'plex2'</span>,<span class="st">'plex3'</span>,<span class="st">'plex4'</span>,<span class="st">'plex5'</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>annotation <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'a'</span>,<span class="st">'b'</span>,<span class="st">'c'</span>,<span class="st">'d'</span>,<span class="st">'e'</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>psmFiles <span class="ot">=</span> <span class="fu">file.path</span>(baseRepository, <span class="st">'database20220929_ewsExpressionDatabase/sorensenSurface/fragpipeAnalysis'</span>, sampleInfo, <span class="st">'psm.tsv'</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">file.exists</span>(psmFiles))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">##process the identification data</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>psmData <span class="ot">=</span> <span class="fu">tibble</span>()</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(sampleInfo)){</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  psmTemp <span class="ot">=</span> <span class="fu">read_tsv</span>(psmFiles[i], <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Peptide, <span class="st">`</span><span class="at">Protein ID</span><span class="st">`</span>, Gene, Spectrum) <span class="sc">%&gt;%</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">sequence =</span> Peptide, </span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">accession =</span> <span class="st">`</span><span class="at">Protein ID</span><span class="st">`</span>,</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">symbol =</span> Gene) <span class="sc">%&gt;%</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">dataset =</span> <span class="fu">tolower</span>(sampleInfo[i]),</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">scan =</span> <span class="fu">as.numeric</span>(<span class="fu">sub</span>(<span class="st">'.*Surface_Plex[0-9]</span><span class="sc">\\</span><span class="st">.(.*)</span><span class="sc">\\</span><span class="st">.[0-9]+</span><span class="sc">\\</span><span class="st">.[0-9]+'</span>, <span class="st">'</span><span class="sc">\\</span><span class="st">1'</span>, Spectrum))) <span class="sc">%&gt;%</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Spectrum)</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  <span class="do">##</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  psmData <span class="ot">=</span> <span class="fu">rbind</span>(psmData, psmTemp)</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>psmData</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 57,596 x 5
   sequence      accession symbol  dataset  scan
   &lt;chr&gt;         &lt;chr&gt;     &lt;chr&gt;   &lt;chr&gt;   &lt;dbl&gt;
 1 GGSDGGGSHR    P14209    CD99    plex1    1614
 2 NEEDEGHSNSSPR Q14103    HNRNPD  plex1    1752
 3 TGSQDQEVHR    P17813    ENG     plex1    1781
 4 EQGQNLAR      P61224    RAP1B   plex1    2192
 5 NANAEPAVQR    P14209    CD99    plex1    2212
 6 TDNSESCVHGK   P22413    ENPP1   plex1    2213
 7 EEQTPQNK      P00338    LDHA    plex1    2284
 8 HWDQNER       P08195    SLC3A2  plex1    2325
 9 DTDGGPK       P53985    SLC16A1 plex1    2357
10 KPAGAAK       P10412    H1-4    plex1    2369
# ... with 57,586 more rows</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">##this is the function we will use to process the quant data</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>combineQuantFiles <span class="ot">=</span> <span class="cf">function</span>(filePath, ...){</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  quantData <span class="ot">=</span> <span class="fu">read_tsv</span>(filePath, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(MS2ScanNumber, <span class="st">`</span><span class="at">126Intensity</span><span class="st">`</span><span class="sc">:</span><span class="st">`</span><span class="at">131CIntensity</span><span class="st">`</span>)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(quantData) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'scan'</span>,<span class="st">'tmt126C'</span>,<span class="st">'tmt127N'</span>,<span class="st">'tmt127C'</span>,<span class="st">'tmt128N'</span>,<span class="st">'tmt128C'</span>,<span class="st">'tmt129N'</span>,</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>                          <span class="st">'tmt129C'</span>,<span class="st">'tmt130N'</span>,<span class="st">'tmt130C'</span>,<span class="st">'tmt131N'</span>,<span class="st">'tmt131C'</span>)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  <span class="do">##</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  dataset <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">'.*EwingSurface_(.*)</span><span class="sc">\\</span><span class="st">.raw_Matrix</span><span class="sc">\\</span><span class="st">.txt'</span>, <span class="st">'</span><span class="sc">\\</span><span class="st">1'</span>, filePath)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  quantData<span class="sc">$</span>dataset <span class="ot">=</span> <span class="fu">tolower</span>(dataset)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  <span class="do">##</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(quantData)</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="do">##process the quant data</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>quantFiles <span class="ot">=</span> <span class="fu">list.files</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/quants/'</span>, <span class="at">sep =</span> <span class="st">''</span>),</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pattern =</span> <span class="st">'_Matrix.txt'</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>quantDataSet <span class="ot">=</span> <span class="fu">lapply</span>(quantFiles, combineQuantFiles)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>New names:
New names:
New names:
New names:
New names:
* `` -&gt; `...102`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>allQuantData <span class="ot">=</span> <span class="fu">do.call</span>(<span class="st">'rbind'</span>, quantDataSet)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="do">##combine all of the data</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>psmQuant <span class="ot">=</span> psmData <span class="sc">%&gt;%</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(allQuantData)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = c("dataset", "scan")</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">##filter the psm data</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>psmQuant <span class="ot">=</span> psmQuant <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sampleSignal =</span> <span class="fu">sum</span>(tmt126C<span class="sc">:</span>tmt131C, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(sampleSignal <span class="sc">&gt;=</span> <span class="dv">1000</span>)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="do">##roll into proteins</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>pro <span class="ot">=</span> psmQuant <span class="sc">%&gt;%</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>sampleSignal) <span class="sc">%&gt;%</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">psmCount =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(sequence, accession, symbol, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="fu">across</span>(tmt126C<span class="sc">:</span>tmt131C, mean, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">psmCount =</span> <span class="fu">sum</span>(psmCount, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pepCount =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset, accession, symbol) <span class="sc">%&gt;%</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="fu">across</span>(tmt126C<span class="sc">:</span>tmt131C, median, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">psmCount =</span> <span class="fu">sum</span>(psmCount, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">pepCount =</span> <span class="fu">sum</span>(pepCount, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'sequence', 'accession', 'symbol'. You can
override using the `.groups` argument.
`summarise()` has grouped output by 'dataset', 'accession'. You can override
using the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="do">##save the data</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pro[pro<span class="sc">$</span>dataset <span class="sc">==</span> <span class="st">'plex1'</span>,], <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_rawProteinDatasetA.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pro[pro<span class="sc">$</span>dataset <span class="sc">==</span> <span class="st">'plex2'</span>,], <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_rawProteinDatasetB.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pro[pro<span class="sc">$</span>dataset <span class="sc">==</span> <span class="st">'plex3'</span>,], <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_rawProteinDatasetC.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pro[pro<span class="sc">$</span>dataset <span class="sc">==</span> <span class="st">'plex4'</span>,], <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_rawProteinDatasetD.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pro[pro<span class="sc">$</span>dataset <span class="sc">==</span> <span class="st">'plex5'</span>,], <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_rawProteinDatasetE.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Perform the IRS normalization.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="do">##read in the annotation data</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>tmtAnno <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/tmt_layout_surface.txt'</span>, <span class="at">sep =</span> <span class="st">''</span>), <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(id, plex, label) <span class="sc">%&gt;%</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">dataset =</span> plex,</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">sampleName =</span> id) <span class="sc">%&gt;%</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">dataset =</span> <span class="fu">paste</span>(<span class="st">'plex'</span>, dataset, <span class="at">sep =</span> <span class="st">''</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sampleName =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">'EMPTY'</span>, sampleName), <span class="fu">paste</span>(sampleName,<span class="st">'_'</span>,dataset,<span class="st">'_'</span>,label, <span class="at">sep =</span> <span class="st">''</span>),</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">'PIS'</span>, sampleName), sampleName, <span class="fu">paste</span>(<span class="st">'ews_'</span>,sampleName, <span class="at">sep =</span> <span class="st">''</span>))))</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="do">##read in the data and combine with the annotation</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>proteinSetA <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_rawProteinDatasetA.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">psma =</span> psmCount,</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">pepa =</span> pepCount)</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(proteinSetA)[<span class="dv">3</span><span class="sc">:</span><span class="dv">13</span>] <span class="ot">=</span> tmtAnno[tmtAnno<span class="sc">$</span>dataset <span class="sc">==</span> <span class="st">'plex1'</span>,]<span class="sc">$</span>sampleName</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>proteinSetB <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_rawProteinDatasetB.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">psmb =</span> psmCount,</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>                <span class="at">pepb =</span> pepCount)</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(proteinSetB)[<span class="dv">3</span><span class="sc">:</span><span class="dv">13</span>] <span class="ot">=</span> tmtAnno[tmtAnno<span class="sc">$</span>dataset <span class="sc">==</span> <span class="st">'plex2'</span>,]<span class="sc">$</span>sampleName</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>proteinSetC <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_rawProteinDatasetC.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">psmc =</span> psmCount,</span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>                <span class="at">pepc =</span> pepCount)</span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(proteinSetC)[<span class="dv">3</span><span class="sc">:</span><span class="dv">13</span>] <span class="ot">=</span> tmtAnno[tmtAnno<span class="sc">$</span>dataset <span class="sc">==</span> <span class="st">'plex3'</span>,]<span class="sc">$</span>sampleName</span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>proteinSetD <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_rawProteinDatasetD.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">psmd =</span> psmCount,</span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>                <span class="at">pepd =</span> pepCount)</span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(proteinSetD)[<span class="dv">3</span><span class="sc">:</span><span class="dv">13</span>] <span class="ot">=</span> tmtAnno[tmtAnno<span class="sc">$</span>dataset <span class="sc">==</span> <span class="st">'plex4'</span>,]<span class="sc">$</span>sampleName</span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>proteinSetE <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_rawProteinDatasetE.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span></span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">psme =</span> psmCount,</span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>                <span class="at">pepe =</span> pepCount)</span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(proteinSetE)[<span class="dv">3</span><span class="sc">:</span><span class="dv">13</span>] <span class="ot">=</span> tmtAnno[tmtAnno<span class="sc">$</span>dataset <span class="sc">==</span> <span class="st">'plex5'</span>,]<span class="sc">$</span>sampleName</span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a><span class="do">##join all of the data</span></span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>proteinSet <span class="ot">=</span> proteinSetA <span class="sc">%&gt;%</span></span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(proteinSetB) <span class="sc">%&gt;%</span></span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(proteinSetC) <span class="sc">%&gt;%</span></span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(proteinSetD) <span class="sc">%&gt;%</span></span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(proteinSetE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = c("accession", "symbol")
Joining, by = c("accession", "symbol")
Joining, by = c("accession", "symbol")
Joining, by = c("accession", "symbol")</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">##raw data check</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>targetSamples <span class="ot">=</span> proteinSet[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="fu">which</span>(<span class="fu">grepl</span>(<span class="st">'ews'</span>, <span class="fu">colnames</span>(proteinSet))))]</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>pooledSamples <span class="ot">=</span> proteinSet[,<span class="fu">c</span>(<span class="fu">which</span>(<span class="fu">grepl</span>(<span class="st">'PIS'</span>, <span class="fu">colnames</span>(proteinSet))))]</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>psmSamples <span class="ot">=</span> proteinSet[,<span class="fu">c</span>(<span class="fu">which</span>(<span class="fu">grepl</span>(<span class="st">'psm'</span>, <span class="fu">colnames</span>(proteinSet))))]</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>rawData <span class="ot">=</span> <span class="fu">cbind</span>(targetSamples, pooledSamples, psmSamples)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>rawData[rawData <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>rawData <span class="ot">=</span> <span class="fu">subset</span>(rawData, <span class="fu">rowSums</span>(<span class="fu">is.na</span>(rawData[,<span class="dv">3</span><span class="sc">:</span><span class="dv">45</span>])) <span class="sc">&lt;</span> <span class="dv">20</span>)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">log2</span>(rawData[,<span class="dv">3</span><span class="sc">:</span><span class="dv">50</span>]), <span class="at">col =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">'red'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'green'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'blue'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'purple'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'pink'</span>,<span class="dv">9</span>),<span class="fu">rep</span>(<span class="st">'grey'</span>,<span class="dv">5</span>)), </span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">notch =</span> <span class="cn">TRUE</span>, <span class="at">main =</span> <span class="st">'Raw data'</span>,</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="st">'TMT Samples'</span>, <span class="at">ylab =</span> <span class="st">'log2 of Intensity'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="sorensenSurface_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDensities</span>(<span class="fu">log2</span>(rawData[,<span class="dv">3</span><span class="sc">:</span><span class="dv">50</span>]), <span class="at">col =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">'red'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'green'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'blue'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'purple'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'pink'</span>,<span class="dv">9</span>),<span class="fu">rep</span>(<span class="st">'grey'</span>,<span class="dv">5</span>)), </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">main =</span> <span class="st">'Raw data'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="sorensenSurface_files/figure-html/unnamed-chunk-4-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">format</span>(<span class="fu">round</span>(<span class="fu">colSums</span>(rawData[,<span class="dv">3</span><span class="sc">:</span><span class="dv">50</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">digits =</span> <span class="dv">0</span>), <span class="at">big.mark =</span> <span class="st">","</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     ews_CT1_T2      ews_ES1_R1   ews_CHLA10_T1      ews_ES6_R2      ews_EW9_T1 
"1,151,844,785" "1,213,331,343" "1,552,198,245" "1,346,825,048" "1,287,370,680" 
     ews_ES6_R1      ews_EW8_R1      ews_ES8_R1 ews_Supermix_R1  ews_CHLA258_R2 
"  845,384,301" "  914,612,594" "1,432,704,803" "1,576,698,153" "1,981,320,774" 
     ews_ES1_R2      ews_EW5_R1      ews_ES4_R2      ews_TC7_R1  ews_NCHEWS1_R1 
"1,126,011,383" "  904,932,754" "  275,476,469" "  538,563,283" "  787,840,291" 
   ews_SKNEP_R1 ews_Supermix_R2     ews_TC32_T1     ews_EW12_T1      ews_ES4_R1 
"1,602,715,129" "1,915,804,753" "2,232,556,323" "1,523,820,475" "  476,889,796" 
    ews_EW13_T1      ews_EW8_R2      ews_ES3_T1      ews_CT1_T1      ews_ES7_T1 
"1,789,585,246" "  959,026,237" "  986,212,150" "1,361,820,968" "  682,098,487" 
ews_Supermix_R3      ews_EW5_R2      ews_ES2_T2  ews_NCHEWS1_R2      ews_ES3_T2 
"1,921,774,462" "1,337,327,717" "1,176,072,870" "  750,437,943" "  978,917,802" 
     ews_EW9_T2   ews_CHLA10_T2  ews_CHLA258_R1 ews_Supermix_R4    ews_SKNEP_R2 
"1,454,878,810" "1,757,800,549" "2,690,919,567" "2,001,078,003" "  457,983,642" 
     ews_TC7_R2      ews_ES2_T1     ews_EW12_T2     ews_TC32_T2      ews_ES7_T2 
"  390,723,897" "1,023,679,273" "1,485,786,621" "2,511,822,080" "  731,123,261" 
     ews_ES8_R2     ews_EW13_T2 ews_Supermix_R5           PIS_1           PIS_2 
"  720,187,324" "1,891,655,738" "1,737,918,646" "1,370,159,259" "1,330,845,548" 
          PIS_3           PIS_4           PIS_5 
"1,368,666,545" "1,599,307,819" "1,367,114,572" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="do">##sample loading normalization</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>exp1Raw <span class="ot">=</span> rawData[<span class="fu">c</span>(<span class="dv">3</span><span class="sc">:</span><span class="dv">11</span>)]</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>exp2Raw <span class="ot">=</span> rawData[<span class="fu">c</span>(<span class="dv">12</span><span class="sc">:</span><span class="dv">19</span>)]</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>exp3Raw <span class="ot">=</span> rawData[<span class="fu">c</span>(<span class="dv">20</span><span class="sc">:</span><span class="dv">28</span>)]</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>exp4Raw <span class="ot">=</span> rawData[<span class="fu">c</span>(<span class="dv">29</span><span class="sc">:</span><span class="dv">36</span>)]</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>exp5Raw <span class="ot">=</span> rawData[<span class="fu">c</span>(<span class="dv">37</span><span class="sc">:</span><span class="dv">45</span>)]</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>target <span class="ot">=</span> <span class="fu">mean</span>(<span class="fu">c</span>(<span class="fu">colSums</span>(exp1Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), </span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colSums</span>(exp2Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), </span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colSums</span>(exp3Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), </span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colSums</span>(exp4Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), </span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colSums</span>(exp5Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>normFacs <span class="ot">=</span> target <span class="sc">/</span> <span class="fu">colSums</span>(exp1Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>exp1Sl <span class="ot">=</span> <span class="fu">sweep</span>(exp1Raw, <span class="dv">2</span>, normFacs, <span class="at">FUN =</span> <span class="st">"*"</span>)</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>normFacs <span class="ot">=</span> target <span class="sc">/</span> <span class="fu">colSums</span>(exp2Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>exp2Sl <span class="ot">=</span> <span class="fu">sweep</span>(exp2Raw, <span class="dv">2</span>, normFacs, <span class="at">FUN =</span> <span class="st">"*"</span>)</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>normFacs <span class="ot">=</span> target <span class="sc">/</span> <span class="fu">colSums</span>(exp3Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>exp3Sl <span class="ot">=</span> <span class="fu">sweep</span>(exp3Raw, <span class="dv">2</span>, normFacs, <span class="at">FUN =</span> <span class="st">"*"</span>)</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>normFacs <span class="ot">=</span> target <span class="sc">/</span> <span class="fu">colSums</span>(exp4Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>exp4Sl <span class="ot">=</span> <span class="fu">sweep</span>(exp4Raw, <span class="dv">2</span>, normFacs, <span class="at">FUN =</span> <span class="st">"*"</span>)</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>normFacs <span class="ot">=</span> target <span class="sc">/</span> <span class="fu">colSums</span>(exp5Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>exp5Sl <span class="ot">=</span> <span class="fu">sweep</span>(exp5Raw, <span class="dv">2</span>, normFacs, <span class="at">FUN =</span> <span class="st">"*"</span>)</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>dataSl <span class="ot">=</span> <span class="fu">cbind</span>(exp1Sl, exp2Sl, exp3Sl, exp4Sl, exp5Sl)</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">log2</span>(dataSl), <span class="at">col =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">'red'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'green'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'blue'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'purple'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'pink'</span>,<span class="dv">9</span>),<span class="fu">rep</span>(<span class="st">'grey'</span>,<span class="dv">5</span>)), </span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">notch =</span> <span class="cn">TRUE</span>, <span class="at">main =</span> <span class="st">"Sample Loading (SL) normalized data"</span>,</span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="st">'TMT Sample'</span>, <span class="at">ylab =</span> <span class="st">'log2 of Intensity'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="sorensenSurface_files/figure-html/unnamed-chunk-4-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDensities</span>(<span class="fu">log2</span>(dataSl), <span class="at">col =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">'red'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'green'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'blue'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'purple'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'pink'</span>,<span class="dv">9</span>),<span class="fu">rep</span>(<span class="st">'grey'</span>,<span class="dv">5</span>)), <span class="at">main =</span> <span class="st">"SL normalization"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="sorensenSurface_files/figure-html/unnamed-chunk-4-4.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="do">##IRS normalization</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>irs <span class="ot">=</span> <span class="fu">tibble</span>(rawData[,<span class="dv">46</span><span class="sc">:</span><span class="dv">50</span>])</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(irs) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'sum1'</span>,<span class="st">'sum2'</span>,<span class="st">'sum3'</span>,<span class="st">'sum4'</span>,<span class="st">'sum5'</span>)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>irs<span class="sc">$</span>average <span class="ot">=</span> <span class="fu">apply</span>(irs, <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="fu">exp</span>(<span class="fu">mean</span>(<span class="fu">log</span>(x),<span class="at">na.rm =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>irs<span class="sc">$</span>fac1 <span class="ot">=</span> irs<span class="sc">$</span>average <span class="sc">/</span> irs<span class="sc">$</span>sum1</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>irs<span class="sc">$</span>fac2 <span class="ot">=</span> irs<span class="sc">$</span>average <span class="sc">/</span> irs<span class="sc">$</span>sum2</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>irs<span class="sc">$</span>fac3 <span class="ot">=</span> irs<span class="sc">$</span>average <span class="sc">/</span> irs<span class="sc">$</span>sum3</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>irs<span class="sc">$</span>fac4 <span class="ot">=</span> irs<span class="sc">$</span>average <span class="sc">/</span> irs<span class="sc">$</span>sum4</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>irs<span class="sc">$</span>fac5 <span class="ot">=</span> irs<span class="sc">$</span>average <span class="sc">/</span> irs<span class="sc">$</span>sum5</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>dataIrs <span class="ot">=</span> exp1Sl <span class="sc">*</span> irs<span class="sc">$</span>fac1</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>dataIrs <span class="ot">=</span> <span class="fu">cbind</span>(dataIrs, exp2Sl <span class="sc">*</span> irs<span class="sc">$</span>fac2)</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>dataIrs <span class="ot">=</span> <span class="fu">cbind</span>(dataIrs, exp3Sl <span class="sc">*</span> irs<span class="sc">$</span>fac3)</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>dataIrs <span class="ot">=</span> <span class="fu">cbind</span>(dataIrs, exp4Sl <span class="sc">*</span> irs<span class="sc">$</span>fac4)</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>dataIrs <span class="ot">=</span> <span class="fu">cbind</span>(dataIrs, exp5Sl <span class="sc">*</span> irs<span class="sc">$</span>fac5)</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">log2</span>(dataIrs), <span class="at">col =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">'red'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'green'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'blue'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'purple'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'pink'</span>,<span class="dv">9</span>),<span class="fu">rep</span>(<span class="st">'grey'</span>,<span class="dv">5</span>)), </span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">main =</span> <span class="st">"Internal Reference Scaling (IRS) normalized data"</span>,</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="st">'TMT Sample'</span>, <span class="at">ylab =</span> <span class="st">'log2 of Intensity'</span>, <span class="at">notch =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="sorensenSurface_files/figure-html/unnamed-chunk-4-5.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDensities</span>(<span class="fu">log2</span>(dataIrs), <span class="at">col =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">'red'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'green'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'blue'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'purple'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'pink'</span>,<span class="dv">9</span>),<span class="fu">rep</span>(<span class="st">'grey'</span>,<span class="dv">5</span>)), <span class="at">main =</span> <span class="st">"IRS data"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="sorensenSurface_files/figure-html/unnamed-chunk-4-6.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">format</span>(<span class="fu">round</span>(<span class="fu">colSums</span>(dataIrs, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">digits =</span> <span class="dv">0</span>), <span class="at">big.mark =</span> <span class="st">","</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     ews_CT1_T2      ews_ES1_R1   ews_CHLA10_T1      ews_ES6_R2      ews_EW9_T1 
"1,252,317,111" "1,261,715,567" "1,261,814,003" "1,269,529,617" "1,269,698,869" 
     ews_ES6_R1      ews_EW8_R1      ews_ES8_R1 ews_Supermix_R1  ews_CHLA258_R2 
"1,259,794,772" "1,267,073,509" "1,257,141,427" "1,268,374,554" "1,350,940,161" 
     ews_ES1_R2      ews_EW5_R1      ews_ES4_R2      ews_TC7_R1  ews_NCHEWS1_R1 
"1,333,176,140" "1,347,154,740" "1,336,093,883" "1,339,955,317" "1,331,792,326" 
   ews_SKNEP_R1 ews_Supermix_R2     ews_TC32_T1     ews_EW12_T1      ews_ES4_R1 
"1,354,486,884" "1,360,714,874" "1,270,505,644" "1,270,915,455" "1,263,320,499" 
    ews_EW13_T1      ews_EW8_R2      ews_ES3_T1      ews_CT1_T1      ews_ES7_T1 
"1,275,316,601" "1,262,615,473" "1,253,838,248" "1,263,225,940" "1,247,729,638" 
ews_Supermix_R3      ews_EW5_R2      ews_ES2_T2  ews_NCHEWS1_R2      ews_ES3_T2 
"1,258,236,476" "1,095,219,562" "1,094,312,882" "1,076,652,167" "1,086,233,167" 
     ews_EW9_T2   ews_CHLA10_T2  ews_CHLA258_R1 ews_Supermix_R4    ews_SKNEP_R2 
"1,086,756,504" "1,076,967,398" "1,080,069,026" "1,089,083,106" "1,233,186,959" 
     ews_TC7_R2      ews_ES2_T1     ews_EW12_T2     ews_TC32_T2      ews_ES7_T2 
"1,235,332,496" "1,249,800,977" "1,240,697,758" "1,241,361,011" "1,236,709,165" 
     ews_ES8_R2     ews_EW13_T2 ews_Supermix_R5 
"1,236,031,812" "1,243,516,989" "1,245,971,910" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="do">##post-normalization clustering</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>colVec <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">'red'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'green'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'blue'</span>,<span class="dv">9</span>), <span class="fu">rep</span>(<span class="st">'purple'</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">'pink'</span>,<span class="dv">9</span>))</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">log2</span>(dataIrs), <span class="at">col =</span> colVec, <span class="at">main =</span> <span class="st">"IRS clusters group by condition"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="sorensenSurface_files/figure-html/unnamed-chunk-4-7.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Recombine the data with the accessions and save the final file.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>finalIrsData <span class="ot">=</span> <span class="fu">as_tibble</span>(<span class="fu">cbind</span>(rawData[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>,<span class="dv">51</span><span class="sc">:</span><span class="dv">55</span>)], dataIrs))</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="do">##save data</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(finalIrsData, <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_proteinSetABCDEIrsNormalized.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Reshape the data and output the final dataset.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="do">##calculate some final scores, using the iBAQ value as a score in this case</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>finalData <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">readRDS</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_proteinSetABCDEIrsNormalized.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>)))</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="do">##for the iBAQ calculation, we need the number of theoretical tryptic peptides</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>proteinFasta <span class="ot">=</span> <span class="fu">readAAStringSet</span>(<span class="st">'D:/databases/projectEwsDlg2/proteomicsIndex/2022-09-27-reviewed-contam-UP000005640.fas'</span>)</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>proteinDatabase <span class="ot">=</span> <span class="fu">tibble</span>(<span class="st">'proteinNames'</span> <span class="ot">=</span> <span class="fu">names</span>(proteinFasta)) <span class="sc">%&gt;%</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">accession =</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(proteinNames, <span class="st">'</span><span class="sc">\\</span><span class="st">|'</span>), <span class="st">'['</span>, <span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">length =</span> <span class="fu">width</span>(proteinFasta)) <span class="sc">%&gt;%</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>proteinNames)</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>trypticPeptides <span class="ot">=</span> <span class="fu">vector</span>()</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>detectableLength <span class="ot">=</span> <span class="fu">vector</span>()</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(proteinFasta)){</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>  aaSeq <span class="ot">=</span> <span class="fu">as.character</span>(proteinFasta[[i]])</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>  seqDigest <span class="ot">=</span> OrgMassSpecR<span class="sc">::</span><span class="fu">Digest</span>(aaSeq, <span class="at">enzyme =</span> <span class="st">'trypsin'</span>, <span class="at">missed =</span> <span class="dv">0</span>, <span class="at">custom =</span> <span class="fu">list</span>(<span class="at">code =</span> <span class="fu">c</span>(<span class="st">'X'</span>,<span class="st">'U'</span>,<span class="st">'Z'</span>,<span class="st">'B'</span>), <span class="at">mass =</span> <span class="fu">c</span>(<span class="dv">50</span>, <span class="dv">60</span>, <span class="dv">70</span>, <span class="dv">80</span>)))</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>  seqDigest<span class="sc">$</span>pepLength <span class="ot">=</span> (seqDigest<span class="sc">$</span>stop <span class="sc">-</span> seqDigest<span class="sc">$</span>start) <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>  seqDigestSub <span class="ot">=</span> <span class="fu">subset</span>(seqDigest, (seqDigest<span class="sc">$</span>pepLength <span class="sc">&gt;</span> <span class="dv">5</span>) <span class="sc">&amp;</span> (seqDigest<span class="sc">$</span>pepLength <span class="sc">&lt;</span> <span class="dv">31</span>))</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>  trypticPeptides <span class="ot">=</span> <span class="fu">c</span>(trypticPeptides, <span class="fu">nrow</span>(seqDigestSub))</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>  detectableLength <span class="ot">=</span> <span class="fu">c</span>(detectableLength, <span class="fu">sum</span>(seqDigestSub<span class="sc">$</span>pepLength, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in OrgMassSpecR::Digest(aaSeq, enzyme = "trypsin", missed = 0, custom =
list(code = c("X", : sequence does not contain cleavage sites

Warning in OrgMassSpecR::Digest(aaSeq, enzyme = "trypsin", missed = 0, custom =
list(code = c("X", : sequence does not contain cleavage sites

Warning in OrgMassSpecR::Digest(aaSeq, enzyme = "trypsin", missed = 0, custom =
list(code = c("X", : sequence does not contain cleavage sites

Warning in OrgMassSpecR::Digest(aaSeq, enzyme = "trypsin", missed = 0, custom =
list(code = c("X", : sequence does not contain cleavage sites

Warning in OrgMassSpecR::Digest(aaSeq, enzyme = "trypsin", missed = 0, custom =
list(code = c("X", : sequence does not contain cleavage sites

Warning in OrgMassSpecR::Digest(aaSeq, enzyme = "trypsin", missed = 0, custom =
list(code = c("X", : sequence does not contain cleavage sites

Warning in OrgMassSpecR::Digest(aaSeq, enzyme = "trypsin", missed = 0, custom =
list(code = c("X", : sequence does not contain cleavage sites

Warning in OrgMassSpecR::Digest(aaSeq, enzyme = "trypsin", missed = 0, custom =
list(code = c("X", : sequence does not contain cleavage sites

Warning in OrgMassSpecR::Digest(aaSeq, enzyme = "trypsin", missed = 0, custom =
list(code = c("X", : sequence does not contain cleavage sites

Warning in OrgMassSpecR::Digest(aaSeq, enzyme = "trypsin", missed = 0, custom =
list(code = c("X", : sequence does not contain cleavage sites

Warning in OrgMassSpecR::Digest(aaSeq, enzyme = "trypsin", missed = 0, custom =
list(code = c("X", : sequence does not contain cleavage sites

Warning in OrgMassSpecR::Digest(aaSeq, enzyme = "trypsin", missed = 0, custom =
list(code = c("X", : sequence does not contain cleavage sites</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co">#now add to the database</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>proteinDatabase<span class="sc">$</span>detectablePeptides <span class="ot">=</span> trypticPeptides</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>proteinDatabase<span class="sc">$</span>detectableLength <span class="ot">=</span> detectableLength</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="do">##incorporate with the proteomics data</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>finalDataSub <span class="ot">=</span> <span class="fu">tibble</span>(<span class="fu">subset</span>(finalData, <span class="fu">rowSums</span>(<span class="fu">is.na</span>(finalData[,<span class="dv">3</span><span class="sc">:</span><span class="dv">7</span>])) <span class="sc">&lt;</span> <span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">names</span>(finalData)[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">'Super'</span>, <span class="fu">colnames</span>(finalData))]) <span class="sc">%&gt;%</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(accession, symbol, ews_CT1_T2<span class="sc">:</span>ews_EW13_T2) <span class="sc">%&gt;%</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(proteinDatabase) <span class="sc">%&gt;%</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">meanIntensity =</span> <span class="fu">rowMeans</span>(<span class="fu">across</span>(ews_CT1_T2<span class="sc">:</span>ews_EW13_T2), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sorensenSurface =</span> <span class="fu">log10</span>(meanIntensity <span class="sc">/</span> detectablePeptides)) <span class="sc">%&gt;%</span></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(accession, symbol, sorensenSurface)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "accession"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="do">##write the data</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(finalDataSub, <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_sorensenSurface.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(finalDataSub, </span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>            <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenSurface/dataset_sorensenSurface.csv'</span>, <span class="at">sep =</span> <span class="st">''</span>),</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">','</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Finish up.</p>
<section id="session-info" class="level3">
<h3 class="anchored" data-anchor-id="session-info">Session info</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.1.3 (2022-03-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_Canada.1252  LC_CTYPE=English_Canada.1252   
[3] LC_MONETARY=English_Canada.1252 LC_NUMERIC=C                   
[5] LC_TIME=English_Canada.1252    

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] OrgMassSpecR_0.5-3          Biostrings_2.62.0          
 [3] XVector_0.34.0              ggrepel_0.9.2              
 [5] DEqMS_1.12.1                DESeq2_1.34.0              
 [7] SummarizedExperiment_1.24.0 Biobase_2.54.0             
 [9] MatrixGenerics_1.6.0        matrixStats_0.62.0         
[11] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
[13] IRanges_2.28.0              S4Vectors_0.32.4           
[15] BiocGenerics_0.40.0         RColorBrewer_1.1-3         
[17] limma_3.50.3                forcats_0.5.2              
[19] stringr_1.4.1               dplyr_1.0.10               
[21] purrr_0.3.5                 readr_2.1.3                
[23] tidyr_1.2.1                 tibble_3.1.8               
[25] ggplot2_3.4.0               tidyverse_1.3.2            

loaded via a namespace (and not attached):
 [1] bitops_1.0-7           fs_1.5.2               bit64_4.0.5           
 [4] lubridate_1.9.0        httr_1.4.4             tools_4.1.3           
 [7] backports_1.4.1        utf8_1.2.2             R6_2.5.1              
[10] DBI_1.1.3              colorspace_2.0-3       withr_2.5.0           
[13] tidyselect_1.2.0       bit_4.0.5              compiler_4.1.3        
[16] cli_3.4.1              rvest_1.0.3            xml2_1.3.3            
[19] DelayedArray_0.20.0    scales_1.2.1           genefilter_1.76.0     
[22] digest_0.6.30          rmarkdown_2.18         pkgconfig_2.0.3       
[25] htmltools_0.5.3        dbplyr_2.2.1           fastmap_1.1.0         
[28] htmlwidgets_1.5.4      rlang_1.0.6            readxl_1.4.1          
[31] RSQLite_2.2.18         rstudioapi_0.14        generics_0.1.3        
[34] jsonlite_1.8.3         vroom_1.6.0            BiocParallel_1.28.3   
[37] googlesheets4_1.0.1    RCurl_1.98-1.9         magrittr_2.0.3        
[40] GenomeInfoDbData_1.2.7 Matrix_1.5-3           Rcpp_1.0.9            
[43] munsell_0.5.0          fansi_1.0.3            lifecycle_1.0.3       
[46] stringi_1.7.8          yaml_2.3.6             zlibbioc_1.40.0       
[49] blob_1.2.3             parallel_4.1.3         crayon_1.5.2          
[52] lattice_0.20-45        splines_4.1.3          haven_2.5.1           
[55] annotate_1.72.0        KEGGREST_1.34.0        hms_1.1.2             
[58] locfit_1.5-9.6         knitr_1.40             pillar_1.8.1          
[61] geneplotter_1.72.0     reprex_2.0.2           XML_3.99-0.12         
[64] glue_1.6.2             evaluate_0.18          modelr_0.1.10         
[67] png_0.1-7              vctrs_0.5.1            tzdb_0.3.0            
[70] cellranger_1.1.0       gtable_0.3.1           assertthat_0.2.1      
[73] cachem_1.0.6           xfun_0.34              xtable_1.8-4          
[76] broom_1.0.1            survival_3.4-0         googledrive_2.0.0     
[79] gargle_1.2.1           memoise_2.0.1          AnnotationDbi_1.56.2  
[82] timechange_0.1.1       ellipsis_0.3.2        </code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    setTimeout(function() {
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>