<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.260">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Christopher Hughes">

<title>sorensenTimecourseProtein</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<script src="sorensenTimecourseProtein_files/libs/clipboard/clipboard.min.js"></script>
<script src="sorensenTimecourseProtein_files/libs/quarto-html/quarto.js"></script>
<script src="sorensenTimecourseProtein_files/libs/quarto-html/popper.min.js"></script>
<script src="sorensenTimecourseProtein_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="sorensenTimecourseProtein_files/libs/quarto-html/anchor.min.js"></script>
<link href="sorensenTimecourseProtein_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link id="quarto-text-highlighting-styles" href="sorensenTimecourseProtein_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet">
<script src="sorensenTimecourseProtein_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="sorensenTimecourseProtein_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="sorensenTimecourseProtein_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">sorensenTimecourseProtein</h1>
</div>





<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Christopher Hughes </p>
          </div>
  </div>
    
    
  </div>
  

</header>

<section id="setting-up-the-environment" class="level2">
<h2 class="anchored" data-anchor-id="setting-up-the-environment">Setting up the environment</h2>
<p>These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'tidyverse'</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'limma'</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'RColorBrewer'</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'DESeq2'</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'DEqMS'</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'ggrepel'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>generalDatasets <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>baseWorkspace <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>baseRepository <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-processing" class="level2">
<h2 class="anchored" data-anchor-id="data-processing">Data processing</h2>
<p>Read in the peptide spectral match data. The TMT layout is as follows:</p>
<p>126 pool 127N empty 127C empty 128N day0 128C day7 129N day9 129C day10 130N day11 130C day14 131N day17 131C day22 132N empty 132C empty 133N pool 133C empty 134N empty</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="do">##set the file locations</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>sampleInfo <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'setA'</span>,<span class="st">'setB'</span>,<span class="st">'setC'</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>annotation <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'a'</span>,<span class="st">'b'</span>,<span class="st">'c'</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>psmFiles <span class="ot">=</span> <span class="fu">file.path</span>(baseRepository, <span class="st">'database20220929_ewsExpressionDatabase/sorensenTimecourseProtein/fragpipeAnalysis'</span>, sampleInfo, <span class="st">'psm.tsv'</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">file.exists</span>(psmFiles))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">##process the identification data</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>psmData <span class="ot">=</span> <span class="fu">tibble</span>()</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(sampleInfo)){</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  psmTemp <span class="ot">=</span> <span class="fu">read_tsv</span>(psmFiles[i], <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Peptide, <span class="st">`</span><span class="at">Protein ID</span><span class="st">`</span>, Gene, Spectrum) <span class="sc">%&gt;%</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">sequence =</span> Peptide, </span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">accession =</span> <span class="st">`</span><span class="at">Protein ID</span><span class="st">`</span>,</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">symbol =</span> Gene) <span class="sc">%&gt;%</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">dataset =</span> <span class="fu">tolower</span>(sampleInfo[i]),</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">scan =</span> <span class="fu">as.numeric</span>(<span class="fu">sub</span>(<span class="st">'.*_HpH_[0-9]+</span><span class="sc">\\</span><span class="st">.(.*)</span><span class="sc">\\</span><span class="st">.[0-9]+</span><span class="sc">\\</span><span class="st">.[0-9]+'</span>, <span class="st">'</span><span class="sc">\\</span><span class="st">1'</span>, Spectrum)),</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">fraction =</span> <span class="fu">sub</span>(<span class="st">'.*_HpH_(.*)</span><span class="sc">\\</span><span class="st">.[0-9]+</span><span class="sc">\\</span><span class="st">.[0-9]+</span><span class="sc">\\</span><span class="st">.[0-9]+'</span>, <span class="st">'</span><span class="sc">\\</span><span class="st">1'</span>, Spectrum)) <span class="sc">%&gt;%</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Spectrum)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  <span class="do">##</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  psmData <span class="ot">=</span> <span class="fu">rbind</span>(psmData, psmTemp)</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>psmData</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 556,568 x 6
   sequence      accession symbol   dataset  scan fraction
   &lt;chr&gt;         &lt;chr&gt;     &lt;chr&gt;    &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;   
 1 ALVELNGVSLIPK Q9UHL9    GTF2IRD1 seta     2633 1       
 2 THQQQHQAAVR   Q6ZN55    ZNF574   seta     2730 1       
 3 MNTNPSR       P40429    RPL13A   seta     2766 1       
 4 AHATGAGPAGR   Q9Y490    TLN1     seta     3081 1       
 5 VTSTQQQTR     P49023    PXN      seta     3105 1       
 6 NVSSATHSAR    Q9UPN4    CEP131   seta     3109 1       
 7 TIEAHSR       Q16555    DPYSL2   seta     3114 1       
 8 RAEATAASTSR   O95721    SNAP29   seta     3118 1       
 9 SGGGGHGSSR    O00571    DDX3X    seta     3124 1       
10 RSTITSR       Q99880    H2BC13   seta     3136 1       
# ... with 556,558 more rows</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">##this is the function we will use to process the quant data</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>combineQuantFiles <span class="ot">=</span> <span class="cf">function</span>(filePath, ...){</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  quantData <span class="ot">=</span> <span class="fu">read_tsv</span>(filePath, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(MS2ScanNumber, <span class="st">`</span><span class="at">126Intensity</span><span class="st">`</span><span class="sc">:</span><span class="st">`</span><span class="at">134NIntensity</span><span class="st">`</span>)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(quantData) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'scan'</span>,<span class="st">'tmt126C'</span>,<span class="st">'tmt127N'</span>,<span class="st">'tmt127C'</span>,<span class="st">'tmt128N'</span>,<span class="st">'tmt128C'</span>,<span class="st">'tmt129N'</span>,</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>                          <span class="st">'tmt129C'</span>,<span class="st">'tmt130N'</span>,<span class="st">'tmt130C'</span>,<span class="st">'tmt131N'</span>,<span class="st">'tmt131C'</span>,<span class="st">'tmt132N'</span>,<span class="st">'tmt132C'</span>,<span class="st">'tmt133N'</span>,<span class="st">'tmt133C'</span>,<span class="st">'tmt134N'</span>)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  <span class="do">##</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  fraction <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">'.*_HpH_(.*)</span><span class="sc">\\</span><span class="st">.raw_Matrix</span><span class="sc">\\</span><span class="st">.txt'</span>, <span class="st">'</span><span class="sc">\\</span><span class="st">1'</span>, filePath)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  dataset <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">'.*ch20201223_A673ShEwsFli1DoxTimecourse(.*)_TMT16_HpH_[0-9]+</span><span class="sc">\\</span><span class="st">.raw_Matrix</span><span class="sc">\\</span><span class="st">.txt'</span>, <span class="st">'</span><span class="sc">\\</span><span class="st">1'</span>, filePath)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  quantData<span class="sc">$</span>fraction <span class="ot">=</span> fraction</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  quantData<span class="sc">$</span>dataset <span class="ot">=</span> <span class="fu">tolower</span>(dataset)</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="do">##</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(quantData)</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="do">##process the quant data</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>quantFiles <span class="ot">=</span> <span class="fu">list.files</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenTimecourseProtein/quants/'</span>, <span class="at">sep =</span> <span class="st">''</span>),</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pattern =</span> <span class="st">'_Matrix.txt'</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>quantDataSet <span class="ot">=</span> <span class="fu">lapply</span>(quantFiles, combineQuantFiles)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
New names:
* `` -&gt; `...140`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>allQuantData <span class="ot">=</span> <span class="fu">do.call</span>(<span class="st">'rbind'</span>, quantDataSet)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="do">##combine all of the data</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>psmQuant <span class="ot">=</span> psmData <span class="sc">%&gt;%</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(allQuantData)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = c("dataset", "scan", "fraction")</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">##filter the psm data</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>psmQuant <span class="ot">=</span> psmQuant <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>tmt127N, <span class="sc">-</span>tmt127C, <span class="sc">-</span>tmt132N, <span class="sc">-</span>tmt132C, <span class="sc">-</span>tmt133C, <span class="sc">-</span>tmt134N) <span class="sc">%&gt;%</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(tmt126C)) <span class="sc">%&gt;%</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sampleSignal =</span> <span class="fu">sum</span>(tmt128N<span class="sc">:</span>tmt131C, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(sampleSignal <span class="sc">&gt;=</span> <span class="dv">800</span>)</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="do">##roll into proteins</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>pro <span class="ot">=</span> psmQuant <span class="sc">%&gt;%</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>sampleSignal) <span class="sc">%&gt;%</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">psmCount =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(sequence, accession, symbol, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="fu">across</span>(tmt126C<span class="sc">:</span>tmt133N, mean, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">psmCount =</span> <span class="fu">sum</span>(psmCount, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pepCount =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset, accession, symbol) <span class="sc">%&gt;%</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="fu">across</span>(tmt126C<span class="sc">:</span>tmt133N, median, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">psmCount =</span> <span class="fu">sum</span>(psmCount, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">pepCount =</span> <span class="fu">sum</span>(pepCount, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'sequence', 'accession', 'symbol'. You can
override using the `.groups` argument.
`summarise()` has grouped output by 'dataset', 'accession'. You can override
using the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="do">##save the data</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pro[pro<span class="sc">$</span>dataset <span class="sc">==</span> <span class="st">'seta'</span>,], <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenTimecourseProtein/dataset_rawProteinDatasetA.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pro[pro<span class="sc">$</span>dataset <span class="sc">==</span> <span class="st">'setb'</span>,], <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenTimecourseProtein/dataset_rawProteinDatasetB.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pro[pro<span class="sc">$</span>dataset <span class="sc">==</span> <span class="st">'setc'</span>,], <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenTimecourseProtein/dataset_rawProteinDatasetC.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Perform the IRS normalization.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>proteinSet1 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenTimecourseProtein/dataset_rawProteinDatasetA.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pool1a =</span> tmt126C,</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">day0a =</span> tmt128N,</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">day7a =</span> tmt128C,</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">day9a =</span> tmt129N,</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">day10a =</span> tmt129C,</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">day11a =</span> tmt130N,</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">day14a =</span> tmt130C,</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">day17a =</span> tmt131N,</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">day22a =</span> tmt131C,</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">pool2a =</span> tmt133N,</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">psma =</span> psmCount,</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">pepa =</span> pepCount)</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>proteinSet2 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenTimecourseProtein/dataset_rawProteinDatasetB.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pool1b =</span> tmt126C,</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">day0b =</span> tmt128N,</span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>                <span class="at">day7b =</span> tmt128C,</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>                <span class="at">day9b =</span> tmt129N,</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>                <span class="at">day10b =</span> tmt129C,</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>                <span class="at">day11b =</span> tmt130N,</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>                <span class="at">day14b =</span> tmt130C,</span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>                <span class="at">day17b =</span> tmt131N,</span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>                <span class="at">day22b =</span> tmt131C,</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>                <span class="at">pool2b =</span> tmt133N,</span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>                <span class="at">psmb =</span> psmCount,</span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>                <span class="at">pepb =</span> pepCount)</span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>proteinSet3 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenTimecourseProtein/dataset_rawProteinDatasetC.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pool1c =</span> tmt126C,</span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>                <span class="at">day0c =</span> tmt128N,</span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>                <span class="at">day7c =</span> tmt128C,</span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>                <span class="at">day9c =</span> tmt129N,</span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>                <span class="at">day10c =</span> tmt129C,</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>                <span class="at">day11c =</span> tmt130N,</span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>                <span class="at">day14c =</span> tmt130C,</span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>                <span class="at">day17c =</span> tmt131N,</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>                <span class="at">day22c =</span> tmt131C,</span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>                <span class="at">pool2c =</span> tmt133N,</span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>                <span class="at">psmc =</span> psmCount,</span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>                <span class="at">pepc =</span> pepCount)</span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>proteinSet <span class="ot">=</span> proteinSet1 <span class="sc">%&gt;%</span></span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(proteinSet2) <span class="sc">%&gt;%</span></span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(proteinSet3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = c("accession", "symbol")
Joining, by = c("accession", "symbol")</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">##raw data check</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>targetSamples <span class="ot">=</span> proteinSet[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="fu">which</span>(<span class="fu">grepl</span>(<span class="st">'day'</span>, <span class="fu">colnames</span>(proteinSet))))]</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>pooledSamples <span class="ot">=</span> proteinSet[,<span class="fu">c</span>(<span class="fu">which</span>(<span class="fu">grepl</span>(<span class="st">'pool'</span>, <span class="fu">colnames</span>(proteinSet))))]</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>psmSamples <span class="ot">=</span> proteinSet[,<span class="fu">c</span>(<span class="fu">which</span>(<span class="fu">grepl</span>(<span class="st">'psm'</span>, <span class="fu">colnames</span>(proteinSet))))]</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>rawData <span class="ot">=</span> <span class="fu">cbind</span>(targetSamples, pooledSamples, psmSamples)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>rawData[rawData <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>rawData <span class="ot">=</span> <span class="fu">subset</span>(rawData, <span class="fu">rowSums</span>(<span class="fu">is.na</span>(rawData[,<span class="dv">3</span><span class="sc">:</span><span class="dv">10</span>])) <span class="sc">&lt;</span> <span class="dv">5</span> <span class="sc">&amp;</span> <span class="fu">rowSums</span>(<span class="fu">is.na</span>(rawData[,<span class="dv">11</span><span class="sc">:</span><span class="dv">18</span>])) <span class="sc">&lt;</span> <span class="dv">5</span> <span class="sc">&amp;</span> <span class="fu">rowSums</span>(<span class="fu">is.na</span>(rawData[,<span class="dv">19</span><span class="sc">:</span><span class="dv">26</span>])) <span class="sc">&lt;</span> <span class="dv">5</span>)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">log2</span>(rawData[,<span class="dv">3</span><span class="sc">:</span><span class="dv">26</span>]), <span class="at">col =</span> <span class="fu">rep</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'red'</span>, <span class="st">'green'</span>, <span class="st">'blue'</span>), <span class="at">each =</span> <span class="dv">8</span>), <span class="dv">3</span>), </span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">notch =</span> <span class="cn">TRUE</span>, <span class="at">main =</span> <span class="st">'Raw data: setA (red), setB (green), setC (blue)'</span>,</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="st">'TMT Samples'</span>, <span class="at">ylab =</span> <span class="st">'log2 of Intensity'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="sorensenTimecourseProtein_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDensities</span>(<span class="fu">log2</span>(rawData[,<span class="dv">3</span><span class="sc">:</span><span class="dv">26</span>]), <span class="at">col =</span> <span class="fu">rep</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'red'</span>, <span class="st">'green'</span>, <span class="st">'blue'</span>), <span class="at">each =</span> <span class="dv">8</span>), <span class="dv">3</span>), </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">main =</span> <span class="st">'Raw data'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="sorensenTimecourseProtein_files/figure-html/unnamed-chunk-4-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">format</span>(<span class="fu">round</span>(<span class="fu">colSums</span>(rawData[,<span class="dv">3</span><span class="sc">:</span><span class="dv">26</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">digits =</span> <span class="dv">0</span>), <span class="at">big.mark =</span> <span class="st">","</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          day0a           day7a           day9a          day10a          day11a 
"2,853,291,059" "3,913,879,694" "4,078,081,420" "3,911,458,604" "3,745,926,025" 
         day14a          day17a          day22a           day0b           day7b 
"4,310,315,919" "3,345,905,539" "3,038,166,684" "2,554,868,913" "3,792,748,748" 
          day9b          day10b          day11b          day14b          day17b 
"3,525,406,123" "4,025,521,297" "3,694,785,048" "3,980,939,597" "3,059,323,130" 
         day22b           day0c           day7c           day9c          day10c 
"3,043,709,797" "2,684,858,168" "3,756,976,389" "3,387,812,695" "4,113,817,256" 
         day11c          day14c          day17c          day22c 
"3,930,597,875" "4,117,644,534" "3,064,272,437" "2,985,682,568" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="do">##sample loading normalization</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>exp1Raw <span class="ot">=</span> rawData[<span class="fu">c</span>(<span class="dv">3</span><span class="sc">:</span><span class="dv">10</span>)]</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>exp2Raw <span class="ot">=</span> rawData[<span class="fu">c</span>(<span class="dv">11</span><span class="sc">:</span><span class="dv">18</span>)]</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>exp3Raw <span class="ot">=</span> rawData[<span class="fu">c</span>(<span class="dv">19</span><span class="sc">:</span><span class="dv">26</span>)]</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>target <span class="ot">=</span> <span class="fu">mean</span>(<span class="fu">c</span>(<span class="fu">colSums</span>(exp1Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="fu">colSums</span>(exp2Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="fu">colSums</span>(exp3Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>normFacs <span class="ot">=</span> target <span class="sc">/</span> <span class="fu">colSums</span>(exp1Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>exp1Sl <span class="ot">=</span> <span class="fu">sweep</span>(exp1Raw, <span class="dv">2</span>, normFacs, <span class="at">FUN =</span> <span class="st">"*"</span>)</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>normFacs <span class="ot">=</span> target <span class="sc">/</span> <span class="fu">colSums</span>(exp2Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>exp2Sl <span class="ot">=</span> <span class="fu">sweep</span>(exp2Raw, <span class="dv">2</span>, normFacs, <span class="at">FUN =</span> <span class="st">"*"</span>)</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>normFacs <span class="ot">=</span> target <span class="sc">/</span> <span class="fu">colSums</span>(exp3Raw, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>exp3Sl <span class="ot">=</span> <span class="fu">sweep</span>(exp3Raw, <span class="dv">2</span>, normFacs, <span class="at">FUN =</span> <span class="st">"*"</span>)</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>dataSl <span class="ot">=</span> <span class="fu">cbind</span>(exp1Sl, exp2Sl, exp3Sl)</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">log2</span>(dataSl), <span class="at">col =</span> <span class="fu">rep</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'red'</span>, <span class="st">'green'</span>, <span class="st">'blue'</span>), <span class="at">each =</span> <span class="dv">8</span>), <span class="dv">3</span>), </span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">notch =</span> <span class="cn">TRUE</span>, <span class="at">main =</span> <span class="st">"Sample Loading (SL) normalized data: </span><span class="sc">\n</span><span class="st">setA (red), setB (green), setC (blue)"</span>,</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="st">'TMT Sample'</span>, <span class="at">ylab =</span> <span class="st">'log2 of Intensity'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="sorensenTimecourseProtein_files/figure-html/unnamed-chunk-4-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDensities</span>(<span class="fu">log2</span>(dataSl), <span class="at">col =</span> <span class="fu">rep</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'red'</span>, <span class="st">'green'</span>, <span class="st">'blue'</span>), <span class="at">each =</span> <span class="dv">8</span>), <span class="dv">3</span>), <span class="at">main =</span> <span class="st">"SL normalization"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="sorensenTimecourseProtein_files/figure-html/unnamed-chunk-4-4.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="do">##IRS normalization</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>irs <span class="ot">=</span> <span class="fu">tibble</span>(<span class="fu">rowSums</span>(rawData[,<span class="dv">27</span><span class="sc">:</span><span class="dv">28</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="fu">rowSums</span>(rawData[,<span class="dv">29</span><span class="sc">:</span><span class="dv">30</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="fu">rowSums</span>(rawData[,<span class="dv">31</span><span class="sc">:</span><span class="dv">32</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(irs) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'sum1'</span>,<span class="st">'sum2'</span>,<span class="st">'sum3'</span>)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>irs<span class="sc">$</span>average <span class="ot">=</span> <span class="fu">apply</span>(irs, <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="fu">exp</span>(<span class="fu">mean</span>(<span class="fu">log</span>(x),<span class="at">na.rm =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>irs<span class="sc">$</span>fac1 <span class="ot">=</span> irs<span class="sc">$</span>average <span class="sc">/</span> irs<span class="sc">$</span>sum1</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>irs<span class="sc">$</span>fac2 <span class="ot">=</span> irs<span class="sc">$</span>average <span class="sc">/</span> irs<span class="sc">$</span>sum2</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>irs<span class="sc">$</span>fac3 <span class="ot">=</span> irs<span class="sc">$</span>average <span class="sc">/</span> irs<span class="sc">$</span>sum3</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>dataIrs <span class="ot">=</span> exp1Sl <span class="sc">*</span> irs<span class="sc">$</span>fac1</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>dataIrs <span class="ot">=</span> <span class="fu">cbind</span>(dataIrs, exp2Sl <span class="sc">*</span> irs<span class="sc">$</span>fac2)</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>dataIrs <span class="ot">=</span> <span class="fu">cbind</span>(dataIrs, exp3Sl <span class="sc">*</span> irs<span class="sc">$</span>fac3)</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">log2</span>(dataIrs), <span class="at">col =</span> <span class="fu">rep</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'red'</span>, <span class="st">'green'</span>, <span class="st">'blue'</span>), <span class="at">each =</span> <span class="dv">8</span>), <span class="dv">3</span>), </span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">main =</span> <span class="st">"Internal Reference Scaling (IRS) normalized data: </span><span class="sc">\n</span><span class="st">setA (red), setB (green), setC (blue)"</span>,</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="st">'TMT Sample'</span>, <span class="at">ylab =</span> <span class="st">'log2 of Intensity'</span>, <span class="at">notch =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 1 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 2 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 3 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 4 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 5 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 6 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 7 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 8 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 9 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 10 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 11 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 12 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 13 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 14 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 15 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 16 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 17 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 18 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 19 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 20 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 21 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 22 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 23 is not drawn</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z$group
== : Outlier (-Inf) in boxplot 24 is not drawn</code></pre>
</div>
<div class="cell-output-display">
<p><img src="sorensenTimecourseProtein_files/figure-html/unnamed-chunk-4-5.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDensities</span>(<span class="fu">log2</span>(dataIrs), <span class="at">col =</span> <span class="fu">rep</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'red'</span>, <span class="st">'green'</span>, <span class="st">'blue'</span>), <span class="at">each =</span> <span class="dv">8</span>), <span class="dv">3</span>), <span class="at">main =</span> <span class="st">"IRS data"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="sorensenTimecourseProtein_files/figure-html/unnamed-chunk-4-6.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">format</span>(<span class="fu">round</span>(<span class="fu">colSums</span>(dataIrs, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">digits =</span> <span class="dv">0</span>), <span class="at">big.mark =</span> <span class="st">","</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          day0a           day7a           day9a          day10a          day11a 
"3,187,537,699" "3,157,426,181" "3,155,915,395" "3,157,857,771" "3,156,403,317" 
         day14a          day17a          day22a           day0b           day7b 
"3,167,287,600" "3,179,849,397" "3,181,003,294" "3,430,261,772" "3,448,253,015" 
          day9b          day10b          day11b          day14b          day17b 
"3,441,634,914" "3,442,722,692" "3,436,712,692" "3,430,313,458" "3,420,130,070" 
         day22b           day0c           day7c           day9c          day10c 
"3,423,364,358" "3,506,479,020" "3,509,603,458" "3,511,761,343" "3,515,571,817" 
         day11c          day14c          day17c          day22c 
"3,516,885,032" "3,508,980,019" "3,508,105,245" "3,503,484,087" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="do">##post-normalization clustering</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>colVec <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">'red'</span>,<span class="dv">8</span>),<span class="fu">rep</span>(<span class="st">'blue'</span>,<span class="dv">8</span>),<span class="fu">rep</span>(<span class="st">'green'</span>,<span class="dv">8</span>))</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">log2</span>(dataIrs), <span class="at">col =</span> colVec, <span class="at">main =</span> <span class="st">"IRS clusters group by condition"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="sorensenTimecourseProtein_files/figure-html/unnamed-chunk-4-7.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Recombine the data with the accessions and save the final file.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>finalIrsData <span class="ot">=</span> <span class="fu">as_tibble</span>(<span class="fu">cbind</span>(rawData[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>,<span class="dv">33</span><span class="sc">:</span><span class="dv">35</span>)], dataIrs))</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="do">##save data</span></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(finalIrsData, <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenTimecourseProtein/dataset_proteinSetABCIrsNormalized.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Perform a differential expression analysis using DEqMS between specific timepoints.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>dataIrsRaw <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">readRDS</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenTimecourseProtein/dataset_proteinSetABCIrsNormalized.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>)))</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(dataIrsRaw) <span class="ot">=</span> dataIrsRaw<span class="sc">$</span>accession</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>dataIrs <span class="ot">=</span> dataIrsRaw[,<span class="fu">c</span>(<span class="fu">which</span>(<span class="fu">grepl</span>(<span class="st">'day0'</span>, <span class="fu">colnames</span>(dataIrsRaw)) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">'day7'</span>, <span class="fu">colnames</span>(dataIrsRaw))))] <span class="co">#I only keep the two time points I want to compare here</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>dataIrsLog <span class="ot">=</span> <span class="fu">log2</span>(dataIrs)</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>dataIrsLog <span class="ot">=</span> <span class="fu">na.omit</span>(dataIrsLog)</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a><span class="do">##make the design table and fit the model</span></span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>cond <span class="ot">=</span> <span class="fu">as.factor</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'day0'</span>,<span class="st">'day7'</span>),<span class="dv">3</span>))</span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a>design <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span><span class="sc">+</span>cond) <span class="co"># 0 means no intercept for the linear model</span></span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(design) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">'cond'</span>,<span class="st">''</span>,<span class="fu">colnames</span>(design))</span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a>contrast <span class="ot">=</span>  <span class="fu">makeContrasts</span>(<span class="at">contrasts =</span> <span class="st">'day0-day7'</span>, <span class="at">levels =</span> design)</span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a>fit1 <span class="ot">=</span> <span class="fu">lmFit</span>(dataIrsLog, design)</span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a>fit2 <span class="ot">=</span> <span class="fu">contrasts.fit</span>(fit1,<span class="at">contrasts =</span> contrast)</span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a>fit3 <span class="ot">=</span> <span class="fu">eBayes</span>(fit2)</span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a><span class="do">##DEqMS analysis</span></span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a>count_columns <span class="ot">=</span> <span class="fu">seq</span>(<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">1</span>)</span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a>psm.count.table <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">count =</span> <span class="fu">rowMins</span>(<span class="fu">as.matrix</span>(dataIrsRaw[,count_columns])), <span class="at">row.names =</span>  dataIrsRaw<span class="sc">$</span>accession)</span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a>fit3<span class="sc">$</span>count <span class="ot">=</span> psm.count.table[<span class="fu">rownames</span>(fit3<span class="sc">$</span>coefficients),<span class="st">'count'</span>]</span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a>fit4 <span class="ot">=</span> <span class="fu">spectraCounteBayes</span>(fit3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Outputs of spectraCounteBayes:</p>
<ul>
<li>object is augmented form of fit object from eBayes in Limma, with the additions being:</li>
<li>sca.t - Spectra Count Adjusted posterior t-value</li>
<li>sca.p - Spectra Count Adjusted posterior p-value</li>
<li>sca.dfprior - DEqMS estimated prior degrees of freedom</li>
<li>sca.priorvar- DEqMS estimated prior variance</li>
<li>sca.postvar - DEqMS estimated posterior variance</li>
<li>model - fitted model</li>
</ul>
<p>Output the results.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="do">##extract the results</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>deqmsResults <span class="ot">=</span> <span class="fu">tibble</span>(<span class="fu">outputResult</span>(fit4, <span class="at">coef_col =</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">accession =</span> gene) <span class="sc">%&gt;%</span></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(dataIrsRaw[,<span class="fu">c</span>(<span class="st">'accession'</span>,<span class="st">'symbol'</span>)]) <span class="sc">%&gt;%</span></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(symbol, logFC) <span class="sc">%&gt;%</span></span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">sorensenTimecourseProtein =</span> logFC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "accession"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="do">##write the data</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(deqmsResults, <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenTimecourseProtein/dataset_sorensenTimecourseProtein.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(deqmsResults, </span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>            <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenTimecourseProtein/dataset_sorensenTimecourseProtein.csv'</span>, <span class="at">sep =</span> <span class="st">''</span>),</span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">','</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Finish up.</p>
<section id="session-info" class="level3">
<h3 class="anchored" data-anchor-id="session-info">Session info</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.1.3 (2022-03-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_Canada.1252  LC_CTYPE=English_Canada.1252   
[3] LC_MONETARY=English_Canada.1252 LC_NUMERIC=C                   
[5] LC_TIME=English_Canada.1252    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggrepel_0.9.2               DEqMS_1.12.1               
 [3] DESeq2_1.34.0               SummarizedExperiment_1.24.0
 [5] Biobase_2.54.0              MatrixGenerics_1.6.0       
 [7] matrixStats_0.62.0          GenomicRanges_1.46.1       
 [9] GenomeInfoDb_1.30.1         IRanges_2.28.0             
[11] S4Vectors_0.32.4            BiocGenerics_0.40.0        
[13] RColorBrewer_1.1-3          limma_3.50.3               
[15] forcats_0.5.2               stringr_1.4.1              
[17] dplyr_1.0.10                purrr_0.3.5                
[19] readr_2.1.3                 tidyr_1.2.1                
[21] tibble_3.1.8                ggplot2_3.4.0              
[23] tidyverse_1.3.2            

loaded via a namespace (and not attached):
 [1] bitops_1.0-7           fs_1.5.2               bit64_4.0.5           
 [4] lubridate_1.9.0        httr_1.4.4             tools_4.1.3           
 [7] backports_1.4.1        utf8_1.2.2             R6_2.5.1              
[10] DBI_1.1.3              colorspace_2.0-3       withr_2.5.0           
[13] tidyselect_1.2.0       bit_4.0.5              compiler_4.1.3        
[16] cli_3.4.1              rvest_1.0.3            xml2_1.3.3            
[19] DelayedArray_0.20.0    scales_1.2.1           genefilter_1.76.0     
[22] digest_0.6.30          rmarkdown_2.18         XVector_0.34.0        
[25] pkgconfig_2.0.3        htmltools_0.5.3        dbplyr_2.2.1          
[28] fastmap_1.1.0          htmlwidgets_1.5.4      rlang_1.0.6           
[31] readxl_1.4.1           RSQLite_2.2.18         rstudioapi_0.14       
[34] generics_0.1.3         jsonlite_1.8.3         vroom_1.6.0           
[37] BiocParallel_1.28.3    googlesheets4_1.0.1    RCurl_1.98-1.9        
[40] magrittr_2.0.3         GenomeInfoDbData_1.2.7 Matrix_1.5-3          
[43] Rcpp_1.0.9             munsell_0.5.0          fansi_1.0.3           
[46] lifecycle_1.0.3        stringi_1.7.8          yaml_2.3.6            
[49] zlibbioc_1.40.0        blob_1.2.3             grid_4.1.3            
[52] parallel_4.1.3         crayon_1.5.2           lattice_0.20-45       
[55] splines_4.1.3          Biostrings_2.62.0      haven_2.5.1           
[58] annotate_1.72.0        KEGGREST_1.34.0        hms_1.1.2             
[61] locfit_1.5-9.6         knitr_1.40             pillar_1.8.1          
[64] geneplotter_1.72.0     reprex_2.0.2           XML_3.99-0.12         
[67] glue_1.6.2             evaluate_0.18          modelr_0.1.10         
[70] png_0.1-7              vctrs_0.5.1            tzdb_0.3.0            
[73] cellranger_1.1.0       gtable_0.3.1           assertthat_0.2.1      
[76] cachem_1.0.6           xfun_0.34              xtable_1.8-4          
[79] broom_1.0.1            survival_3.4-0         googledrive_2.0.0     
[82] gargle_1.2.1           memoise_2.0.1          AnnotationDbi_1.56.2  
[85] timechange_0.1.1       ellipsis_0.3.2        </code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    setTimeout(function() {
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>