<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.260">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Christopher Hughes">

<title>sorensenTimecourseRna</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<script src="sorensenTimecourseRna_files/libs/clipboard/clipboard.min.js"></script>
<script src="sorensenTimecourseRna_files/libs/quarto-html/quarto.js"></script>
<script src="sorensenTimecourseRna_files/libs/quarto-html/popper.min.js"></script>
<script src="sorensenTimecourseRna_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="sorensenTimecourseRna_files/libs/quarto-html/anchor.min.js"></script>
<link href="sorensenTimecourseRna_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link id="quarto-text-highlighting-styles" href="sorensenTimecourseRna_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet">
<script src="sorensenTimecourseRna_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="sorensenTimecourseRna_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="sorensenTimecourseRna_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">sorensenTimecourseRna</h1>
</div>





<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Christopher Hughes </p>
          </div>
  </div>
    
    
  </div>
  

</header>

<section id="setting-up-the-environment" class="level2">
<h2 class="anchored" data-anchor-id="setting-up-the-environment">Setting up the environment</h2>
<p>These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#packages</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'tidyverse'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>-- Attaching packages --------------------------------------- tidyverse 1.3.2 --
v ggplot2 3.4.0      v purrr   0.3.5 
v tibble  3.1.8      v dplyr   1.0.10
v tidyr   1.2.1      v stringr 1.4.1 
v readr   2.1.3      v forcats 0.5.2 
-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'RColorBrewer'</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'ggplot2'</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'tximport'</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'GenomicFeatures'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics

Attaching package: 'BiocGenerics'

The following objects are masked from 'package:dplyr':

    combine, intersect, setdiff, union

The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs

The following objects are masked from 'package:base':

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min

Loading required package: S4Vectors
Loading required package: stats4

Attaching package: 'S4Vectors'

The following objects are masked from 'package:dplyr':

    first, rename

The following object is masked from 'package:tidyr':

    expand

The following objects are masked from 'package:base':

    expand.grid, I, unname

Loading required package: IRanges

Attaching package: 'IRanges'

The following objects are masked from 'package:dplyr':

    collapse, desc, slice

The following object is masked from 'package:purrr':

    reduce

The following object is masked from 'package:grDevices':

    windows

Loading required package: GenomeInfoDb
Loading required package: GenomicRanges
Loading required package: AnnotationDbi
Loading required package: Biobase
Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.


Attaching package: 'AnnotationDbi'

The following object is masked from 'package:dplyr':

    select</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'DESeq2'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment
Loading required package: MatrixGenerics
Loading required package: matrixStats

Attaching package: 'matrixStats'

The following objects are masked from 'package:Biobase':

    anyMissing, rowMedians

The following object is masked from 'package:dplyr':

    count


Attaching package: 'MatrixGenerics'

The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars

The following object is masked from 'package:Biobase':

    rowMedians</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'org.Hs.eg.db'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'ggrepel'</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#directories</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>generalDatasets <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>baseWorkspace <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>baseRepository <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-processing" class="level2">
<h2 class="anchored" data-anchor-id="data-processing">Data processing</h2>
<p>Read in the sample annotation details and make the txdb that we can use later on for annotation gene identifiers.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co">#read a table containing the sample information for our RNAseq runs</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="fu">paste</span>(baseRepository, <span class="st">'/sequencing20210421_a673TimecourseRnaSeqOutput/sampleInfo.txt'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>samples</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 24 x 6
   sampleName cell          treatment barcode experiment batch
   &lt;chr&gt;      &lt;chr&gt;         &lt;chr&gt;     &lt;chr&gt;   &lt;chr&gt;      &lt;chr&gt;
 1 day0       a673shEwsFli1 none      ATCACG  PX1955     setA 
 2 day0       a673shEwsFli1 none      ATCACG  PX1956     setB 
 3 day0       a673shEwsFli1 none      ATCACG  PX1957     setC 
 4 day7       a673shEwsFli1 none      CGATGT  PX1955     setA 
 5 day7       a673shEwsFli1 none      CGATGT  PX1956     setB 
 6 day7       a673shEwsFli1 none      CGATGT  PX1957     setC 
 7 day9       a673shEwsFli1 none      TTAGGC  PX1955     setA 
 8 day9       a673shEwsFli1 none      TTAGGC  PX1956     setB 
 9 day9       a673shEwsFli1 none      TTAGGC  PX1957     setC 
10 day10      a673shEwsFli1 none      TGACCA  PX1955     setA 
# ... with 14 more rows</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co">#use the sample info to build a file list</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>files <span class="ot">=</span> <span class="fu">file.path</span>(baseRepository, <span class="st">'sequencing20210421_a673TimecourseRnaSeqOutput'</span>, <span class="st">'quants'</span>, <span class="fu">paste</span>(samples<span class="sc">$</span>barcode, <span class="st">'_'</span>, samples<span class="sc">$</span>batch, <span class="at">sep =</span> <span class="st">''</span>), <span class="st">'quant.sf'</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">file.exists</span>(files))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<p>For Salmon analysis, I am generally following the documentation found <a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">here</a> and <a href="http://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html">here</a>. When we import our data, we want a table that allows us to link gene and transcript identifiers. For this we use the GTF associated with our database files that we used during the alignment process.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co">#build the txdb from the gtf file</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>myTxdb <span class="ot">=</span> <span class="fu">makeTxDbFromGFF</span>(<span class="st">'D:/databases/projectEwsDlg2/baseGenomeFiles/genome.gtf'</span>)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>k <span class="ot">=</span> <span class="fu">keys</span>(myTxdb, <span class="at">keytype =</span> <span class="st">'TXNAME'</span>)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>tx2gene <span class="ot">=</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(myTxdb, k, <span class="st">'GENEID'</span>, <span class="st">'TXNAME'</span>)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tx2gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             TXNAME            GENEID
1 ENST00000456328.2 ENSG00000223972.5
2 ENST00000450305.2 ENSG00000223972.5
3 ENST00000473358.1 ENSG00000243485.5
4 ENST00000469289.1 ENSG00000243485.5
5 ENST00000607096.1 ENSG00000284332.1
6 ENST00000606857.1 ENSG00000268020.3</code></pre>
</div>
</div>
<p>Read in the Salmon data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="co">#read the salmon data</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>txi <span class="ot">=</span> <span class="fu">tximport</span>(files, </span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">type =</span> <span class="st">'salmon'</span>, </span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">tx2gene =</span> tx2gene)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(txi)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "abundance"           "counts"              "length"             
[4] "countsFromAbundance"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(txi<span class="sc">$</span>counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       [,1]     [,2]     [,3]     [,4]     [,5]     [,6]
ENSG00000000003.15 2652.539 2119.085 2332.999 3138.041 2874.567 3252.989
ENSG00000000005.6     0.000    2.000    0.000    2.000    2.000    3.000
ENSG00000000419.14 2851.309 2287.998 2415.567 2178.854 2003.019 2277.191
ENSG00000000457.14  309.999  255.001  226.000  458.000  464.001  509.000
ENSG00000000460.17 1432.964 1173.974 1242.000  706.000  729.452  794.000
ENSG00000000938.13    2.000    3.000    0.000    0.000    2.000    1.000
                       [,7]     [,8]     [,9]    [,10]    [,11]    [,12]
ENSG00000000003.15 2431.000 2822.353 2778.054 1663.000 1838.000 2087.001
ENSG00000000005.6     0.000    0.000    1.000    2.000    0.000    0.000
ENSG00000000419.14 1921.000 2209.000 2096.001 1550.795 1742.809 1753.997
ENSG00000000457.14  359.000  414.999  333.001  263.000  332.082  334.000
ENSG00000000460.17  354.001  466.000  383.001  523.000  576.000  575.001
ENSG00000000938.13    3.000    4.000    1.000    1.000    1.000    0.000
                      [,13]    [,14]    [,15]    [,16]    [,17]    [,18]
ENSG00000000003.15 2825.999 2612.000 2682.001 2657.919 3228.245 2454.007
ENSG00000000005.6     0.000    0.000    0.000    0.000    0.000    0.000
ENSG00000000419.14 2452.167 2281.000 2263.001 2681.828 3253.333 2456.237
ENSG00000000457.14  425.999  390.000  365.000  339.000  393.028  325.999
ENSG00000000460.17  554.001  528.117  545.002 1251.211 1571.005 1181.258
ENSG00000000938.13    1.000    2.000    1.000    2.000    1.000    0.000
                      [,19]    [,20]    [,21]    [,22]    [,23]    [,24]
ENSG00000000003.15 1528.000 1584.000 1482.000 2813.000 3622.502 2824.001
ENSG00000000005.6     1.000    0.000    0.000    2.000    4.000    1.000
ENSG00000000419.14 3318.012 3208.001 2882.079 2394.529 3196.449 2500.667
ENSG00000000457.14  555.000  539.000  490.000  294.000  387.000  261.999
ENSG00000000460.17  609.000  620.999  630.000 1121.001 1488.452 1115.001
ENSG00000000938.13    4.000    1.000    2.000    1.000    0.000    1.000</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="co">#perform the deseq analysis</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>ddsTxi <span class="ot">=</span> <span class="fu">DESeqDataSetFromTximport</span>(txi,</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> samples,</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> sampleName)</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">=</span> <span class="fu">DESeq</span>(ddsTxi)</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">=</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds)) <span class="sc">&gt;=</span> <span class="dv">10</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">=</span> dds[keep,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Extract the DESeq data and plot for the different comparisons of interest.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="co">#parse the deseq data - change the first two values to get different comparisons</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>daysToCompare <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'day0'</span>,<span class="st">'day9'</span>,<span class="st">'day10'</span>,<span class="st">'day11'</span>,<span class="st">'day14'</span>,<span class="st">'day17'</span>,<span class="st">'day22'</span>)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(daysToCompare)){</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  datasetFirst <span class="ot">=</span> daysToCompare[i]</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  datasetSecond <span class="ot">=</span> <span class="st">'day7'</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">'sampleName'</span>, datasetFirst, datasetSecond))</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>  ens.str <span class="ot">=</span> <span class="fu">substr</span>(<span class="fu">rownames</span>(res), <span class="dv">1</span>, <span class="dv">15</span>)</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  res<span class="sc">$</span>symbol <span class="ot">=</span> <span class="fu">mapIds</span>(org.Hs.eg.db,</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>                      <span class="at">keys=</span>ens.str,</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>                      <span class="at">column=</span><span class="st">"SYMBOL"</span>,</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>                      <span class="at">keytype=</span><span class="st">"ENSEMBL"</span>,</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>                      <span class="at">multiVals=</span><span class="st">"first"</span>)</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  resOrdered <span class="ot">=</span> res[<span class="fu">order</span>(res<span class="sc">$</span>pvalue),]</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>  <span class="co">#save the data</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>  outputData <span class="ot">=</span> <span class="fu">as.data.frame</span>(resOrdered) <span class="sc">%&gt;%</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(symbol)) <span class="sc">%&gt;%</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(symbol, log2FoldChange) <span class="sc">%&gt;%</span></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">sorensenTimecourseRna =</span> log2FoldChange) <span class="sc">%&gt;%</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(symbol) <span class="sc">%&gt;%</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">sorensenTimecourseRna =</span> <span class="fu">mean</span>(sorensenTimecourseRna, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(outputData, </span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>          <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenTimecourseRna/dataset_sorensenTimecourseRna_'</span>, datasetFirst, <span class="st">'-'</span>, datasetSecond, <span class="st">'.rds'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.csv</span>(outputData, </span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="fu">paste</span>(baseRepository, <span class="st">'/database20220929_ewsExpressionDatabase/sorensenTimecourseRna/dataset_sorensenTimecourseRna_'</span>, datasetFirst, <span class="st">'-'</span>, datasetSecond, <span class="st">'.csv'</span>, <span class="at">sep =</span> <span class="st">''</span>))</span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We are done here for now. Wrap up below.</p>
<section id="session-info" class="level3">
<h3 class="anchored" data-anchor-id="session-info">Session info</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.1.3 (2022-03-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_Canada.1252  LC_CTYPE=English_Canada.1252   
[3] LC_MONETARY=English_Canada.1252 LC_NUMERIC=C                   
[5] LC_TIME=English_Canada.1252    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggrepel_0.9.2               org.Hs.eg.db_3.14.0        
 [3] DESeq2_1.34.0               SummarizedExperiment_1.24.0
 [5] MatrixGenerics_1.6.0        matrixStats_0.62.0         
 [7] GenomicFeatures_1.46.5      AnnotationDbi_1.56.2       
 [9] Biobase_2.54.0              GenomicRanges_1.46.1       
[11] GenomeInfoDb_1.30.1         IRanges_2.28.0             
[13] S4Vectors_0.32.4            BiocGenerics_0.40.0        
[15] tximport_1.22.0             RColorBrewer_1.1-3         
[17] forcats_0.5.2               stringr_1.4.1              
[19] dplyr_1.0.10                purrr_0.3.5                
[21] readr_2.1.3                 tidyr_1.2.1                
[23] tibble_3.1.8                ggplot2_3.4.0              
[25] tidyverse_1.3.2            

loaded via a namespace (and not attached):
 [1] googledrive_2.0.0        colorspace_2.0-3         rjson_0.2.21            
 [4] ellipsis_0.3.2           XVector_0.34.0           fs_1.5.2                
 [7] rstudioapi_0.14          bit64_4.0.5              fansi_1.0.3             
[10] lubridate_1.9.0          xml2_1.3.3               splines_4.1.3           
[13] cachem_1.0.6             geneplotter_1.72.0       knitr_1.40              
[16] jsonlite_1.8.3           Rsamtools_2.10.0         annotate_1.72.0         
[19] broom_1.0.1              dbplyr_2.2.1             png_0.1-7               
[22] compiler_4.1.3           httr_1.4.4               backports_1.4.1         
[25] assertthat_0.2.1         Matrix_1.5-3             fastmap_1.1.0           
[28] gargle_1.2.1             cli_3.4.1                htmltools_0.5.3         
[31] prettyunits_1.1.1        tools_4.1.3              gtable_0.3.1            
[34] glue_1.6.2               GenomeInfoDbData_1.2.7   rappdirs_0.3.3          
[37] Rcpp_1.0.9               cellranger_1.1.0         vctrs_0.5.1             
[40] Biostrings_2.62.0        rtracklayer_1.54.0       xfun_0.34               
[43] rvest_1.0.3              timechange_0.1.1         lifecycle_1.0.3         
[46] restfulr_0.0.15          XML_3.99-0.12            googlesheets4_1.0.1     
[49] zlibbioc_1.40.0          scales_1.2.1             vroom_1.6.0             
[52] hms_1.1.2                parallel_4.1.3           yaml_2.3.6              
[55] curl_4.3.3               memoise_2.0.1            biomaRt_2.50.3          
[58] stringi_1.7.8            RSQLite_2.2.18           genefilter_1.76.0       
[61] BiocIO_1.4.0             filelock_1.0.2           BiocParallel_1.28.3     
[64] rlang_1.0.6              pkgconfig_2.0.3          bitops_1.0-7            
[67] evaluate_0.18            lattice_0.20-45          GenomicAlignments_1.30.0
[70] htmlwidgets_1.5.4        bit_4.0.5                tidyselect_1.2.0        
[73] magrittr_2.0.3           R6_2.5.1                 generics_0.1.3          
[76] DelayedArray_0.20.0      DBI_1.1.3                pillar_1.8.1            
[79] haven_2.5.1              withr_2.5.0              survival_3.4-0          
[82] KEGGREST_1.34.0          RCurl_1.98-1.9           modelr_0.1.10           
[85] crayon_1.5.2             utf8_1.2.2               BiocFileCache_2.2.1     
[88] tzdb_0.3.0               rmarkdown_2.18           progress_1.2.2          
[91] locfit_1.5-9.6           grid_4.1.3               readxl_1.4.1            
[94] blob_1.2.3               reprex_2.0.2             digest_0.6.30           
[97] xtable_1.8-4             munsell_0.5.0           </code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    setTimeout(function() {
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>