---
title: "Mapping tandem repeats"
output:
  html_notebook:
      code_folding: none
---

This document details processing of tandem repeat data output from [Tandem Repeats Finder](https://github.com/Benson-Genomics-Lab/TRF). 

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning = FALSE}
###########################################################
library('tidyverse')
library('rtracklayer')
library('GenomicRanges')
library('wiggleplotr')
```

<div style="margin-bottom:50px;"></div>

Set the base folders where we will do the work.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'
```

<div style="margin-bottom:50px;"></div>

## Data processing

### hg38 analysis

Read in and process the tandem repeats data.

```{r}
###########################################################
baseLocation = 'C:/Users/chughes/Documents/bccrc/databases/projectEwsDlg2/baseGenomeFiles/tandemRepeats'
repFiles = list.files(baseLocation, pattern = '*.dat', full.names = TRUE)

##
repColNames = c('start','end','period',
                'copyNumber','consensusSize','percentMatches',
                'percentIndels','score','A','C','G','T','entropy',
                'repeatedSeq','fullSeq')
#
allReps = tibble()
for (i in 1:length(repFiles)){
  inData = read_table(repFiles[i], skip = 15, col_names = repColNames) %>%
    dplyr::mutate(chromosome = sub('.*tandemRepeats\\/(.*)\\.fa\\.2\\.7\\.7\\.80\\.10\\.50\\.500\\.dat','\\1',repFiles[i])) %>%
    dplyr::mutate(strand = ifelse(grepl('^[AG][AG][AG][AG]$', repeatedSeq), '+', 
                                ifelse(grepl('^[TC][TC][TC][TC]$', repeatedSeq), '-', 'false'))) %>%
    dplyr::filter(strand != 'false') %>%
    dplyr::mutate(isMsat = ifelse(strand == '+' & G >= 40 & G <=60, 'yes',
                                  ifelse(strand == '-' & T >= 40 & T <= 60, 'yes', 'no'))) %>%
    dplyr::filter(isMsat == 'yes')
  #
  allReps = rbind(allReps, inData)
}

#
allReps2 = allReps %>%
  dplyr::mutate(chromosome = factor(chromosome, levels = c('chr1','chr2','chr3','chr4','chr5','chr6','chr7',
                                                       'chr8','chr9','chr10','chr11','chr12','chr13','chr14',
                                                       'chr15','chr16','chr17','chr18','chr19','chr20','chr21',
                                                       'chr22','chrX','chrY'))) %>%
  dplyr::arrange(chromosome, start) %>%
  dplyr::mutate(score = seq(1,10370,1)) %>%
  dplyr::mutate(msat_name = paste('msat',score, sep = ''))
##
saveRDS(allReps2, paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_msatRepeatsHg38NotAnnotated.rds',sep=''))
```

<div style="margin-bottom:50px;"></div>

Now annotate these repeats with respect to genes. First thing to do is get the gene regions out of the GTF.

```{r}
###########################################################
gtf = as_tibble(as.data.frame(rtracklayer::import('C:/Users/chughes/Documents/bccrc/databases/projectEwsDlg2/baseGenomeFiles/genome.gtf'))) %>%
  dplyr::filter(type == 'gene')

#
saveRDS(gtf, paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_genesOnlyFromGtf.rds',sep=''))
```

<div style="margin-bottom:50px;"></div>

Now we can use these gene regions to annotate the repeats data.

```{r}
###########################################################
gtfGenes = readRDS(paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_genesOnlyFromGtf.rds',sep=''))
msatRepeats = readRDS(paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_msatRepeatsHg38NotAnnotated.rds',sep=''))

##
gfGtf = makeGRangesFromDataFrame(gtfGenes)
gfMsat = makeGRangesFromDataFrame(msatRepeats)
overlapSet = GenomicRanges::findOverlaps(gfMsat, gfGtf, type = 'within')

##
msatAnnotation = tibble('qHits' = queryHits(overlapSet),
                        'sHits' = subjectHits(overlapSet),
                        'geneId' = gtfGenes[subjectHits(overlapSet), 'gene_name']) %>%
  group_by(qHits) %>%
  dplyr::mutate(multiGeneHits = paste(geneId, collapse = ',')) %>%
  dplyr::select(qHits, multiGeneHits) %>%
  dplyr::rename(genes = 'multiGeneHits') %>%
  unique()

##
msatRepeats$inGene = NA
msatRepeats[msatAnnotation$qHits,'inGene'] = msatAnnotation$genes
```

<div style="margin-bottom:50px;"></div>

This theoretically works, but maybe we want to use [bedtools closest](https://bedtools.readthedocs.io/en/latest/content/tools/closest.html) for this instead. In order to do that, we need to make bed format files for our two sets we want to compare.

```{r}
###########################################################
gtfGenes = readRDS(paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_genesOnlyFromGtf.rds',sep=''))
msatRepeats = readRDS(paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_msatRepeatsHg38NotAnnotated.rds',sep=''))

#
gtfBed = gtfGenes %>%
  dplyr::select(seqnames, start, end, gene_name, strand) %>%
  dplyr::mutate(seqnames = factor(seqnames, levels = c('chr1','chr2','chr3','chr4','chr5','chr6','chr7',
                                                       'chr8','chr9','chr10','chr11','chr12','chr13','chr14',
                                                       'chr15','chr16','chr17','chr18','chr19','chr20','chr21',
                                                       'chr22','chrX','chrY'))) %>%
  dplyr::arrange(seqnames, start) %>%
  dplyr::mutate(score = seq(1,60649,1)) %>%
  dplyr::select(seqnames, start, end, gene_name, score, strand) %>%
  dplyr::filter(!is.na(seqnames))
write.table(gtfBed, paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_gtfGenesOnly.bed',sep = ''),
            col.names = FALSE, row.names = FALSE, quote = FALSE, sep = '\t')


#
msatBed = msatRepeats %>%
  dplyr::select(chromosome, start, end, msat_name, score, strand)
write.table(msatBed, paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_msatRepeats.bed',sep = ''),
            col.names = FALSE, row.names = FALSE, quote = FALSE, sep = '\t')

```

<div style="margin-bottom:50px;"></div>

Combine the output from bedtools with the original msat data and with some RNA expression information.

```{r}
###########################################################
##
#read and combine the results
closestGene = read_tsv(paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_hg38MsatClosestGenesUnstranded.tsv',sep=''),
                       col_names = c('seqname_a','start_a','end_a','msat_name','index_a','strand_a',
                                     'seqname_b','start_b','end_b','gene','index_b','strand_b','distanceAtoB')) %>%
  left_join(msatRepeats)

##
#add rna expression data
rnaExp = read_csv(paste(baseRepository, '/sequencing20210421_a673TimecourseRnaSeqOutput/dataset_deseq2_day7-day0.csv',sep=''),
                  col_names = c('ensg','baseMean','log2FoldChange','lfcSE','stat','pvalue','padj','gene'))

closestGeneRna = closestGene %>%
  left_join(rnaExp)
```

<div style="margin-bottom:50px;"></div>

Add EwS specificity scores.

```{r}
###########################################################
##
#get the ccle data for our genes of interest
genesOfInterest = unique(closestGeneRna$gene)
ccle = readRDS(paste(generalDatasets, '/depmap21Q4/CCLE_expression_parsed.rds', sep = ''))
cells = ccle %>%
  dplyr::select(gene, log2Tpm, lineage_subtype) %>%
  dplyr::mutate(isEwing = ifelse(grepl('Ewing', lineage_subtype), 'yes', 'no')) %>%
  dplyr::select(-lineage_subtype) %>%
  dplyr::filter(gene %in% genesOfInterest)
##

ewsSpecScores = data.frame()
for (i in 1:length(genesOfInterest)){
  targetGene = cells %>%
    dplyr::filter(gene == genesOfInterest[i]) %>%
    group_by(isEwing) %>%
    dplyr::summarise(medRna = median(log2Tpm, na.rm = TRUE))
  if (nrow(targetGene) < 2){
    #message('no value for ', rnaProGenes[i])
    targetGeneScore = data.frame('gene' = genesOfInterest[i],
                               'medRna' = NA)
    ewsSpecScores = rbind(ewsSpecScores, targetGeneScore)
  } else {
      targetGeneScore = data.frame('gene' = genesOfInterest[i],
                               'medRna' = targetGene[targetGene$isEwing == 'no', 'medRna'] - targetGene[targetGene$isEwing == 'yes', 'medRna'])
      ewsSpecScores = rbind(ewsSpecScores, targetGeneScore)
  }
}
ewsCellScores = ewsSpecScores %>%
  dplyr::rename(cellScore = medRna)

##
#assign to our msat data
closestGeneRnaEws = closestGeneRna %>%
  left_join(ewsCellScores)

##
#save the data
saveRDS(closestGeneRnaEws, paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_msatRepeatsRnaEws.rds',sep=''))
write.table(closestGeneRnaEws, paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_msatRepeatsRnaEws.tsv',sep=''),
            col.names = TRUE, row.names = FALSE, quote = FALSE, sep = '\t')
```
























<div style="margin-bottom:50px;"></div>

### hg19 analysis

We can do the same analysis for hg19, as below.

Read in and process the tandem repeats data for the hg19 analysis.

```{r}
###########################################################
baseLocation = 'C:/Users/chughes/Documents/bccrc/databases/projectEwsDlg2/hg19/tandemRepeats'
repFiles = list.files(baseLocation, pattern = '*.dat', full.names = TRUE)

##
repColNames = c('start','end','period',
                'copyNumber','consensusSize','percentMatches',
                'percentIndels','score','A','C','G','T','entropy',
                'repeatedSeq','fullSeq')
#
allReps = tibble()
for (i in 1:length(repFiles)){
  inData = read_table(repFiles[i], skip = 15, col_names = repColNames) %>%
    dplyr::mutate(chromosome = sub('.*tandemRepeats\\/(.*)\\.fa\\.2\\.7\\.7\\.80\\.10\\.50\\.500\\.dat','\\1',repFiles[i])) %>%
    dplyr::mutate(strand = ifelse(grepl('^[AG][AG][AG][AG]$', repeatedSeq), '+', 
                                ifelse(grepl('^[TC][TC][TC][TC]$', repeatedSeq), '-', 'false'))) %>%
    dplyr::filter(strand != 'false') %>%
    dplyr::mutate(isMsat = ifelse(strand == '+' & G >= 40 & G <=60, 'yes',
                                  ifelse(strand == '-' & T >= 40 & T <= 60, 'yes', 'no'))) %>%
    dplyr::filter(isMsat == 'yes')
  #
  allReps = rbind(allReps, inData)
}

#
saveRDS(allReps, paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_msatRepeatsHg19NotAnnotated.rds',sep=''))
```

<div style="margin-bottom:50px;"></div>

Write a bed file for each of the msat sets.

```{r}
###########################################################
hg19 = readRDS(paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_msatRepeatsHg19NotAnnotated.rds',sep=''))
hg38 = readRDS(paste(baseRepository, '/sequencing20211129_grunewaldEwsAtlasWgs/dataset_msatRepeatsHg38NotAnnotated.rds',sep=''))

#
hg19Bed = hg19 %>%
  dplyr::select(chromosome, start, end, consensusSize, copyNumber) %>%
  dplyr::mutate(name = paste('mSat',seq(1,nrow(hg19),1),sep='')) 
write.table(hg19Bed, paste(baseRepository,'/sequencing20211129_grunewaldEwsAtlasWgs/dataset_hg19MsatRepeats.bed',sep=''),
            col.names = FALSE, row.names = FALSE, quote = FALSE, sep = '\t')
  
#
hg38Bed = hg38 %>%
  dplyr::select(chromosome, start, end, consensusSize, copyNumber) %>%
  dplyr::mutate(name = paste('mSat',seq(1,nrow(hg38),1),sep='')) 
write.table(hg38Bed, paste(baseRepository,'/sequencing20211129_grunewaldEwsAtlasWgs/dataset_hg38MsatRepeats.bed',sep=''),
            col.names = FALSE, row.names = FALSE, quote = FALSE, sep = '\t')
```

<div style="margin-bottom:50px;"></div>

Make a plot of the DLG2 gene from a GTF.

```{r}
###########################################################
gtf = as_tibble(as.data.frame(rtracklayer::import('C:/Users/chughes/Documents/bccrc/databases/projectEwsDlg2/baseGenomeFiles/genome.gtf')))
dlg2Gtf = gtf %>%
  dplyr::filter(grepl('DLG2$', gene_name))
  #filter(grepl('NM_001300983.1$', transcript_id))
  dplyr::rename(gene_name = gene_id, gene_id = seqnames) %>%
  mutate('seqnames' = 'chr11')
##
dlg2Meta = tibble('transcript_id' = 'NM_001300983.1',
                  'gene_id' = 'NC_000011.10',
                  'gene_name' = 'DLG2',
                  'strand' = '-')
##
dlg2Exons = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('exon', type)), split.field = 'transcript_id')
##
dlg2Cds = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('CDS', type)), split.field = 'transcript_id')
#

```




