<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Christopher Hughes">

<title>Sorensen RNA-seq data processing</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="processingProteomicsResults_files/libs/clipboard/clipboard.min.js"></script>
<script src="processingProteomicsResults_files/libs/quarto-html/quarto.js"></script>
<script src="processingProteomicsResults_files/libs/quarto-html/popper.min.js"></script>
<script src="processingProteomicsResults_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="processingProteomicsResults_files/libs/quarto-html/anchor.min.js"></script>
<link href="processingProteomicsResults_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="processingProteomicsResults_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="processingProteomicsResults_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="processingProteomicsResults_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="processingProteomicsResults_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Sorensen RNA-seq data processing</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Christopher Hughes </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<section id="details" class="level2">
<h2 class="anchored" data-anchor-id="details">Details</h2>
<p>This document describes processing of proteomics data for Ashish related to lysosome proteomics. These are SILAC data. I processed them using DIA-NN with the following protocol:</p>
<p>Specify a spectral library - can be an empirical one or an in silico predicted one. Then add the following commands to the ‘Additional options’ textbox: –fixed-mod SILAC,0.0,KR,label –lib-fixed-mod SILAC –channels SILAC,L,KR,0:0; SILAC,H,KR,8.014199:10.008269 –peak-translation –original-mods</p>
<p>–fixed-mod here declares SILAC as a label –lib-fixed-mod in silico labels the library with it. As masses are 0, it just adds SILAC to precursor names –channels declares channels. Channel ‘L’ is declared with masses 0 on K and 0 on R. Channel ‘H’ with masses 8 on K and 10 on R. –relaxed-prot-inf - as usual, recommended for benchmarking if the library does not have protein groups assembled properly already.</p>
<p>Report: Precursor.Translated - special quantity calculated when having multiple channels - most likely the best one. If you need protein quantities, need to calculate those based on Precursor.Translated with MaxLFQ, either using iq R package (recommended) or diann R package. Channel.Q.Value - probability that a given precursor is indeed present in the channel Translated.Quality - reflects how much evidence there was for the particular Precursor.Translated (the quantity I suggest to use), that is how many fragments could be use to calculate ratios between channels.</p>
<p>–matrix-ch-qvalue and –matrix-tr-qvalue set the channel and translated q-value filtering for the output matrices. These commands can be useful if you use the matrices as the output.</p>
</section>
<section id="setting-up-the-environment" class="level2">
<h2 class="anchored" data-anchor-id="setting-up-the-environment">Setting up the environment</h2>
<p>These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'tidyverse'</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'iq'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>generalDatasets <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>baseWorkspace <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToOthers'</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>baseRepository <span class="ot">=</span> <span class="st">'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToOthers'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-processing" class="level2">
<h2 class="anchored" data-anchor-id="data-processing">Data processing</h2>
<p>Read in data from DIA-NN and process with the IQ package.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>raw <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'D:/msDataTemp/proteomics20221006_ashishLysoIpSilac/report.tsv'</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">File.Name =</span> <span class="fu">sub</span>(<span class="st">'.*8_(.*)_diaBoth</span><span class="sc">\\</span><span class="st">.mzML'</span>, <span class="st">'</span><span class="sc">\\</span><span class="st">1'</span>, File.Name)) <span class="sc">%&gt;%</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Channel.Q.Value <span class="sc">&lt;=</span> <span class="fl">0.01</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Protein.Group =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">'-L'</span>, Precursor.Id), <span class="fu">paste</span>(Protein.Group,<span class="st">'_light'</span>,<span class="at">sep=</span><span class="st">''</span>),</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">'-M'</span>, Precursor.Id), <span class="fu">paste</span>(Protein.Group,<span class="st">'_medium'</span>,<span class="at">sep=</span><span class="st">''</span>), <span class="fu">paste</span>(Protein.Group,<span class="st">'_heavy'</span>,<span class="at">sep=</span><span class="st">''</span>)))) <span class="sc">%&gt;%</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Genes =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">'-L'</span>, Precursor.Id), <span class="fu">paste</span>(Genes,<span class="st">'_light'</span>,<span class="at">sep=</span><span class="st">''</span>),</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">'-M'</span>, Precursor.Id), <span class="fu">paste</span>(Genes,<span class="st">'_medium'</span>,<span class="at">sep=</span><span class="st">''</span>), <span class="fu">paste</span>(Genes,<span class="st">'_heavy'</span>,<span class="at">sep=</span><span class="st">''</span>))))</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(raw, <span class="st">'D:/msDataTemp/proteomics20221006_ashishLysoIpSilac/report-parsed.tsv'</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>test <span class="ot">=</span> <span class="fu">process_long_format</span>(<span class="st">'D:/msDataTemp/proteomics20221006_ashishLysoIpSilac/report-parsed.tsv'</span>, </span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>                           <span class="at">output_filename =</span> <span class="st">'D:/msDataTemp/proteomics20221006_ashishLysoIpSilac/report-pgFromIq.tsv'</span>,</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>                           <span class="at">annotation_col =</span> <span class="st">'Genes'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Command: --sample File.Name --primary Protein.Group --secondary Precursor.Id --quant Fragment.Quant.Corrected --annotation Genes --filter-double-less Q.Value 0.01 --filter-double-less PG.Q.Value 0.01 --intensity_col_sep ; --na_string 0 D:/msDataTemp/proteomics20221006_ashishLysoIpSilac/report-parsed.tsv 

Sample column:
    File.Name
Protein column:
    Protein.Group
Ion column(s):
    Precursor.Id
Quant column:
    Fragment.Quant.Corrected
Annotation column(s):
    Genes
Double less filter(s):
    Q.Value &lt; 0.010000
    PG.Q.Value &lt; 0.010000
Quant column separator:
    ;
Quant column NA string (in addition to NaN):
    0

Using 4 threads ...

# lines read (excluding headers)      = 360164
# quantitative values after filtering = 2719514

# samples  = 5
# proteins = 13947</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Removing low intensities...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Barplotting raw data ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Median normalization ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Barplotting after normalization ...</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>nrow = 2719497, # proteins = 13947, # samples = 5
Using 15 threads...
0%
5%
10%
15%
20%
26%
31%
36%
41%
46%
51%
57%
62%
67%
73%
78%
83%
88%
93%
98%
Completed.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Writing to: D:/msDataTemp/proteomics20221006_ashishLysoIpSilac/report-pgFromIq.tsv</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>raw.parsed <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'D:/msDataTemp/proteomics20221006_ashishLysoIpSilac/report-pgFromIq.tsv'</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">accession =</span> <span class="fu">sub</span>(<span class="st">'(.*)_.*'</span>, <span class="st">'</span><span class="sc">\\</span><span class="st">1'</span>, Protein.Group),</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">symbol =</span> <span class="fu">sub</span>(<span class="st">'(.*)_.*'</span>, <span class="st">'</span><span class="sc">\\</span><span class="st">1'</span>, Genes),</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">quantLabel =</span> <span class="fu">sub</span>(<span class="st">'.*_(.*)'</span>, <span class="st">'</span><span class="sc">\\</span><span class="st">1'</span>, Protein.Group)) <span class="sc">%&gt;%</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(accession, symbol, quantLabel, rep1, rep2, rep3, rep4, rep5)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(raw.parsed, <span class="st">'D:/msDataTemp/proteomics20221006_ashishLysoIpSilac/report-parsed-quants.tsv'</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Done with these data for now.</p>
<section id="session-info" class="level3">
<h3 class="anchored" data-anchor-id="session-info">Session info</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################################</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_Canada.utf8  LC_CTYPE=English_Canada.utf8   
[3] LC_MONETARY=English_Canada.utf8 LC_NUMERIC=C                   
[5] LC_TIME=English_Canada.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] iq_1.9.7        forcats_0.5.2   stringr_1.5.0   dplyr_1.0.10   
 [5] purrr_1.0.0     readr_2.1.3     tidyr_1.2.1     tibble_3.1.8   
 [9] ggplot2_3.4.0   tidyverse_1.3.2

loaded via a namespace (and not attached):
 [1] lubridate_1.9.0     assertthat_0.2.1    digest_0.6.31      
 [4] utf8_1.2.2          R6_2.5.1            cellranger_1.1.0   
 [7] backports_1.4.1     reprex_2.0.2        evaluate_0.19      
[10] httr_1.4.4          pillar_1.8.1        rlang_1.0.6        
[13] googlesheets4_1.0.1 readxl_1.4.1        rstudioapi_0.14    
[16] rmarkdown_2.19      googledrive_2.0.0   htmlwidgets_1.6.1  
[19] bit_4.0.5           munsell_0.5.0       broom_1.0.2        
[22] compiler_4.2.2      modelr_0.1.10       xfun_0.36          
[25] pkgconfig_2.0.3     htmltools_0.5.4     tidyselect_1.2.0   
[28] fansi_1.0.3         crayon_1.5.2        tzdb_0.3.0         
[31] dbplyr_2.2.1        withr_2.5.0         grid_4.2.2         
[34] jsonlite_1.8.4      gtable_0.3.1        lifecycle_1.0.3    
[37] DBI_1.1.3           magrittr_2.0.3      scales_1.2.1       
[40] cli_3.5.0           stringi_1.7.8       vroom_1.6.0        
[43] fs_1.5.2            xml2_1.3.3          ellipsis_0.3.2     
[46] generics_0.1.3      vctrs_0.5.1         tools_4.2.2        
[49] bit64_4.0.5         glue_1.6.2          hms_1.1.2          
[52] parallel_4.2.2      fastmap_1.1.0       yaml_2.3.6         
[55] timechange_0.1.1    colorspace_2.0-3    gargle_1.2.1       
[58] rvest_1.0.3         knitr_1.41          haven_2.5.1        </code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>