---
title: "Analysis of published YBX1 PARCLIP data"
output:
  html_notebook:
      code_folding: none
---

This document describes exploration of some PARCLIP data from a published study (PMID: 31358969). The paper performed a few different analyses related to YBX1, but here we will focus on their PARCLIP data that was performed in duplicate. The raw data for the PARCLIP experiments were uploaded to GSE133671, but unfortunately, they only give basic peak call files that are lacking any sort of scores or information on how they were parsed. As a result, I reprocessed the raw sequencing data using the script found [here](/rawSequencingDataProcessingShellScript.md).

## Linked analysis

This file is a continuation of the work done in:

* [20200526_explorationYbx1ParclipPublished](https://github.com/chrishuges/projectsWorkspace/tree/master/sorensenLab/relatedToYbx1/20200526_explorationYbx1ParclipPublished)

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, echo = FALSE}
library('tidyverse')
library('GenomicFeatures')
library('ggplot2')
library('RColorBrewer')
library('org.Hs.eg.db')
library('GenomicRanges')
library('RMariaDB')
library('bamsignals')
```

I want to set a base directory that we can use as a link to the directory where we will do most of the work.

```{r}
baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

Lastly, we need to call a source file that has some functions we will need. For simplicity and ease of tracking, I make a copy of this file in each directory that I will work in in order to be able to trace the exact one functions that are used, in case I change them and mistakenly don't push it to GitHub.

```{r}
source(paste(baseWorkspace, '/20200526_explorationYbx1ParclipPublished/userDefinedFunctions.R', sep = ''))
```

## Data processing

First we need to build a transcriptome index that we will use for gene assignment in one of the referenced functions. Use the help function for this to find a server that is close to you, it makes a big difference in the speed. Ensembl v99 is the last one for hg19 and this is what we used in the alignment for ease of comparing to other published studies.

```{r}
txdbEnsembl = makeTxDbFromBiomart(dataset = 'hsapiens_gene_ensembl',
                                  host = 'uswest.ensembl.org')
#now save the txdb for later use
saveDb(txdbEnsembl, file = paste(baseRepository, 
                                 '/20200526_explorationYbx1ParclipPublished/TxDb.Hsapiens.ENSEMBL.hg19.v99.sqlite', sep = ''))
```

Annotate the YBX1 PARCLIP data that was obtained from wavClusteR using functions defined in the source file referenced above.

```{r}
#########################################################################################
##get the data for the two replicates
parclip1 = readRDS(paste(baseRepository, '/20200526_explorationYbx1ParclipPublished/SRR9623531.wavClusters.rds', sep = ''))
parclip2 = readRDS(paste(baseRepository, '/20200526_explorationYbx1ParclipPublished/SRR9623532.wavClusters.rds', sep = ''))

##overlap the replicates to only retain regions identified in both
parclipIntersect = GenomicRanges::intersect(parclip1, parclip2)

##in case you don't have the txdb loaded in already
#txdbEnsembl = loadDb(file = paste(baseRepository, 
#                                  '/20200526_explorationYbx1ParclipPublished/TxDb.Hsapiens.ENSEMBL.hg19.v99.sqlite', sep = ''))
txdbEnsembl = loadDb(file = 'C:/Users/chris/OneDrive/Documents/bccrc/databases/txdb/TxDb.Hsapiens.ENSEMBL.hg19.v99.sqlite')
##now annotate this with our user-defined function
parclipAnnotated = wavclusterPeakAnnotation(as_tibble(parclipIntersect), txdbEnsembl, removeMito = TRUE)
```

Remove clusters that are too small and add read counts based on the replicates.

```{r}
############################################################################
##remove clusters that are too small
parclipLargeClusters = subset(parclipAnnotated, parclipAnnotated$width >= 12)

##read the raw bam files
bampath1 = paste(baseRepository, '/20200526_explorationYbx1ParclipPublished/SRR9623531.trimmed.rrna.bowtie.mapped.sorted.bam', sep = '')
bampath2 = paste(baseRepository, '/20200526_explorationYbx1ParclipPublished/SRR9623532.trimmed.rrna.bowtie.mapped.sorted.bam', sep = '')

##get the max read count in a region, replicate 1
bamCoverage1 = bamCoverage(bampath1, makeGRangesFromDataFrame(parclipLargeClusters), verbose=FALSE)
bamCounts1 = unlist(lapply(as.list(bamCoverage1), max))
parclipLargeClusters$readCountRep1 = bamCounts1

##get the max read count in a region, replicate 2
bamCoverage2 = bamCoverage(bampath2, makeGRangesFromDataFrame(parclipLargeClusters), verbose=FALSE)
bamCounts2 = unlist(lapply(as.list(bamCoverage2), max))
parclipLargeClusters$readCountRep2 = bamCounts2

##combine with the original data and take the sum of the replicates
parclipLargeClusters$readCountSum = as.integer(rowSums(parclipLargeClusters[,c('readCountRep1','readCountRep2')], na.rm = TRUE))
```

Now annotate the location in the read peak where the maximum read count is observed, for each peak.

```{r}
############################################################################
###find the max location for each peak
bamMax1 = unlist(lapply(as.list(bamCoverage1), function(x) as.integer(mean(which(x == max(x))))))
bamMax2 = unlist(lapply(as.list(bamCoverage2), function(x) as.integer(mean(which(x == max(x))))))
parclipLargeClusters$peakMaxRep1 = parclipLargeClusters$start + bamMax1
parclipLargeClusters$peakMaxRep2 = parclipLargeClusters$start + bamMax2
```

Now annotate the purity for each gene. This is a bit rough, but for all the peaks in each gene, it basically calculates what proportion they contribute to the total signal of that gene. If a gene just has non-specific reads across it, I expect that the peaks will all have low proportion scores. If there are a couple of large peaks, then I expect these to rise above and have higher scores. It is not perfect, and I can envision a couple scenarios where it will fall down, but I think as long as I am not too strict in the filter level it should be ok.

```{r}
############################################################################
##find the purity for all peaks within a gene
gnSet = unique(parclipLargeClusters$txdbSymbol)
parclipPurity = tibble()
for (i in 1:length(gnSet)){
  purityCalculation = parclipLargeClusters[parclipLargeClusters$txdbSymbol == gnSet[i],] %>%
    arrange(desc(readCountSum)) %>%
    mutate(peakPurity = readCountSum / sum(readCountSum)) 
  parclipPurity = rbind(parclipPurity, purityCalculation)
}

###make a quick plot of the purity values
ggplot(parclipPurity, aes(peakPurity)) +
  geom_density(size = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.1)) +
  labs(x = 'peak purity', y = 'density', title = 'peak purity calculation') +
  geom_vline(xintercept = 0.25, linetype = 'dashed', color = brewer.pal(9,'Spectral')[1]) +
  theme_classic()
ggsave(paste(baseWorkspace, '/20200526_explorationYbx1ParclipPublished/density_yb1Parclip_peakPurityScores.pdf', sep = ''),
       useDingbats = FALSE,
       width = 4,
       height = 4)
```
Now I can assign the peak rank. This is essentially just the rank of all peaks assigned to a single gene, based on their read counts.

```{r}
############################################################################
##now assign the peak rank
parclipPurityFiltered = parclipPurity %>% filter(peakPurity >= 0.25)
gnSet = unique(parclipPurityFiltered$txdbSymbol)
parclipPeakRank = tibble()
for (i in 1:length(gnSet)){
  purityFilter = parclipPurityFiltered[parclipPurityFiltered$txdbSymbol == gnSet[i],] %>%
    arrange(desc(readCountSum)) %>%
    mutate(peakRank = seq(1,length(readCountSum),1))
  parclipPeakRank = rbind(parclipPeakRank, purityFilter)
}

###this is the final dataset, save it
saveRDS(parclipPeakRank,
        paste(baseRepository, '/20200526_explorationYbx1ParclipPublished/dataset_yb1ParclipAnnotatedPeakSet.rds', sep = ''))
write.table(parclipPeakRank,
        paste(baseRepository, '/20200526_explorationYbx1ParclipPublished/dataset_yb1ParclipAnnotatedPeakSet.csv', sep = ''),
        col.names = TRUE, row.names = FALSE, quote = FALSE, sep = ',')
```

## Session info

```{r}
sessionInfo()
```