---
title: "Comparing EwS signature genes with CRISPR survival"
output:
  html_notebook:
      code_folding: none
---

This document details the analysis of data from DepMap for dependency to overall survival in different studies.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning=FALSE}
##########################################################################################
library('tidyverse')
library('vroom')
library('survival')
library('survminer')
library('ggplot2')
library('RColorBrewer')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

Read in the gene set of interest from the Aynaud single cell RNAseq paper.

```{r}
##########################################################################################
ewsSet = read_tsv(paste(baseRepository, '/sequencing20210419_survivalAndBafComplexIntersection/aynaud_ewsSignatureSet.txt', sep = ''))
ewsSet
```

<div style="margin-bottom:50px;"></div>

Get the CRISPR data.

```{r}
##########################################################################################
anno = read_csv(paste(generalDatasets, '/depmap21Q1/sample_info.csv', sep = '')) %>%
  dplyr::select(DepMap_ID, stripped_cell_line_name, Subtype, lineage_subtype, lineage_molecular_subtype) %>%
  dplyr::filter(grepl('Ewing', Subtype) | grepl('Ewing', lineage_subtype))
crispr = vroom(paste(generalDatasets, '/depmap21Q1/Achilles_gene_effect.csv', sep = '')) %>%
  dplyr::filter(DepMap_ID %in% anno$DepMap_ID) %>%
  pivot_longer(cols = `A1BG (1)`:`ZZZ3 (26009)`, names_to = 'gene', values_to = 'rnaExp') %>%
  dplyr::mutate('symbol' = sub('(.*) \\([0-9]+\\).*$', '\\1', gene)) %>%
  dplyr::filter(symbol %in% ewsSet$symbol) %>%
  dplyr::select(symbol, rnaExp) %>%
  dplyr::group_by(symbol) %>%
  dplyr::summarise(medRnaExp = median(rnaExp, na.rm = TRUE))
dim(crispr)
```

<div style="margin-bottom:50px;"></div>

Get the survival data for the EwS signature genes.

```{r, message = FALSE}
##########################################################################################
esExpression = readRDS(paste(baseRepository, '/microarray20201203_dlg2SavolaSurvival/dataset_savolaExpression.rds', sep = ''))
esPhenotype = readRDS(paste(baseRepository, '/microarray20201203_dlg2SavolaSurvival/dataset_savolaPhenotype.rds', sep = ''))

##
survivalData = function(expressionData, phenotypeData, geneOfInterest){
  ##
  geneOfInterestExprs = expressionData %>%
    dplyr::filter(symbol == geneOfInterest)
  
  ##check for data
  if (nrow(geneOfInterestExprs) >= 1){
    #print('Gene found.')
    geneOfInterestExprs = expressionData %>%
      dplyr::filter(arraySymbolOne == geneOfInterest)
    geneOfInterestExprs$medExprs = apply(geneOfInterestExprs[,which(grepl('GS',colnames(geneOfInterestExprs)))[1]:ncol(geneOfInterestExprs)], 
                                       1, function(x) mean(x, na.rm = TRUE))
    geneOfInterestSort = geneOfInterestExprs %>%
      dplyr::arrange(desc(medExprs))
    ########
    #keep the top probe for the gene, based on the expression calculated above
    geneSurvivalInput = geneOfInterestSort[1,1:(ncol(geneOfInterestSort) - 1)] %>%
      pivot_longer(cols = colnames(geneOfInterestSort)[which(grepl('GS',colnames(geneOfInterestSort)))[1]]:colnames(geneOfInterestSort)[(ncol(geneOfInterestSort) - 1)], 
                 names_to = 'geo_accession', values_to = 'rnaExprs') %>%
      right_join(phenotypeData) %>%
      dplyr::arrange(rnaExprs)
      ########
    geneCutPoint = surv_cutpoint(geneSurvivalInput,
                               time = 'ovs',
                               event = 'status',
                               variables = 'rnaExprs')
    geneSurvivalCat = surv_categorize(geneCutPoint)
    geneSurvivalInput$geneLevel = geneSurvivalCat$rnaExprs
    geneSurvivalInput$geneLevel = factor(geneSurvivalInput$geneLevel, levels = c('low','high'))
    #calculation of the different survival metrics based on our data
    survivalFit = survfit(Surv(ovs, status) ~ geneLevel, data = geneSurvivalInput)
    coxStats = coxph(Surv(ovs, status) ~ geneLevel, data = geneSurvivalInput)
    coxZScore = coef(coxStats)/sqrt(diag(vcov(coxStats)))
    return(coxZScore)
  }
  
  
  ###
  if (nrow(geneOfInterestExprs) < 1){
    #print('Gene not found in first pass, searching secondary identifier annotation.')
    geneOfInterestExprs = expressionData %>%
      dplyr::filter(arraySymbolOne == geneOfInterest)
    
    if (nrow(geneOfInterestExprs) < 1){
      #print('Gene not found in second pass, searching third identifier annotation.')
      geneOfInterestExprs = expressionData %>%
        dplyr::filter(arraySymbolTwo == geneOfInterest)
    }
    
    if (nrow(geneOfInterestExprs) < 1){
      #print('Gene not found. Moving on to next entry.')
      return(NA)
    } else {
      geneOfInterestExprs$medExprs = apply(geneOfInterestExprs[,which(grepl('GS',colnames(geneOfInterestExprs)))[1]:ncol(geneOfInterestExprs)], 
                                       1, function(x) mean(x, na.rm = TRUE))
      geneOfInterestSort = geneOfInterestExprs %>%
        dplyr::arrange(desc(medExprs))
      ########
      #keep the top probe for the gene, based on the expression calculated above
      geneSurvivalInput = geneOfInterestSort[1,1:(ncol(geneOfInterestSort) - 1)] %>%
        pivot_longer(cols = colnames(geneOfInterestSort)[which(grepl('GS',colnames(geneOfInterestSort)))[1]]:colnames(geneOfInterestSort)[(ncol(geneOfInterestSort) - 1)], 
                 names_to = 'geo_accession', values_to = 'rnaExprs') %>%
        right_join(phenotypeData) %>%
        dplyr::arrange(rnaExprs)
        ########
      geneCutPoint = surv_cutpoint(geneSurvivalInput,
                               time = 'ovs',
                               event = 'status',
                               variables = 'rnaExprs')
      geneSurvivalCat = surv_categorize(geneCutPoint)
      geneSurvivalInput$geneLevel = geneSurvivalCat$rnaExprs
      geneSurvivalInput$geneLevel = factor(geneSurvivalInput$geneLevel, levels = c('low','high'))
      #calculation of the different survival metrics based on our data
      survivalFit = survfit(Surv(ovs, status) ~ geneLevel, data = geneSurvivalInput)
      coxStats = coxph(Surv(ovs, status) ~ geneLevel, data = geneSurvivalInput)
      coxZScore = coef(coxStats)/sqrt(diag(vcov(coxStats)))
      return(coxZScore)
    }
  }
}

##
survivalDataOutput = data.frame()
for (i in 1:length(ewsSet$symbol)){
  #print(paste('gene', i))
  survivalScore = data.frame('symbol' = ewsSet$symbol[i],
                             'survivalScore' = survivalData(esExpression, esPhenotype, ewsSet$symbol[i]))
  row.names(survivalScore) = NULL
  survivalDataOutput = rbind(survivalDataOutput, survivalScore)
}
```

<div style="margin-bottom:50px;"></div>

Link up with the CRISPR data. Look [here](https://www.r-bloggers.com/2016/12/cox-proportional-hazards-model/) for some info on the Z-score meaning.

```{r}
##########################################################################################
allData = crispr %>%
  left_join(survivalDataOutput)
dim(allData)

##
ggplot(allData, aes(survivalScore, medRnaExp)) +
  geom_point(size = 2, color = brewer.pal(8,'Spectral')[1]) +
  labs(x = 'Wald statistic', y = 'CRISPR gene effect', title = 'EWS-FLI1 survival') +
  scale_y_continuous(limits = c(-2.2, 1), breaks = seq(-2,1,0.5)) +
  scale_x_continuous(limits = c(-6,4), breaks = seq(-6,4,2)) +
  geom_hline(yintercept = -1, linetype = 'dashed') +
  geom_vline(xintercept = 0, linetype = 'dashed') +
  theme_classic()
ggsave(paste(baseRepository, '/sequencing20210419_survivalAndBafComplexIntersection/scatter_savolaSurvivalFitnessSignature.pdf', sep = ''),
       height = 4, width = 4, useDingbats = FALSE)
```


<div style="margin-bottom:50px;"></div>

Get the survival data for the EwS signature genes.

```{r, message = FALSE}
##########################################################################################
esExpression = readRDS(paste(baseRepository, '/microarray20201203_dlg2PostelVinaySurvival/dataset_postelVinayExpression.rds', sep = ''))
esPhenotype = readRDS(paste(baseRepository, '/microarray20201203_dlg2PostelVinaySurvival/dataset_postelVinayPhenotype.rds', sep = ''))

##
survivalDataOutput = data.frame()
for (i in 1:length(ewsSet$symbol)){
  #print(paste('gene', i))
  survivalScore = data.frame('symbol' = ewsSet$symbol[i],
                             'survivalScore' = survivalData(esExpression, esPhenotype, ewsSet$symbol[i]))
  row.names(survivalScore) = NULL
  survivalDataOutput = rbind(survivalDataOutput, survivalScore)
}

##
allData = crispr %>%
  left_join(survivalDataOutput)
dim(allData)

##
ggplot(allData, aes(survivalScore, medRnaExp)) +
  geom_point(size = 2, color = brewer.pal(8,'Spectral')[1]) +
  labs(x = 'Wald statistic', y = 'CRISPR gene effect', title = 'EWS-FLI1 survival') +
  scale_y_continuous(limits = c(-2.2, 1), breaks = seq(-2,1,0.5)) +
  scale_x_continuous(limits = c(-4,4), breaks = seq(-6,4,2)) +
  geom_hline(yintercept = -1, linetype = 'dashed') +
  geom_vline(xintercept = 0, linetype = 'dashed') +
  theme_classic()
ggsave(paste(baseRepository, '/sequencing20210419_survivalAndBafComplexIntersection/scatter_postelVinaySurvivalFitnessSignature.pdf', sep = ''),
       height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

## Session info

```{r}
##########################################################################################
sessionInfo()
```

































