---
title: "Coverage maps for DGL2"
output:
  html_notebook:
      code_folding: none
---

This document details working with aligned RNAseq data to create coverage maps for DLG2 using the NCBI annotation rather than ENSEMBL.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning = FALSE}
##########################################################################################
library('tidyverse')
library('ggplot2')
library('RColorBrewer')
library('rtracklayer')
library('wiggleplotr')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

First, read the GTF file that we can use to get the DLG2 annotation details.

```{r}
##########################################################################################
totalGtf = as.data.frame(rtracklayer::import.gff('D:/databases/projectEwsDlg2/refseqFiles/GRCh38_latest_genomic.gtf.gz'))
```

<div style="margin-bottom:50px;"></div>

Subset the GTF and make the coverage plot. The main transcript I am going to use is NM_001300983.1. 

```{r}
##########################################################################################
dlg2Gtf = totalGtf %>%
  filter(grepl('NM_001300983.1$', transcript_id)) %>%
  dplyr::rename(gene_name = gene_id, gene_id = seqnames) %>%
  mutate('seqnames' = 'chr11')
##
dlg2Meta = tibble('transcript_id' = 'NM_001300983.1',
                  'gene_id' = 'NC_000011.10',
                  'gene_name' = 'DLG2',
                  'strand' = '-')
##
dlg2Exons = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('exon', type)), split.field = 'transcript_id')
##
dlg2Cds = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('CDS', type)), split.field = 'transcript_id')
##
sampleData = tibble(sample_id = c(paste('SRR5163', seq(665,711,1), sep = ''), paste('SRR5163', seq(713,757,1), sep = '')), 
                    condition = factor(c(rep('ews',92)), levels = c('ews')),
                    scaling_factor = 1)
##
bigwigFiles = file.path(baseRepository,'sequencing20211109_ewsCohortDbgapPmid25010205/results',paste(sampleData$sample_id,'.sorted.bw',sep=''))
all(file.exists(bigwigFiles))

##
sampleData = sampleData %>%
  dplyr::mutate(bigWig = bigwigFiles)
as.data.frame(sampleData)
##
trackData = dplyr::mutate(sampleData, track_id = condition, colour_group = condition)
#trackData = dplyr::mutate(sampleData, track_id = sample_id, colour_group = condition)
##
plotCoverage(dlg2Exons, dlg2Cds, 
             dlg2Meta, trackData,
             heights = c(2,1), fill_palette = getGenotypePalette(),
             rescale_introns = TRUE)
ggsave(paste(baseRepository, '/sequencing20211109_ewsCohortDbgapPmid25010205/coverage_dlg2CoverageRefseqDbgapCohort.pdf', sep = ''),
         height = 2, width = 2, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

Now repeat this for a the short form of DLG2, NM_001377978.1. 

```{r}
##########################################################################################
dlg2Gtf = totalGtf %>%
  filter(grepl('NM_001377978.1$', transcript_id)) %>%
  dplyr::rename(gene_name = gene_id, gene_id = seqnames) %>%
  mutate('seqnames' = 'chr11')
##
dlg2Meta = tibble('transcript_id' = 'NM_001377978.1',
                  'gene_id' = 'NC_000011.10',
                  'gene_name' = 'DLG2',
                  'strand' = '-')
##
dlg2Exons = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('exon', type)), split.field = 'transcript_id')
##
dlg2Cds = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('CDS', type)), split.field = 'transcript_id')
##
sampleData = tibble(sample_id = c(paste('SRR5163', seq(671,711,1), sep = ''),paste('SRR5163', seq(713,757,1), sep = '')), 
                    condition = factor(c(rep('ews',86)), levels = c('ews')),
                    scaling_factor = 1)
##
bigwigFiles = file.path(baseRepository,'sequencing20211109_ewsCohortDbgapPmid25010205/results',paste(sampleData$sample_id,'.chr11.bw',sep=''))
all(file.exists(bigwigFiles))

##
sampleData = sampleData %>%
  dplyr::mutate(bigWig = bigwigFiles)
as.data.frame(sampleData)
##
trackData = dplyr::mutate(sampleData, track_id = condition, colour_group = condition)
#trackData = dplyr::mutate(sampleData, track_id = sample_id, colour_group = condition)
##
plotCoverage(dlg2Exons, dlg2Cds, 
             dlg2Meta, trackData,
             heights = c(2,1), fill_palette = getGenotypePalette(),
             rescale_introns = TRUE)
ggsave(paste(baseRepository, '/sequencing20211109_ewsCohortDbgapPmid25010205/coverage_dlg2ShortCoverageRefseqDbgapCohort.pdf', sep = ''),
         height = 2, width = 2, useDingbats = FALSE)
```


<div style="margin-bottom:50px;"></div>

I want to see if I can do this with line plots instead. 

```{r}
##########################################################################################
dlg2Gtf = totalGtf %>%
  filter(grepl('NM_001300983.1$', transcript_id)) %>%
  dplyr::rename(gene_name = gene_id, gene_id = seqnames) %>%
  mutate('seqnames' = 'chr11')
##
dlg2Meta = tibble('transcript_id' = 'NM_001300983.1',
                  'gene_id' = 'NC_000011.10',
                  'gene_name' = 'DLG2',
                  'strand' = '-')
##
dlg2Exons = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('exon', type)), split.field = 'transcript_id')
##
dlg2Cds = makeGRangesListFromDataFrame(dlg2Gtf %>%
  filter(grepl('CDS', type)), split.field = 'transcript_id')
##
sampleData = tibble(sample_id = c(paste('SRR5163', seq(665,711,1), sep = ''), paste('SRR5163', seq(713,757,1), sep = '')), 
                    condition = c(paste('ews',seq(1,92,1),sep='')),
                    scaling_factor = 1)
##
bigwigFiles = file.path(baseRepository,'sequencing20211109_ewsCohortDbgapPmid25010205/results',paste(sampleData$sample_id,'.sorted.bw',sep=''))
all(file.exists(bigwigFiles))

##
sampleData = sampleData %>%
  dplyr::mutate(bigWig = bigwigFiles)
as.data.frame(sampleData)
##
trackData = dplyr::mutate(sampleData, track_id = 'RNA-seq', colour_group = condition)
#trackData = dplyr::mutate(sampleData, track_id = sample_id, colour_group = condition)
##
test = plotCoverage(dlg2Exons, dlg2Cds, 
             dlg2Meta, trackData,
             heights = c(2,1), fill_palette = c(rep(brewer.pal(3,'Greys')[2],92)),
             rescale_introns = TRUE,
             coverage_type = 'line')
#ggsave(paste(baseRepository, '/sequencing20211109_ewsCohortDbgapPmid25010205/coverage_dlg2CoverageRefseqDbgapCohort.pdf', sep = ''),
#         height = 2, width = 2, useDingbats = FALSE)
```

## Session info

```{r}
##########################################################################################
sessionInfo()
```









