---
title: "Analysis of MG63.3 YBX1 GUC clones by proteomics"
output:
  html_notebook:
      code_folding: none
---

This document details the analysis of proteomics data acquired to compare MG63.3 clones created with the GUC guide targeting YBX1. The raw MS data were compiled to peptide identifications using ProteomeDiscoverer.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning=FALSE}
##########################################################################################
library('tidyverse')
library('ggplot2')
library('RColorBrewer')
library('vroom')
library('DEqMS')
library('pheatmap')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

Read the protein annotation details.

```{r}
##########################################################################################
proteinAnnotation = readRDS(paste(baseRepository, '/proteomics20200705_mg633CrisprYbx1GucClones/uniprotHumanJul2020.fasta.annotated.rds', sep = ''))
```

<div style="margin-bottom:50px;"></div>

Read the peptide results and parse to a more usable format. The channel 127C is empty.

```{r}
##########################################################################################
psm = vroom(paste(baseRepository, '/proteomics20201022_mg633Ybx1GucClones/ch_20201020_mg633Ybx1GucClones_TMT16_HpH_allFrac_PSMs.txt', sep = '')) %>%
  dplyr::select(`Protein Accessions`, `Annotated Sequence`, `SPS Mass Matches in Percent`, `Abundance 126`:`Abundance 127N`, `Abundance 128N`:`Abundance 134N`) %>%
  mutate(accession = sapply(str_split(`Protein Accessions`, ';'), '[', 1)) %>%
  mutate(sequence = toupper(sub('.*\\.(.*)\\..*$', '\\1', `Annotated Sequence`))) %>%
  filter(`SPS Mass Matches in Percent` >= 50) %>%
  dplyr::select(accession, sequence, `Abundance 126`:`Abundance 134N`) %>%
  left_join(proteinAnnotation) %>%
  dplyr::select(accession, gene, sequence, `Abundance 126`:`Abundance 134N`) %>%
  filter(!grepl('^sp$', accession)) %>%
  mutate(gene = gsub('-', NA, gene, fixed = TRUE)) %>%
  mutate(psms = 1)
colnames(psm)[4:18] = c('parent','guc6','guc3','guc7','guc5','guc14','guc24','guc9','guc21',
                                 'guc22','guc20','guc8','guc19','guc16','guc2')
```

<div style="margin-bottom:50px;"></div>

Filter the data to remove less confident PSMs.

```{r}
##########################################################################################
psmFiltered1 = subset(psm, rowSums(is.na(psm[,4:18])) < 12)
psmFiltered2 = subset(psmFiltered1, rowSums(psmFiltered1[,4:18], na.rm = TRUE) > 100)
psmFiltered2[,4:18] = log2(psmFiltered2[,4:18])
quantInputNormalized = medianSweeping(psmFiltered2[,c(1,3:18)], group_col = 1)
boxplot(quantInputNormalized, las = 2, ylab = 'log2 ratio', main = 'normalized TMT data')
```

<div style="margin-bottom:50px;"></div>

Extract the protein data from the normalized object.

```{r}
##########################################################################################
psmCounts = as.data.frame(table(psmFiltered2$gene))
colnames(psmCounts)[1] = 'gene'
pro = as_tibble(quantInputNormalized) %>%
  mutate(accession = row.names(quantInputNormalized)) %>%
  left_join(proteinAnnotation) %>%
  left_join(psmCounts)
```

<div style="margin-bottom:50px;"></div>

Look at YBX1 expression. It is a good diagnostic plot because it should be variable and reduced in the GUC clones.

```{r}
##########################################################################################
goi = pro %>%
  filter(grepl('YBX1', gene)) %>%
  pivot_longer(cols = parent:guc2, names_to = 'clone', values_to = 'expression')
##
ggplot(goi, aes(x = reorder(clone, -expression), expression)) +
  geom_point(size = 2, color = c(brewer.pal(3,'Greys')[2], rep(brewer.pal(3,'OrRd')[3],14))) +
  scale_y_continuous(limits = c(-4,4), breaks = seq(-4,4,2)) +
  labs(x = 'hetKO Clone', y = 'log2(Median intensity relative to Parent)', title = 'hetKO YB-1 abundance') +
  theme_classic()
ggsave(paste(baseRepository, '/proteomics20201022_mg633Ybx1GucClones/scatter_ybx1AbundanceAcrossClones.pdf', sep = ''),
       height = 2, width = 2, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

Take a quick look at YB-3 abundance, as well as some others.

```{r}
##########################################################################################
geneSet = c('YBX1','YBX3','CDH2','NUP210','STK26','TPD52','PSPC1','CCN2')
for (i in 1:length(geneSet)){
  print(i)
  goi = pro %>%
    filter(grepl(paste('^',geneSet[i],'$', sep = ''), gene)) %>%
    pivot_longer(cols = parent:guc2, names_to = 'clone', values_to = 'expression') %>%
    mutate(clone = factor(clone, levels = c('parent','guc5','guc24','guc3','guc9','guc22','guc14','guc16','guc7','guc19','guc21','guc8','guc2','guc6','guc20'))) %>%
    arrange(clone) %>%
    mutate(cloneRank = seq(1,15,1))
  ##
  ggplot(goi, aes(clone, expression)) +
    geom_point(size = 4, color = c(brewer.pal(8,'Greys')[7], rep(brewer.pal(8,'RdBu')[1],14)), alpha = 0.5) +
    scale_y_continuous(limits = c(-4,4), breaks = seq(-4,4,2)) +
    labs(x = 'GUC Clone', y = 'log2(Normalized protein abundance)', title = paste('GUC clone ',geneSet[i], ' abundance', sep = '')) +
    stat_smooth(aes(cloneRank, expression), method = lm, color = brewer.pal(8,'RdBu')[8]) +
    theme_classic()
  ggsave(paste(baseRepository, '/proteomics20201022_mg633Ybx1GucClones/scatter_', geneSet[i], 'AbundanceAcrossClonesWithModel.pdf', sep = ''),
         height = 4, width = 4, useDingbats = FALSE)
}
```

<div style="margin-bottom:50px;"></div>

Calculate an MSD and slope values relative to YB-1 to look for things that might be related.

```{r}
##########################################################################################
proScaled = ungroup(pro) %>%
  dplyr::select(accession, gene, parent,guc5,guc24,guc3,guc9,guc22,guc14,guc16,guc7,guc19,guc21,guc8,guc2,guc6,guc20)
yb1Set = proScaled[grepl('YBX1',proScaled$gene),]
proScaled$msd = apply(proScaled[,4:17], 1, function(x)
  sum((x - yb1Set[,4:17])^2)/14)
##
goiSlope = vector()
for(i in 1:nrow(pro)){
  goi = proScaled[i,'accession']
  goiData = proScaled %>%
    filter(accession == goi$accession) %>%
    pivot_longer(cols = parent:guc2, names_to = 'clone', values_to = 'expression') %>%
    mutate(clone = factor(clone, levels = c('parent','guc5','guc24','guc3','guc9','guc22','guc14','guc16','guc7','guc19','guc21','guc8','guc2','guc6','guc20'))) %>%
    arrange(clone) %>%
    mutate(cloneRank = seq(1,15,1))
  goiSlope[i] = lm(expression ~ cloneRank, data = goiData)$coefficients[2]
}
proScaled$slope = goiSlope
```

<div style="margin-bottom:50px;"></div>

Plot these slope and MSD data.

```{r}
##########################################################################################
ggplot(proScaled, aes(slope, -log10(msd)))+
  geom_point(size = 4, color = brewer.pal(8,'RdBu')[1], alpha = 0.5) +
  scale_y_continuous(limits = c(-1,1), breaks = seq(-1,1,0.25)) +
  scale_x_continuous(limits = c(-0.5,0.25), breaks = seq(-0.5,0.25,0.25)) +
  labs(x = 'Regression line slope', y = '-log10(Mean squared deviation)', title = 'GUC clone YBX1 relationship') +
  theme_classic()
ggsave(paste(baseRepository, '/proteomics20201022_mg633Ybx1GucClones/scatter_msdLinearModelYbx1.pdf', sep = ''),
         height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

Make a heatmap to just look at genes with high global variance.

```{r}
##########################################################################################
pro$cloneVariance = apply(pro[,1:15], 1, function(x) var(x, na.rm = TRUE))
proSort = pro %>%
  arrange(-cloneVariance) %>%
  filter(Freq > 1)

pheatmap(proSort[1:500,1:15], labels_row = proSort$gene, fontsize_row = 5)
```

<div style="margin-bottom:50px;"></div>

This is not super useful, so maybe it is better if we use specific gene sets. I am interested in cell cycle because these cells seem to have a defect here.

```{r}
##########################################################################################
geneSet = read_tsv(paste(generalDatasets, '/geneSets/hallmarkG2mCheckpoint.txt', sep = ''), skip = 1)
colnames(geneSet) = 'gene'
proGeneSet = pro[,c(1:15,17)] %>%
  filter(gene %in% geneSet$gene)
proGeneSet[,2:15] = apply(proGeneSet[,2:15], 2, function(x) x - proGeneSet$parent)
##
pheatmap(proGeneSet[,c(2,4,6,9:15)],
         color = colorRampPalette(rev(brewer.pal(8, name = "RdYlBu")))(50),
         labels_row = proGeneSet$gene, 
         fontsize_row = 2,
         cellwidth = 5,
         filename = paste(baseRepository, '/proteomics20201022_mg633Ybx1GucClones/heatmap_hallmarkCellCycle.pdf', sep = ''),
         height = 4, width = 4)
```

<div style="margin-bottom:50px;"></div>

Same thing, but for TP53 targets.

```{r}
##########################################################################################
geneSet = read_tsv(paste(generalDatasets, '/geneSets/hallmarkTp53Pathway.txt', sep = ''), skip = 1)
colnames(geneSet) = 'gene'
proGeneSet = pro[,c(1:15,17)] %>%
  filter(gene %in% geneSet$gene)
proGeneSet[,2:15] = apply(proGeneSet[,2:15], 2, function(x) x - proGeneSet$parent)
##
pheatmap(proGeneSet[,c(2,4,6,9:15)],
         color = colorRampPalette(rev(brewer.pal(8, name = "RdYlBu")))(50),
         labels_row = proGeneSet$gene, 
         fontsize_row = 2,
         cellwidth = 5,
         filename = paste(baseRepository, '/proteomics20201022_mg633Ybx1GucClones/heatmap_hallmarkTp53Pathway.pdf', sep = ''),
         height = 4, width = 4)
```

<div style="margin-bottom:50px;"></div>

Same thing, but for E2F1 targets.

```{r}
##########################################################################################
geneSet = read_tsv(paste(generalDatasets, '/geneSets/hallmakrE2fTarget.txt', sep = ''), skip = 1)
colnames(geneSet) = 'gene'
proGeneSet = pro[,c(1:15,17)] %>%
  filter(gene %in% geneSet$gene)
proGeneSet[,2:15] = apply(proGeneSet[,2:15], 2, function(x) x - proGeneSet$parent)
##
pheatmap(proGeneSet[,c(2,4,6,9:15)],
         color = colorRampPalette(rev(brewer.pal(8, name = "RdYlBu")))(50),
         labels_row = proGeneSet$gene, 
         fontsize_row = 2,
         cellwidth = 5,
         filename = paste(baseRepository, '/proteomics20201022_mg633Ybx1GucClones/heatmap_hallmarkE2fTarget.pdf', sep = ''),
         height = 4, width = 4)
```

<div style="margin-bottom:50px;"></div>

Same thing, but for TGFBeta targets.

```{r}
##########################################################################################
geneSet = read_tsv(paste(generalDatasets, '/geneSets/hallmarkTgfBetaSignaling.txt', sep = ''), skip = 1)
colnames(geneSet) = 'gene'
proGeneSet = pro[,c(1:15,17)] %>%
  filter(gene %in% geneSet$gene)
proGeneSet[,2:15] = apply(proGeneSet[,2:15], 2, function(x) x - proGeneSet$parent)
##
pheatmap(proGeneSet[,c(2,4,6,9:15)],
         color = colorRampPalette(rev(brewer.pal(8, name = "RdYlBu")))(50),
         labels_row = proGeneSet$gene, 
         fontsize_row = 2,
         cellwidth = 5,
         filename = paste(baseRepository, '/proteomics20201022_mg633Ybx1GucClones/heatmap_hallmarkTgfBetaSignaling.pdf', sep = ''),
         height = 4, width = 4)
```




















```{r}
##########################################################################################
cond = as.factor(c('high','high','high','low','low','low'))
design = model.matrix(~0+cond) 
colnames(design) = gsub("cond","",colnames(design))
```

<div style="margin-bottom:50px;"></div>

Now we can build the Limma model.

```{r}
##########################################################################################
geneMatrix = as.matrix(quantInputNormalized[,c(1,10,3,11,13,14)])
limmaFit1 = lmFit(geneMatrix, design)
limmaContrasts = c('high-low')
limmaContrastDesign =  makeContrasts(contrasts = limmaContrasts, levels = design)
limmaFit2 = eBayes(contrasts.fit(limmaFit1, contrasts = limmaContrastDesign))
```

<div style="margin-bottom:50px;"></div>

Now for the actual DEqMS work. We will first

```{r}
##########################################################################################
psmCountTable = as.data.frame(table(psmFiltered2$accession))
rownames(psmCountTable) = psmCountTable$Var1
limmaFit2$count = psmCountTable[rownames(limmaFit2$coefficients),2]
limmaFit3 = spectraCounteBayes(limmaFit2)
head(limmaFit3$coefficients)
##
deqmsResults = outputResult(limmaFit3, coef_col = 1)
colnames(deqmsResults)[7] = 'accession'
deqmsResults = left_join(deqmsResults, pro[,16:17])
#write.table(deqmsResults, 
#            paste(baseRepository, '/proteomics20200705_mg633CrisprYbx1GucClones/dataset_deqmsGuc4-Parent.csv', sep = ''),
#            sep = ',', row.names = FALSE, quote = FALSE)
```


