---
title: "Ewing polysomeSeq analysis"
output:
  html_notebook:
      code_folding: none
---

This document details the analysis of sequencing data from the Renata to study polysome changes in Ewing sarcoma tumours. 

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning=FALSE}
##########################################################################################
library('tidyverse')
library('RColorBrewer')
library('ggplot2')
library('tximport')
library('GenomicFeatures')
library('DESeq2')
library('org.Hs.eg.db')
library('ggrepel')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

Make the samples data frame.

```{r, message = FALSE}
##########################################################################################
samples = tibble('sample' = c(paste('P14057_10',seq(65,99,1),sep=''), paste('P14057_1',seq(100,102,1),sep='')),
                 'experiment' = rep(c('tot','poly'),19),
                 'treatment' = c(rep('entinostat',14), rep('control',12), rep('control',6), rep('entinostat',6)),
                 'type' = c(rep('cell',26), rep('cdx',12)))
samples$identifier = c(paste(samples$experiment,samples$type,samples$treatment,sep=''))

##
files = file.path(baseRepository, 'sequencing20210322_ewingPolysomeSeqRenata', paste(samples$sample, '_quant', sep = ''), 'quant.sf')
all(file.exists(files))

##
myTxdb = makeTxDbFromGFF('C:/Users/chughes/Documents/bccrc/databases/hg38/gencode_gtf/2230c535660fb4774114bfa966a62f823fdb6d21acf138d4.gtf.gz')
k = keys(myTxdb, keytype = 'TXNAME')
tx2gene = AnnotationDbi::select(myTxdb, k, 'GENEID', 'TXNAME')
```

<div style="margin-bottom:50px;"></div>

Now import the data to a transcriptome object and then export the counts and abundance data to files.

```{r}
##########################################################################################
txi = tximport(files, 
               type = 'salmon', 
               tx2gene = tx2gene)
names(txi)
head(txi$counts)

##
txExport = as.data.frame(txi$counts)
colnames(txExport) = samples$sample
txExport$ensembl = row.names(txExport)
ens.str = substr(rownames(txExport), 1, 15)
txExport$symbol = mapIds(org.Hs.eg.db,
                     keys=ens.str,
                     column="SYMBOL",
                     keytype="ENSEMBL",
                     multiVals="first")
saveRDS(txExport[,c(39:40,1:38)], 
        paste(baseRepository, '/sequencing20210322_ewingPolysomeSeqRenata/dataset_rawCountsSalmonAllData.rds', sep = ''))
write.table(txExport[,c(39:40,1:38)], 
          file = paste(baseRepository, '/sequencing20210322_ewingPolysomeSeqRenata/dataset_rawCountsSalmonAllData.tsv', sep = ''),
          quote = FALSE, col.names = TRUE, row.names = FALSE, sep = '\t')

##
txExport = as.data.frame(txi$abundance)
colnames(txExport) = samples$sample
txExport$ensembl = row.names(txExport)
ens.str = substr(rownames(txExport), 1, 15)
txExport$symbol = mapIds(org.Hs.eg.db,
                     keys=ens.str,
                     column="SYMBOL",
                     keytype="ENSEMBL",
                     multiVals="first")
saveRDS(txExport[,c(39:40,1:38)], 
        paste(baseRepository, '/sequencing20210322_ewingPolysomeSeqRenata/dataset_tpmSalmonAllData.rds', sep = ''))
write.table(txExport[,c(39:40,1:38)], 
          file = paste(baseRepository, '/sequencing20210322_ewingPolysomeSeqRenata/dataset_tpmSalmonAllData.tsv', sep = ''),
          quote = FALSE, col.names = TRUE, row.names = FALSE, sep = '\t')
```
<div style="margin-bottom:50px;"></div>

Now we can construct the DESeq2 dataset.

```{r}
##########################################################################################
ddsTxi = DESeqDataSetFromTximport(txi,
                                  colData = samples,
                                  design = ~ identifier)
##
dds = DESeq(ddsTxi)
keep = rowSums(counts(dds)) >= 10
dds = dds[keep,]
```

<div style="margin-bottom:50px;"></div>

Perform comparisons between samples.

```{r}
##########################################################################################
datasetFirst = 'totcellcontrol'
datasetSecond = 'totcellentinostat'
res = results(dds, contrast = c('identifier', datasetFirst, datasetSecond))
ens.str = substr(rownames(res), 1, 15)
res$symbol = mapIds(org.Hs.eg.db,
                     keys=ens.str,
                     column="SYMBOL",
                     keytype="ENSEMBL",
                     multiVals="first")
resOrdered = res[order(res$pvalue),]

##
saveRDS(as.data.frame(resOrdered), 
        paste(baseRepository, '/sequencing20210322_ewingPolysomeSeqRenata/dataset_deseq2_', datasetFirst, '-', datasetSecond, '.rds', sep = ''))
write.csv(as.data.frame(resOrdered), 
          file = paste(baseRepository, '/sequencing20210322_ewingPolysomeSeqRenata/dataset_deseq2_', datasetFirst, '-', datasetSecond, '.csv', sep = ''))

##
#resLFC = lfcShrink(dds, coef = paste('treatment_',datasetFirst,'_vs_',datasetSecond, sep = ''), type = "apeglm")
#resLFC
#plotMA(resLFC, ylim = c(-2,2))
rnaExp = as.data.frame(resOrdered)
rnaExp$logPValue = -log10(rnaExp$padj)
rnaExp$logPValueScaled = ifelse(rnaExp$logPValue > 12, 12, rnaExp$logPValue)
rnaExp$pColors = ifelse(rnaExp$padj <= 0.001 & rnaExp$log2FoldChange >= 1, brewer.pal(8,'Spectral')[1], 
                        ifelse(rnaExp$padj <= 0.001 & rnaExp$log2FoldChange <= -1, brewer.pal(8,'Spectral')[8], brewer.pal(11,'RdGy')[8]))
rnaExp$pText = ifelse(grepl('YBX1',rnaExp$symbol), rnaExp$symbol, '')
##
ggplot(rnaExp, aes(log2FoldChange, logPValueScaled)) +
  geom_point(size = 1, color = rnaExp$pColors, alpha = 0.75) +
  labs(x = paste('log2(',datasetFirst,' - ',datasetSecond,')', sep = ''), y = '-log10(Adjusted p-value)', title = paste(datasetFirst,' vs ',datasetSecond, sep='')) +
  geom_text_repel(label = rnaExp$pText, nudge_x = 2, nudge_y = -1) +
  scale_x_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) +
  scale_y_continuous(limits = c(0,12), breaks = seq(0,12,2)) +
  geom_vline(xintercept = c(-1,1), linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.001), linetype = 'dashed') +
  theme_classic()
##
ggsave(paste(baseRepository, '/sequencing20210322_ewingPolysomeSeqRenata/scatter_deseq2_', datasetFirst, '-', datasetSecond, '.pdf', sep = ''),
       height = 4, width = 4, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

Lets cross reference these data with some of the CLIPseq data that we have from Chen et al.

```{r}
##########################################################################################
tempData = readRDS(paste(baseRepository, '/sequencing20210322_ewingPolysomeSeqRenata/dataset_deseq2_polycellcontrol-polycellentinostat.rds', sep = ''))
##
clip = read_csv(paste(baseRepository, '/sequencing20200527_ybx1ParclipChen/dataset_yb1ParclipAnnotatedPeakSet.csv',sep = '')) %>%
  dplyr::select(txdbSymbol, region, readCountSum) %>%
  dplyr::rename(symbol = txdbSymbol) %>%
  dplyr::left_join(tempData)



ggplot(clip, aes(region, log2FoldChange)) +
  geom_boxplot() +
  scale_y_continuous(limits = c(-10,10))

table(clip$region)
```

<div style="margin-bottom:50px;"></div>

Lets compare trancriptome to translatome.

```{r}
##########################################################################################
poly = readRDS(paste(baseRepository, '/sequencing20210322_ewingPolysomeSeqRenata/dataset_deseq2_polycellcontrol-polycellentinostat.rds', sep = ''))
total = readRDS(paste(baseRepository, '/sequencing20210322_ewingPolysomeSeqRenata/dataset_deseq2_totcellcontrol-totcellentinostat.rds', sep = ''))

allData = total %>%
  dplyr::select(symbol, log2FoldChange) %>%
  dplyr::filter(!is.na(symbol)) %>%
  dplyr::rename(total = log2FoldChange) %>%
  left_join(poly) %>%
  dplyr::select(symbol, total, log2FoldChange) %>%
  dplyr::rename(poly = log2FoldChange)

##
allData$pColors = ifelse(allData$total > 2 & allData$poly > 2, brewer.pal(8,'Spectral')[1], 
                        ifelse(allData$total < -2 & allData$poly < -2, brewer.pal(8,'Spectral')[8], brewer.pal(11,'RdGy')[8]))
ggplot(allData, aes(total, poly)) +
  geom_point(size = 1, color = allData$pColors, alpha = 0.75) +
  labs(x = 'log2(total dmso - entinostat)', y = 'log2(polysome dmso - entinostat)', title = 'Entinostat translatome') +
  #geom_text_repel(label = rnaExp$pText, nudge_x = 2, nudge_y = -1) +
  scale_x_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) +
  scale_y_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) +
  geom_vline(xintercept = c(-2,2), linetype = 'dashed') +
  geom_hline(yintercept = c(-2,2), linetype = 'dashed') +
  theme_classic()
##
ggsave(paste(baseRepository, '/sequencing20210322_ewingPolysomeSeqRenata/scatter_deseq2_polysome-total.pdf', sep = ''),
       height = 4, width = 4, useDingbats = FALSE)

cor(allData$total, allData$poly, use='pairwise.complete.obs')

```










