---
title: "Looking in DepMap for DLG2"
output:
  html_notebook:
      code_folding: none
---

This document details the analysis of RNAseq data from DepMap.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning=FALSE}
##########################################################################################
library('tidyverse')
library('ggplot2')
library('RColorBrewer')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

Read the parsed RNA data from Depmap.

```{r}
##########################################################################################
rna = readRDS(paste(generalDatasets, '/depmap21Q4/CCLE_expression_parsed.rds', sep = ''))
```

<div style="margin-bottom:50px;"></div>

Process the RNA data for our gene of interest, removing lineages that don't have enough measurements to be of interest.

```{r}
##########################################################################################
goi = rna %>%
  filter(grepl('^DLG2$', gene)) %>%
  filter(!(lineage_subtype %in% names(which(table(lineage_subtype) <= 3)))) %>%
  filter(!is.na(lineage_subtype))
```

<div style="margin-bottom:50px;"></div>

Make a plot for DLG2 across the lineage_subtype column.

```{r}
##########################################################################################
ggplot(goi, aes(x = reorder(lineage_subtype, log2Tpm, median, na.rm = TRUE), log2Tpm)) +
  geom_boxplot(size = 0.5, outlier.shape = NA) +
  labs(x = 'Lineage subtype in DepMap', y = 'log2(Transcripts per million)', title = 'DLG2 RNA expression in DepMap') +
  scale_y_continuous(limits = c(0,8), breaks = seq(0,8,2)) +
  geom_hline(yintercept = 2, linetype = 'dashed') +
  theme_classic() +
  #theme(axis.text.x = element_blank())
  theme(axis.text.x = element_text(angle = 45, size = 6, hjust = 1))
ggsave(paste(baseRepository, '/database20201202_dlg2RnaExpressionDepMap/boxplot_dlg2RnaExpressionLineageSubtypeDepmap.pdf', sep = ''),
       height = 4, width = 6, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

I also want to make a plot of both fusions.

```{r}
##########################################################################################
goi = rna %>%
  filter(grepl('^DLG2$', gene)) %>%
  filter(lineage_subtype == 'Ewing_sarcoma') %>%
  filter(!is.na(lineage_molecular_subtype) & !grepl('FEV', lineage_molecular_subtype))
```

<div style="margin-bottom:50px;"></div>

Make a plot for DLG2 across the lineage_molecular_subtype column.

```{r}
##########################################################################################
ggplot(goi, aes(x = reorder(lineage_molecular_subtype, log2Tpm, median, na.rm = TRUE), log2Tpm)) +
  geom_boxplot(aes(fill = lineage_molecular_subtype), size = 0.5, outlier.shape = NA, width = 0.5) +
  geom_point(aes(color = lineage_molecular_subtype), size = 1, position = position_jitterdodge(jitter.width = 0.25)) +
  labs(x = 'Lineage subtype in DepMap', y = 'log2(TPM + 1)', title = 'DLG2 by fusion') +
  scale_y_continuous(limits = c(0,8), breaks = seq(0,8,2)) +
  scale_color_manual(values = c(rep(brewer.pal(3,'OrRd')[3],2))) +
  scale_fill_manual(values = c(rep(brewer.pal(3,'Greys')[2],2))) +
  #geom_hline(yintercept = 2, linetype = 'dashed') +
  theme_classic() +
  theme(legend.position = 'none')
ggsave(paste(baseRepository, '/database20201202_dlg2RnaExpressionDepMap/boxplot_dlg2RnaExpressionByFusionDepmap.pdf', sep = ''),
       height = 2, width = 2, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

Consolidate all low expression cases into a single box.

```{r}
##########################################################################################
tissuesOfInterest = c('Ewing_sarcoma', 'SCLC', 'non_hodgkin_lymphoma')
rnaSubset = goi %>%
  dplyr::mutate(tissueClass = ifelse(lineage_subtype %in% tissuesOfInterest, lineage_subtype, 'Other'))

##
ggplot(rnaSubset, aes(x = reorder(tissueClass, -log2Tpm, median, na.rm = TRUE), log2Tpm, fill = tissueClass)) +
  geom_violin(size = 0.25, width = 0.75) +
  #geom_boxplot(size = 0.5, outlier.shape = NA, width = 0.5) +
  labs(x = 'Tissue Specification in DepMap', y = 'log2(TPM + 1)', title = 'DLG2 in DepMap') +
  scale_fill_manual(values = c(rep(brewer.pal(3,'OrRd')[3],2), brewer.pal(3,'Greys')[2], rep(brewer.pal(3,'OrRd')[3],1))) +
  scale_y_continuous(limits = c(0,8), breaks = seq(0,8,2)) +
  theme_classic() +
  theme(legend.position = 'none')
ggsave(paste(baseRepository, '/database20201202_dlg2RnaExpressionDepMap/boxplot_dlg2RnaExpressionSmtsDepMapSubset.pdf', sep = ''),
       height = 2, width = 2, useDingbats = FALSE)
```

<div style="margin-bottom:50px;"></div>

One other thing I want to do is pull out the cell lines that contain expression for DLG2 so I can reprocess their RNAseq data in SRA. I downloaded a metadata table with all of the SRA accession details for PRJNA523380. 

```{r}
##########################################################################################
dlg2 = rna %>%
  dplyr::filter(grepl('^DLG2$', gene)) %>%
  dplyr::filter(log2Tpm >= 2)

##SRA table
sra = read_csv(paste(generalDatasets,'/depmap21Q4/SraRunTable.txt',sep = '')) %>%
  dplyr::select(Run, `Assay Type`, Cell_Line) %>%
  dplyr::rename(cell_line_name = 'Cell_Line')

##
dlg2Runs = dlg2 %>%
  dplyr::left_join(sra) %>%
  dplyr::filter(!is.na(Run))

##
write.table(dlg2Runs, paste(baseRepository,'/database20201202_dlg2RnaExpressionDepMap/dataset_sraRunsWithDlg2Expression.tsv', sep = ''),
            col.names = TRUE, row.names = FALSE, quote = FALSE, sep = '\t')
```


<div style="margin-bottom:50px;"></div>

## Session info

```{r}
##########################################################################################
sessionInfo()
```
