---
title: "Looking in DepMap for DLG2"
output:
  html_notebook:
      code_folding: none
---

This document details the analysis of RNAseq data from DepMap.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r, message = FALSE, warning=FALSE}
##########################################################################################
library('tidyverse')
library('ggplot2')
library('RColorBrewer')
```

<div style="margin-bottom:50px;"></div>

I want to set a base directory that we can use as a link to the directory where we will do most of the work. I use two directories here because the Workspace is what is pushed to GitHub and contains scripts and plot files, but the Repository is where more of the big data is stored that does not get pushed.

```{r}
##########################################################################################
generalDatasets = 'C:/Users/chughes/Documents/bccrc/projectsRepository/generalDatasets'
baseWorkspace = 'C:/Users/chughes/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToDlg2'
baseRepository = 'C:/Users/chughes/Documents/bccrc/projectsRepository/sorensenLab/relatedToDlg2'
#generalDatasets = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/generalDatasets'
#baseWorkspace = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsWorkspace/sorensenLab/relatedToYbx1'
#baseRepository = 'C:/Users/chris/OneDrive/Documents/bccrc/projectsRepository/sorensenLab/relatedToYbx1'
```

<div style="margin-bottom:50px;"></div>

## Data processing

Read the parsed RNA data from Depmap.

```{r}
##########################################################################################
rna = readRDS(paste(generalDatasets, '/depmap20Q4/CCLE_expression_parsed.rds', sep = ''))
```

<div style="margin-bottom:50px;"></div>

Process the RNA data for our gene of interest, removing lineages that don't have enough measurements to be of interest.

```{r}
##########################################################################################
goi = rna %>%
  filter(grepl('^DLG2$', gene)) %>%
  filter(!(lineage_subtype %in% names(which(table(lineage_subtype) <= 3)))) %>%
  filter(!is.na(lineage_subtype))
```

<div style="margin-bottom:50px;"></div>

Make a plot for DLG2 across the lineage_subtype column.

```{r}
##########################################################################################
ggplot(goi, aes(x = reorder(lineage_subtype, log2Tpm, median, na.rm = TRUE), log2Tpm)) +
  geom_boxplot(size = 0.5, outlier.shape = NA) +
  labs(x = 'Lineage subtype in DepMap', y = 'log2(Transcripts per million)', title = 'DLG2 RNA expression in DepMap') +
  scale_y_continuous(limits = c(0,8), breaks = seq(0,8,2)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 8))
ggsave(paste(baseRepository, '/database20201202_dlg2RnaExpressionDepMap/boxplot_dlg2RnaExpressionLineageSubtypeDepmap.pdf', sep = ''),
       height = 4, width = 6, useDingbats = FALSE)
```



<div style="margin-bottom:50px;"></div>

## Session info

```{r}
sessionInfo()
```
